{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## python script for processing RAD data in ``ustacks`` ##"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**PURPOSE** This script renames files from barcode name to sample name, and then generates a shell script that runs ``ustacks`` according to arguments you input at the command line."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**INPUTS** \n",
    "<br> python  \n",
    "{arg0}[filename.py]\n",
    "<br> {arg1}[barcodes & samples text file] \n",
    "<br> {arg2}[start directory] \n",
    "<br> {arg3}[end directory]\n",
    "<br><br> For example, <br><br>\n",
    "``$ python pypipe_ustacks.py barcodes_samplenames.txt raw_data processed_data``"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**OUTPUTS** \n",
    "<br> [1] Four files per one input file\n",
    "<br> [2] These files are: (1) alleles, (2) models, (3) snps, & (4) tags"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is what the script looks like:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "##########################################################################################\n",
    "# \n",
    "### ``ustacks``\n",
    "#\n",
    "# PURPOSE: This step aligns identical RAD tags within an individual into stacks & provides data for calling SNPs\n",
    "# INPUT: fastq or gzfastq files\n",
    "# OUTPUT: 4 files - alleles, modules, snps, tags\n",
    "# \n",
    "# \n",
    "### WHEN RUNNING THIS SCRIPT, YOUR INPUTS AT THE COMMAND LINE ARE:\n",
    "# python  \n",
    "# {0}[pipeline filename] \n",
    "# {1}[barcodes & samples textfile] \n",
    "# {2}[start directory] \n",
    "# {3}[end directory]\n",
    "# \n",
    "### DEPENDENCIES\n",
    "# \n",
    "# [1] You need a file where first column is barcode and second is unique sample name\n",
    "# \n",
    "### WARNINGS\n",
    "# \n",
    "# [1] ustacks only works when your working directory is one direcotry above the folders you are\n",
    "#\t\tcalling and storing data in \n",
    "# \n",
    "##########################################################################################\n",
    "\n",
    "# --- [A] call necessary modules\n",
    "\n",
    "import subprocess\n",
    "import sys \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# --- [B] Rename your files by the sample name\n",
    "\n",
    "new_file = open(\"new_filenames_shell.txt\", \"w\") # new txt file\n",
    "dir = sys.argv[2] # directory files that need names changed\n",
    "firststr = \"cd \" + dir + \"\\n\"\n",
    "new_file.write(firststr)\n",
    "\n",
    "myfile = open(sys.argv[1], \"r\")\t\t\t\t# myfile =  tab delimited file mentioned, called after your script\n",
    "\n",
    "for line in myfile:\t\t\t\t\t\t   # loop through each line\n",
    "\tlinelist=line.strip().split()\t\t   # splits into list by white space\n",
    "\tbarcodefile = linelist[0] + \"_1.fq.gz\"\n",
    "\t# print barcodefile \t\t\t\t   # troubleshooting loop\n",
    "\tsamplefile = linelist[1] + \"_1.fq.gz\"\n",
    "\t# print samplefile \t\t\t\t\t   # troubleshooting loop\n",
    "\tnewstring = \"mv\" + \"\\t\" + barcodefile + \"\\t\" + samplefile + \"\\n\"\n",
    "\t# print newstring  # troubleshooting loop\n",
    "\tnew_file.write(newstring)\n",
    "\n",
    "myfile.close()\n",
    "new_file.close()\n",
    "\n",
    "# run the script you just made as a shell script to rename your files\n",
    "subprocess.call(['sh new_filenames_shell.txt'], shell=True)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "### --- [C] Make shell script to run all samples through command line through ``ustacks``\n",
    "\n",
    "# ``ustacks`` requires an arbitrary integer for every sample, although unclear how it gets used as it does not become the name of the file\n",
    "\n",
    "\n",
    "# name your 'from' and 'to' directories that will go in each line of your ustacks shell script\n",
    "dirfrom = sys.argv[2]\n",
    "dirto = sys.argv[3]\n",
    "\n",
    "newfile2 = open(\"ustacks_shell.txt\", \"w\")\t # make ustacks shell script to run through terminal\n",
    "myfile = open(\"new_filenames_shell.txt\", \"r\")\t#open the file with a list of barcodes + sample IDs\n",
    "\n",
    "# dir = sys.argv[2] # directory files that need names changed\n",
    "# firststr = \"cd \" + dir + \"\\n\"\n",
    "# new_file.write(firststr)\n",
    "\n",
    "dir2 = sys.argv[2] # directory with files that we want to run ustacks on\n",
    "firststr_2 = \"cd \" + dir2 + \"\\n\"\n",
    "newfile2.write(firststr_2)\n",
    "\n",
    "ID_int = 001\t\t\t\t\t\t\t\t# start integer counter\n",
    "for line in myfile: \t\t\t#for each line in the barcode file\t\n",
    "\tlinelist=line.strip().split()\t\n",
    "\tsampID = linelist[2] \t\t\t\t\t#save the second object as \"sampID\"\n",
    "\tif ID_int < 10: \n",
    "\t\tustacks_code = \"ustacks -t gzfastq -f \" + dirfrom + \"/\" + sampID + \" -r -d -o \" + dirto + \" -i 00\" + str(ID_int) + \" -m 5 -M 3 -p 10\" + \"\\n\"\n",
    "\t\t\t\t\t\t\t\t#create a line of code for ustacks that includes the new sample ID (with 2 leading 0s)\n",
    "\telif ID_int >= 10 & ID_int < 100: \n",
    "\t\tustacks_code = \"ustacks -t gzfastq -f \" + dirfrom + \"/\" + sampID + \" -r -d -o \" + dirto + \" -i 0\" + str(ID_int) + \" -m 5 -M 3 -p 10\" + \"\\n\"\n",
    "\t\t\t\t\t\t\t\t#create a line of code for ustacks that includes the new sample ID (with 1 leading 0)\n",
    "\telse: \n",
    "\t\tustacks_code = \"ustacks -t gzfastq -f \" + dirfrom + \"/\" + sampID + \" -r -d -o \" + dirto + \" -i \" + str(ID_int) + \" -m 5 -M 3 -p 10\" + \"\\n\"\n",
    "\t\t\t\t\t\t\t\t#create a line of code for ustacks that includes the new sample ID (with no leading 0s)\n",
    "\tnewfile2.write(ustacks_code)\t#append this new line of code to the output file\n",
    "\tID_int += 1\n",
    "\n",
    "myfile.close()\n",
    "newfile2.close()\n",
    "\n",
    "# run this new script through the terminal\n",
    "subprocess.call(['sh ustacks_shell.txt'], shell=True)\n",
    "\n",
    "##########################################################################################\n",
    "\n",
    "### --- DOCUMENTATION FOR USTACKS\n",
    "# ustacks -t file_type -f file_path [-d] [-r] [-o path] [-i id] [-m min_cov] [-M max_dist] [-p num_threads] [-R] [-H] [-h]\n",
    "# t — input file Type. Supported types: fasta, fastq, gzfasta, or gzfastq.\n",
    "# f — input file path.\n",
    "# o — output path to write results.\n",
    "# i — SQL ID to insert into the output to identify this sample.\n",
    "# m — Minimum depth of coverage required to create a stack (default 2).\n",
    "# M — Maximum distance (in nucleotides) allowed between stacks (default 2).\n",
    "# N — Maximum distance allowed to align secondary reads to primary stacks (default: M + 2).\n",
    "# R — retain unused reads.\n",
    "# H — disable calling haplotypes from secondary reads.\n",
    "# p — enable parallel execution with num_threads threads.\n",
    "# h — display this help messsage.\n",
    "# Stack assembly options:\n",
    "# \n",
    "# r — enable the Removal algorithm, to drop highly-repetitive stacks (and nearby errors) from the algorithm.\n",
    "# d — enable the Deleveraging algorithm, used for resolving over merged tags.\n",
    "# --max_locus_stacks [num] — maximum number of stacks at a single de novo locus (default 3).\n",
    "# --k_len [len] — specify k-mer size for matching between alleles and loci (automatically calculated by default).\n",
    "# Gapped assembly options:\n",
    "# \n",
    "# --gapped — preform gapped alignments between stacks.\n",
    "# --max_gaps — number of gaps allowed between stacks before merging (default: 2).\n",
    "# --min_aln_len — minimum length of aligned sequence in a gapped alignment (default: 0.80).\n",
    "# Model options:\n",
    "# \n",
    "# --model_type [type] — either 'snp' (default), 'bounded', or 'fixed'\n",
    "# For the SNP or Bounded SNP model:\n",
    "# --alpha [num] — chi square significance level required to call a heterozygote or homozygote, either 0.1, 0.05 (default), 0.01, or 0.001.\n",
    "# For the Bounded SNP model:\n",
    "# --bound_low [num] — lower bound for epsilon, the error rate, between 0 and 1.0 (default 0).\n",
    "# --bound_high [num] — upper bound for epsilon, the error rate, between 0 and 1.0 (default 1).\n",
    "# For the Fixed model:\n",
    "# --bc_err_freq [num] — specify the barcode error frequency, between 0 and 1.0.\n",
    "##########################################################################################\n",
    "# \n",
    "### ``ustacks``\n",
    "#\n",
    "# PURPOSE: This step aligns identical RAD tags within an individual into stacks & provides data for calling SNPs\n",
    "# INPUT: fastq or gzfastq files\n",
    "# OUTPUT: 4 files - alleles, modules, snps, tags\n",
    "# \n",
    "# \n",
    "### WHEN RUNNING THIS SCRIPT, YOUR INPUTS AT THE COMMAND LINE ARE:\n",
    "# python  \n",
    "# {0}[pipeline filename] \n",
    "# {1}[barcodes & samples textfile] \n",
    "# {2}[start directory] \n",
    "# {3}[end directory]\n",
    "# \n",
    "### DEPENDENCIES\n",
    "# \n",
    "# [1] You need a file where first column is barcode and second is unique sample name\n",
    "# \n",
    "### WARNINGS\n",
    "# \n",
    "# [1] ustacks only works when your working directory is one direcotry above the folders you are\n",
    "#\t\tcalling and storing data in \n",
    "# \n",
    "##########################################################################################\n",
    "\n",
    "# --- [A] call necessary modules\n",
    "\n",
    "import subprocess\n",
    "import sys \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# --- [B] Rename your files by the sample name\n",
    "\n",
    "new_file = open(\"new_filenames_shell.txt\", \"w\") # new txt file\n",
    "dir = sys.argv[2] # directory files that need names changed\n",
    "firststr = \"cd \" + dir + \"\\n\"\n",
    "new_file.write(firststr)\n",
    "\n",
    "myfile = open(sys.argv[1], \"r\")\t\t\t\t# myfile =  tab delimited file mentioned, called after your script\n",
    "\n",
    "for line in myfile:\t\t\t\t\t\t   # loop through each line\n",
    "\tlinelist=line.strip().split()\t\t   # splits into list by white space\n",
    "\tbarcodefile = linelist[0] + \"_1.fq.gz\"\n",
    "\t# print barcodefile \t\t\t\t   # troubleshooting loop\n",
    "\tsamplefile = linelist[1] + \"_1.fq.gz\"\n",
    "\t# print samplefile \t\t\t\t\t   # troubleshooting loop\n",
    "\tnewstring = \"mv\" + \"\\t\" + barcodefile + \"\\t\" + samplefile + \"\\n\"\n",
    "\t# print newstring  # troubleshooting loop\n",
    "\tnew_file.write(newstring)\n",
    "\n",
    "myfile.close()\n",
    "new_file.close()\n",
    "\n",
    "# run the script you just made as a shell script to rename your files\n",
    "subprocess.call(['sh new_filenames_shell.txt'], shell=True)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "### --- [C] Make shell script to run all samples through command line through ``ustacks``\n",
    "\n",
    "# ``ustacks`` requires an arbitrary integer for every sample, although unclear how it gets used as it does not become the name of the file\n",
    "\n",
    "\n",
    "# name your 'from' and 'to' directories that will go in each line of your ustacks shell script\n",
    "dirfrom = sys.argv[2]\n",
    "dirto = sys.argv[3]\n",
    "\n",
    "newfile2 = open(\"ustacks_shell.txt\", \"w\")\t # make ustacks shell script to run through terminal\n",
    "myfile = open(\"new_filenames_shell.txt\", \"r\")\t#open the file with a list of barcodes + sample IDs\n",
    "\n",
    "# dir = sys.argv[2] # directory files that need names changed\n",
    "# firststr = \"cd \" + dir + \"\\n\"\n",
    "# new_file.write(firststr)\n",
    "\n",
    "dir2 = sys.argv[2] # directory with files that we want to run ustacks on\n",
    "firststr_2 = \"cd \" + dir2 + \"\\n\"\n",
    "newfile2.write(firststr_2)\n",
    "\n",
    "ID_int = 001\t\t\t\t\t\t\t\t# start integer counter\n",
    "for line in myfile: \t\t\t#for each line in the barcode file\t\n",
    "\tlinelist=line.strip().split()\t\n",
    "\tsampID = linelist[2] \t\t\t\t\t#save the second object as \"sampID\"\n",
    "\tif ID_int < 10: \n",
    "\t\tustacks_code = \"ustacks -t gzfastq -f \" + dirfrom + \"/\" + sampID + \" -r -d -o \" + dirto + \" -i 00\" + str(ID_int) + \" -m 5 -M 3 -p 10\" + \"\\n\"\n",
    "\t\t\t\t\t\t\t\t#create a line of code for ustacks that includes the new sample ID (with 2 leading 0s)\n",
    "\telif ID_int >= 10 & ID_int < 100: \n",
    "\t\tustacks_code = \"ustacks -t gzfastq -f \" + dirfrom + \"/\" + sampID + \" -r -d -o \" + dirto + \" -i 0\" + str(ID_int) + \" -m 5 -M 3 -p 10\" + \"\\n\"\n",
    "\t\t\t\t\t\t\t\t#create a line of code for ustacks that includes the new sample ID (with 1 leading 0)\n",
    "\telse: \n",
    "\t\tustacks_code = \"ustacks -t gzfastq -f \" + dirfrom + \"/\" + sampID + \" -r -d -o \" + dirto + \" -i \" + str(ID_int) + \" -m 5 -M 3 -p 10\" + \"\\n\"\n",
    "\t\t\t\t\t\t\t\t#create a line of code for ustacks that includes the new sample ID (with no leading 0s)\n",
    "\tnewfile2.write(ustacks_code)\t#append this new line of code to the output file\n",
    "\tID_int += 1\n",
    "\n",
    "myfile.close()\n",
    "newfile2.close()\n",
    "\n",
    "# run this new script through the terminal\n",
    "subprocess.call(['sh ustacks_shell.txt'], shell=True)\n",
    "\n",
    "##########################################################################################\n",
    "\n",
    "### --- DOCUMENTATION FOR USTACKS\n",
    "# ustacks -t file_type -f file_path [-d] [-r] [-o path] [-i id] [-m min_cov] [-M max_dist] [-p num_threads] [-R] [-H] [-h]\n",
    "# t — input file Type. Supported types: fasta, fastq, gzfasta, or gzfastq.\n",
    "# f — input file path.\n",
    "# o — output path to write results.\n",
    "# i — SQL ID to insert into the output to identify this sample.\n",
    "# m — Minimum depth of coverage required to create a stack (default 2).\n",
    "# M — Maximum distance (in nucleotides) allowed between stacks (default 2).\n",
    "# N — Maximum distance allowed to align secondary reads to primary stacks (default: M + 2).\n",
    "# R — retain unused reads.\n",
    "# H — disable calling haplotypes from secondary reads.\n",
    "# p — enable parallel execution with num_threads threads.\n",
    "# h — display this help messsage.\n",
    "# Stack assembly options:\n",
    "# \n",
    "# r — enable the Removal algorithm, to drop highly-repetitive stacks (and nearby errors) from the algorithm.\n",
    "# d — enable the Deleveraging algorithm, used for resolving over merged tags.\n",
    "# --max_locus_stacks [num] — maximum number of stacks at a single de novo locus (default 3).\n",
    "# --k_len [len] — specify k-mer size for matching between alleles and loci (automatically calculated by default).\n",
    "# Gapped assembly options:\n",
    "# \n",
    "# --gapped — preform gapped alignments between stacks.\n",
    "# --max_gaps — number of gaps allowed between stacks before merging (default: 2).\n",
    "# --min_aln_len — minimum length of aligned sequence in a gapped alignment (default: 0.80).\n",
    "# Model options:\n",
    "# \n",
    "# --model_type [type] — either 'snp' (default), 'bounded', or 'fixed'\n",
    "# For the SNP or Bounded SNP model:\n",
    "# --alpha [num] — chi square significance level required to call a heterozygote or homozygote, either 0.1, 0.05 (default), 0.01, or 0.001.\n",
    "# For the Bounded SNP model:\n",
    "# --bound_low [num] — lower bound for epsilon, the error rate, between 0 and 1.0 (default 0).\n",
    "# --bound_high [num] — upper bound for epsilon, the error rate, between 0 and 1.0 (default 1).\n",
    "# For the Fixed model:\n",
    "# --bc_err_freq [num] — specify the barcode error frequency, between 0 and 1.0.\n"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
