{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## python script for processing RAD data in ``cstacks`` ##"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**PURPOSE** This script performs line counts on your fasta or fastq files, then sorts them in descending order. Then, it picks a pre-determined subset of these to use to form a catalog of loci. Lastly, it generates a shell script to run ``cstacks`` on this subset of samples."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**INPUTS** \n",
    "<br> python  \n",
    "{arg0}[filename.py]\n",
    "<br> {arg1}[changed filenames file] \n",
    "<br> {arg2}[wd for data files] \n",
    "<br> {arg3}[# individuals you want in your catalog]\n",
    "<br> {arg4}[batch number]\n",
    "<br> {arg5}[output directory]\n",
    "<br> {arg6}[# mismatches allowed]\n",
    "<br> {arg7}[# threads to use]\n",
    "<br><br> For example, <br><br>\n",
    "``$ python pypipe_ustacks.py new_filenames.txt ../raw_data 10 2 ../processed_data 3 3``"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**OUTPUTS** \n",
    "<br> Three files per one input file\n",
    "<br> These files are: (1) catalog alleles, (2) catalog snps, & (3) catalog tags"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This script looks like:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "##########################################################################################\n",
    "# \n",
    "###--- ``cstacks`` script\n",
    "#\n",
    "# PURPOSE: ustacks creates a catalog from a subset of individuals to call SNPs\n",
    "# INPUT: ustacks out put files for specified number of individuals with most sequence reads\n",
    "# OUTPUT: catalog file + associated files\n",
    "# \n",
    "#### WHEN RUNNING THIS SCRIPT, INPUTS AT THE COMMAND LINE ARE:\n",
    "# python \n",
    "# {0}[pypipe_cstacks.py] \n",
    "# {1}[shell script with changed file names] \t\n",
    "# {2}[directory of input files] \n",
    "# {3}[# individuals for cstacks] \n",
    "# {4}[batch number] \n",
    "# {5}[output directory] \n",
    "# {6}[num mismatches allowed] \n",
    "# {7}[num threads]\n",
    "# \n",
    "### DEPENDENCIES: \n",
    "# [1] Your file names coming out of ustacks cannot have a period other than before file extension (not sure if true! check?)\n",
    "# \n",
    "### WARNINGS:\n",
    "# [1] If you have to rerun this script, it will append onto it! make sure no file w name or else won't even run!\n",
    "# \n",
    "##########################################################################################\n",
    "\n",
    "### --- [A] Call necessary modules\n",
    "\n",
    "import sys \n",
    "import subprocess\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "### --- [B] Count lines in each sequence file\n",
    "\n",
    "myfile = open(sys.argv[1], \"r\")\t#open the file with your list of barcodes and sample IDs\n",
    "\n",
    "dirfrom = sys.argv[2] # get directory for input files\n",
    "firststr = \"cd \" + \"sys.argv[2]\" + \"\\n\" # write first line of shell to cd to this directory\n",
    "\n",
    "filestring = \"\"\n",
    "samplename_list = [] # to be used in loop later in this script\n",
    "\n",
    "for line in myfile: \t\t\t\t\t#for each line in the barcode file\n",
    "\tlinelist = line.strip().split()\t\t#make a list of character strings broken by tabs\n",
    "\tsampID = linelist[2]\t\t\t\t#pick out file name\n",
    "\tsamplename_list.append(sampID)\n",
    "\tnewstring = \"gunzip -c \" + sampID + \" | wc -l >> cstacks_linecount.txt\\n\" # make line of code to run at command line\n",
    "\tfilestring += newstring # add to a list of strings we'll write to a file\n",
    "myfile.close()\n",
    "# print filestring # CHECK^\n",
    "\n",
    "\n",
    "filestring = firststr + filestring # write cd line before rest of file string\n",
    "\n",
    "#create a new file where the ustacks code will go, write string to file, close\n",
    "newfile = open(\"cstacks_linecount_shell.txt\", \"w\")\n",
    "newfile.write(filestring)\n",
    "newfile.close()\n",
    "\n",
    "# run shell script that will calculate line counts\n",
    "subprocess.call(['sh cstacks_linecount_shell.txt'], shell=True)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "### --- [C] Get sample names for specified # of samples with most sequence reads \n",
    "\n",
    "linecounts = open(\"cstacks_linecount.txt\", \"r\") # read in line counts file\n",
    "linecounts_list = [] # initiate a list for the line counts so I can get.item later\n",
    "\n",
    "for line in linecounts:\n",
    "\tcount = line.strip().split() # get line count\n",
    "\tlinecounts_list.append(line)\n",
    "\n",
    "list_samp_ct = [] # initiate empty list\n",
    "i = 0 # start counter\n",
    "\n",
    "# --- CHECK^ if lists look normal\n",
    "# print \"sample name list \"\n",
    "# print samplename_list\n",
    "# print \"line counts list \"\n",
    "# print linecounts_list\n",
    "\n",
    "for item in linecounts_list:\n",
    "\tnew_item = [samplename_list[i], linecounts_list[i]]\n",
    "\tlist_samp_ct.append(new_item)\n",
    "\ti += 1\n",
    "\t\n",
    "def getKey(item): # so that sorted will sort by second item in list\n",
    "\treturn item[1]\n",
    "sortedlist = sorted(list_samp_ct, key = getKey, reverse = True)\n",
    "# print sortedlist # CHECK^\n",
    "\n",
    "with open('all_sorted_name_counts.txt', 'w') as file:\n",
    "    file.writelines('\\t'.join(i) + '\\n' for i in sortedlist) # makes file\n",
    "    \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "### --- [D] Write and run shell script for ``cstacks``\n",
    "\n",
    "cstacks_shell = \"\"\n",
    "firststr = \"cstacks -b \" + sys.argv[4] + \" \"\n",
    "cstacks_shell += firststr\n",
    "\n",
    "endrange = int(sys.argv[3]) # set end of range for loop\n",
    "for i in range(0, endrange):\n",
    "\tfilename = samplename_list[i]\n",
    "\ttrmd_filename = filename.rsplit(\".\",2)[0]\n",
    "\t# print trmd_filename # CHECK^\n",
    "\tstring = \"-s \" + trmd_filename + \" \"\n",
    "\tcstacks_shell += string\n",
    "laststr = \"-o \" + sys.argv[5] + \" -n \" + sys.argv[6] + \" -p \" + sys.argv[7]\n",
    "cstacks_shell += laststr\n",
    "# print cstacks_shell # CHECK^\n",
    "\n",
    "cstacks_shell_txt = write(\"cstacks_shell.txt\", \"w\")\n",
    "cstacks_shell_txt.write(cstacks_shell)\n",
    "cstacks_shell_txt.close()\n",
    "\n",
    "# run shell script\n",
    "subprocess.call([\"cstacks_shell.txt\"], shell = True)\n",
    "\n",
    "##########################################################################################\n",
    "\n",
    "### --- DOCUMENTATION FOR CSTACKS\n",
    "\n",
    "# cstacks -b batch_id -s sample_file [-s sample_file_2 ...] [-o path] [-n num] [-g] [-p num_threads] [--catalog path] [-h]\n",
    "# p — enable parallel execution with num_threads threads.\n",
    "# b — MySQL ID of this batch.\n",
    "# s — TSV file from which to load radtags.\n",
    "# o — output path to write results.\n",
    "# m — include tags in the catalog that match to more than one entry.\n",
    "# n — number of mismatches allowed between sample tags when generating the catalog.\n",
    "# g — base catalog matching on genomic location, not sequence identity.\n",
    "# h — display this help messsage.\n",
    "# Catalog editing:\n",
    "# \n",
    "# --catalog [path] — provide the path to an existing catalog. cstacks will add data to this existing catalog.\n",
    "# Advanced options:\n",
    "# \n",
    "# --report_mmatches — report query loci that match more than one catalog locus.\n"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
