{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Purpose\n",
    "\n",
    "The purpose of this notebook is to document my full application of the Stacks pipeline on my cod data. I've been writing scripts to work on a subset of my files, and now I want to start running them on all of my files.\n",
    "\n",
    "## ``process_radtags`` \n",
    "\n",
    "I don't have to run this on my data because the previous user of the data already did. In my script for running ``process_radtags``, however, I also use this opportunity to make folders for the output of each of the Stacks programs. So I'm doing that manually here. Below, you can see these directories plus two folders of cod data (two lanes of data):\n",
    "\n",
    "![image](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Notebooks/images_for_notebooks/Screen%20Shot%202016-11-16%20at%203.10.45%20PM.png?raw=true)\n",
    "\n",
    "## ``ustacks`` \n",
    "\n",
    "I'll be working off of my hard drive, so first I had to change directory to there."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Time\\ Machine\\ Backups/Cod-Time-Series-Data/ "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Another issue is that I have two lanes. It'd be easiest if I could work in only one directory, but their barcodes are redundant. So my first step is to rename the files by adding *_L1* for Library 1 and *_L2* for Library 2. I'll do that in python before the file extension. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here's the content of the script ``add_lib_to_filename.py``:\n",
    "\n",
    "```\n",
    "# arguments\n",
    "# [1] directory name with files to rename \n",
    "# [2] text you would like to add to the end of each filename before file extensions\n",
    "\n",
    "# assumptions\n",
    "# [1] file extention = fq.qz\n",
    "# [2] you are one directory above the directory that you want to rename the files of\n",
    "# [3] your files aren't already renamed! --- write an if else loop to make sure it hasn't been run yet when you have the time\n",
    "\n",
    "import sys\n",
    "import subprocess\n",
    "\n",
    "# make a text file that has each thing in this directory on a line; each line should be each file in the lane\n",
    "\n",
    "string1 = \"\" # initiate string\n",
    "\n",
    "string1 += 'ls ' + sys.argv[1] + ' > ./dircontents.txt' # add ls, wd, redirect to text file\n",
    "\n",
    "firstshell = open(\"getcontents_shell.txt\", \"w\")\n",
    "firstshell.write(string1)\n",
    "firstshell.close()\n",
    "\n",
    "subprocess.call([\"sh getcontents_shell.txt\"], shell = True)\n",
    "\n",
    "# now that file is made for contents of directory, read that file in\n",
    "contents = open(\"dircontents.txt\", \"r\")\n",
    "\n",
    "\n",
    "# second shell = renaming script\n",
    "rename_w_lib_shell = open(\"rename_w_lib_shell.txt\", \"w\")\n",
    "\n",
    "string2 = \"\"\n",
    "string2 += \"cd \" + sys.argv[1] + \"\\n\"\n",
    "\n",
    "\n",
    "# loop that will break the script if it's already run to make sure you don't rename files into something wrong!\n",
    "for line in contents:\n",
    "\tlinelist = line.strip().split(\".\")\n",
    "\tif linelist[0][-4] == \"L\":\n",
    "\t\tprint \"CHECK TO SEE IF YOU HAVE ALREADY RENAMED THESE FILES. SCRIPT PAUSED AS WARNING BECAUSE FILES APPEAR RENAMED.\"\n",
    "\t\tsys.exit() # exit script\n",
    "\telse:\n",
    "\t\tcontinue\n",
    "contents.close()\n",
    "\n",
    "\n",
    "\n",
    "contents = open(\"dircontents.txt\", \"r\")\n",
    "for line in contents:\n",
    "\toriginal = line.strip()\n",
    "\tfilename_list = line.strip().split(\".\") # make list out of whole file name with extensions\n",
    "\tfilename_wo_ext = filename_list[0]\n",
    "\tfile_ext = \".\" + filename_list[1] + \".\" + filename_list[2]\n",
    "\twordlist = filename_wo_ext.split(\"_\") # split filename without extensions into parts by underscore\n",
    "\tnewfilename = wordlist[0] + sys.argv[2] + \"_\" + wordlist[1]\n",
    "\tnewfilename += file_ext\n",
    "\tstring2 += \"mv\" + \"\\t\" + original + \"\\t\" + newfilename + \"\\n\"\n",
    "\n",
    "rename_w_lib_shell.write(string2)\n",
    "rename_w_lib_shell.close()\n",
    "contents.close()\n",
    "\n",
    "# call renaming shell script\n",
    "subprocess.call([\"sh rename_w_lib_shell.txt\"], shell = True)\n",
    "```\n",
    "\n",
    "Running the code looks like this:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!python add_lib_to_filename.py process_radtags_out/cod_lib1 _L1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!python add_lib_to_filename.py process_radtags_out/cod_lib2 _L2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now my files are renamed with the library information after the barcode before their 1 or 2 that was used to signify forward and reverse for paired end data.\n",
    "<br><br>\n",
    "From\n",
    "<br>\n",
    "![image](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Notebooks/images_for_notebooks/Screen%20Shot%202016-11-16%20at%206.15.59%20PM.png?raw=true)\n",
    "<br>\n",
    "To\n",
    "![image](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Notebooks/images_for_notebooks/Screen%20Shot%202016-11-16%20at%206.15.48%20PM.png?raw=true)\n",
    "<br>\n",
    "<br>\n",
    "And then I can group the two directories of libraries together.\n",
    "<br>\n",
    "![image](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Notebooks/images_for_notebooks/Screen%20Shot%202016-11-17%20at%209.09.21%20AM.png?raw=true)\n",
    "<br>\n",
    "Now I want to run ``ustacks`` on all the files in this folder. I can use my handy dandy script I wrote a couple of weeks ago. It looks like this:\n",
    "\n",
    "```\n",
    "##########################################################################################\n",
    "# \n",
    "### ``ustacks``\n",
    "#\n",
    "# PURPOSE: This step aligns identical RAD tags within an individual into stacks & provides data for calling SNPs\n",
    "# INPUT: fastq or gzfastq files\n",
    "# OUTPUT: 4 files - alleles, modules, snps, tags\n",
    "# \n",
    "# \n",
    "### WHEN RUNNING THIS SCRIPT, YOUR INPUTS AT THE COMMAND LINE ARE:\n",
    "# python  \n",
    "# {0}[pipeline filename] \n",
    "# {1}[barcodes & samples textfile] \n",
    "# {2}[start directory] \n",
    "# {3}[end directory]\n",
    "# \n",
    "### DEPENDENCIES\n",
    "# \n",
    "# [1] You need a file where first column is barcode and second is unique sample name\n",
    "# \n",
    "### WARNINGS\n",
    "# \n",
    "# [1] ustacks only works when your working directory is one direcotry above the folders you are\n",
    "#\t\tcalling and storing data in \n",
    "# \n",
    "##########################################################################################\n",
    "\n",
    "# --- [A] call necessary modules\n",
    "\n",
    "import subprocess\n",
    "import sys \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# --- [B] Rename your files by the sample name\n",
    "\n",
    "new_file = open(\"new_filenames_shell.txt\", \"w\") # new txt file\n",
    "dir = sys.argv[2] # directory files that need names changed\n",
    "firststr = \"cd \" + dir + \"\\n\" + \"pwd\\n\"\n",
    "new_file.write(firststr)\n",
    "\n",
    "myfile = open(sys.argv[1], \"r\")\t\t\t\t# myfile =  tab delimited file mentioned, called after your script\n",
    "\n",
    "for line in myfile:\t\t\t\t\t\t   # loop through each line\n",
    "\tlinelist=line.strip().split()\t\t   # splits into list by white space\n",
    "\tbarcodefile1 = linelist[0] + \"_1.fq.gz\" # forward\n",
    "\tbarcodefile2 = linelist[0] + \"_2.fq.gz\" # reverse\n",
    "\tsamplefile1 = linelist[1] + \"_1.fq.gz\" # forward\n",
    "\tsamplefile2 = linelist[1] + \"_2.fq.gz\" # reverse\n",
    "\tnewstring = \"mv\" + \"\\t\" + barcodefile1 + \"\\t\" + samplefile1 + \"\\n\" + \"mv\" + \"\\t\" + barcodefile2 + \"\\t\" + samplefile2 + \"\\n\"\n",
    "\t# print newstring  # troubleshooting loop\n",
    "\tnew_file.write(newstring)\n",
    "\n",
    "myfile.close()\n",
    "new_file.close()\n",
    "\n",
    "# run the script you just made as a shell script to rename your files\n",
    "subprocess.call(['sh new_filenames_shell.txt'], shell=True)\n",
    "\n",
    " \n",
    "\n",
    "\n",
    "\n",
    "### --- [C] Make shell script to run all samples through command line through ``ustacks``\n",
    "\n",
    "# ``ustacks`` requires an arbitrary integer for every sample, although unclear how it gets used as it does not become the name of the file\n",
    "\n",
    "\n",
    "# name your 'from' and 'to' directories that will go in each line of your ustacks shell script\n",
    "dirfrom = sys.argv[2]\n",
    "dirto = sys.argv[3]\n",
    "\n",
    "newfile2 = open(\"ustacks_shell.txt\", \"w\")\t # make ustacks shell script to run through terminal\n",
    "myfile = open(\"new_filenames_shell.txt\", \"r\")\t#open the file with a list of barcodes + sample IDs\n",
    "\n",
    "# dir = sys.argv[2] # directory files that need names changed\n",
    "# firststr = \"cd \" + dir + \"\\n\"\n",
    "# new_file.write(firststr)\n",
    "\n",
    "dir2 = sys.argv[2] # directory with files that we want to run ustacks on\n",
    "\n",
    "ID_int = 001\t\t\t\t\t\t\t\t# start integer counter\n",
    "\n",
    "lines = myfile.readlines()[2:] # skip first two lines because just cd and pwd\n",
    "\n",
    "\n",
    "ID_int = 001\t\t\t\t\t\t\t\t# start integer counter\n",
    "for line in lines: \t\t\t#for each line in the barcode file\t\n",
    "\tlinelist=line.strip().split()\t\n",
    "\tsampID = linelist[2] \t\t\t\t\t#save the second object as \"sampID\"\n",
    "\tif ID_int < 10: \n",
    "\t\tustacks_code = \"ustacks -t gzfastq -f \" + dirfrom + \"/\" + sampID + \" -r -d -o \" + dirto + \" -i 00\" + str(ID_int) + \" -m 5 -M 3 -p 10\" + \"\\n\"\n",
    "\t\t\t\t\t\t\t\t#create a line of code for ustacks that includes the new sample ID (with 2 leading 0s)\n",
    "\telif ID_int >= 10 & ID_int < 100: \n",
    "\t\tustacks_code = \"ustacks -t gzfastq -f \" + dirfrom + \"/\" + sampID + \" -r -d -o \" + dirto + \" -i 0\" + str(ID_int) + \" -m 5 -M 3 -p 10\" + \"\\n\"\n",
    "\t\t\t\t\t\t\t\t#create a line of code for ustacks that includes the new sample ID (with 1 leading 0)\n",
    "\telse: \n",
    "\t\tustacks_code = \"ustacks -t gzfastq -f \" + dirfrom + \"/\" + sampID + \" -r -d -o \" + dirto + \" -i \" + str(ID_int) + \" -m 5 -M 3 -p 10\" + \"\\n\"\n",
    "\t\t\t\t\t\t\t\t#create a line of code for ustacks that includes the new sample ID (with no leading 0s)\n",
    "\tnewfile2.write(ustacks_code)\t#append this new line of code to the output file\n",
    "\tID_int += 1\n",
    "\n",
    "myfile.close()\n",
    "newfile2.close()\n",
    "\n",
    "# run this new script through the terminal\n",
    "subprocess.call(['sh ustacks_shell.txt'], shell=True)\n",
    "\n",
    "##########################################################################################\n",
    "\n",
    "```\n",
    "\n",
    "And I run it like this:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data/process_radtags_out\n",
      "mv: AAACGG_L1_1.fq.gz: No such file or directory\n",
      "mv: AAACGG_L1_2.fq.gz: No such file or directory\n",
      "mv: GCCGTA_L1_1.fq.gz: No such file or directory\n",
      "mv: GCCGTA_L1_2.fq.gz: No such file or directory\n",
      "mv: ACTCTT_L1_1.fq.gz: No such file or directory\n",
      "mv: ACTCTT_L1_2.fq.gz: No such file or directory\n",
      "mv: TTCTAG_L1_1.fq.gz: No such file or directory\n",
      "mv: TTCTAG_L1_2.fq.gz: No such file or directory\n",
      "mv: ATTCCG_L1_1.fq.gz: No such file or directory\n",
      "mv: ATTCCG_L1_2.fq.gz: No such file or directory\n",
      "mv: CCGCAT_L1_1.fq.gz: No such file or directory\n",
      "mv: CCGCAT_L1_2.fq.gz: No such file or directory\n",
      "mv: CGAGGC_L1_1.fq.gz: No such file or directory\n",
      "mv: CGAGGC_L1_2.fq.gz: No such file or directory\n",
      "mv: CGCAGA_L1_1.fq.gz: No such file or directory\n",
      "mv: CGCAGA_L1_2.fq.gz: No such file or directory\n",
      "mv: GAGAGA_L1_1.fq.gz: No such file or directory\n",
      "mv: GAGAGA_L1_2.fq.gz: No such file or directory\n",
      "mv: GGGGCG_L1_1.fq.gz: No such file or directory\n",
      "mv: GGGGCG_L1_2.fq.gz: No such file or directory\n",
      "mv: TAATTC_L1_1.fq.gz: No such file or directory\n",
      "mv: TAATTC_L1_2.fq.gz: No such file or directory\n",
      "mv: TTTAAT_L1_1.fq.gz: No such file or directory\n",
      "mv: TTTAAT_L1_2.fq.gz: No such file or directory\n",
      "mv: AGCCAT_L1_1.fq.gz: No such file or directory\n",
      "mv: AGCCAT_L1_2.fq.gz: No such file or directory\n",
      "mv: AGTAGG_L1_1.fq.gz: No such file or directory\n",
      "mv: AGTAGG_L1_2.fq.gz: No such file or directory\n",
      "mv: ATAGTA_L1_1.fq.gz: No such file or directory\n",
      "mv: ATAGTA_L1_2.fq.gz: No such file or directory\n",
      "mv: CGGTCC_L1_1.fq.gz: No such file or directory\n",
      "mv: CGGTCC_L1_2.fq.gz: No such file or directory\n",
      "mv: CCATTT_L1_1.fq.gz: No such file or directory\n",
      "mv: CCATTT_L1_2.fq.gz: No such file or directory\n",
      "mv: CAGGCA_L1_1.fq.gz: No such file or directory\n",
      "mv: CAGGCA_L1_2.fq.gz: No such file or directory\n",
      "mv: GCAGAT_L1_1.fq.gz: No such file or directory\n",
      "mv: GCAGAT_L1_2.fq.gz: No such file or directory\n",
      "mv: GTCATC_L1_1.fq.gz: No such file or directory\n",
      "mv: GTCATC_L1_2.fq.gz: No such file or directory\n",
      "mv: GACTCT_L1_1.fq.gz: No such file or directory\n",
      "mv: GACTCT_L1_2.fq.gz: No such file or directory\n",
      "mv: TATCAC_L1_1.fq.gz: No such file or directory\n",
      "mv: TATCAC_L1_2.fq.gz: No such file or directory\n",
      "mv: TGGGGA_L1_1.fq.gz: No such file or directory\n",
      "mv: TGGGGA_L1_2.fq.gz: No such file or directory\n",
      "mv: TCTCGG_L1_1.fq.gz: No such file or directory\n",
      "mv: TCTCGG_L1_2.fq.gz: No such file or directory\n",
      "mv: TGACAA_L2_1.fq.gz: No such file or directory\n",
      "mv: TGACAA_L2_2.fq.gz: No such file or directory\n",
      "mv: AATATC_L2_1.fq.gz: No such file or directory\n",
      "mv: AATATC_L2_2.fq.gz: No such file or directory\n",
      "mv: ACAGCG_L2_1.fq.gz: No such file or directory\n",
      "mv: ACAGCG_L2_2.fq.gz: No such file or directory\n",
      "mv: GATCGT_L2_1.fq.gz: No such file or directory\n",
      "mv: GATCGT_L2_2.fq.gz: No such file or directory\n",
      "mv: GGTTTG_L2_1.fq.gz: No such file or directory\n",
      "mv: GGTTTG_L2_2.fq.gz: No such file or directory\n",
      "mv: CTCGCC_L2_1.fq.gz: No such file or directory\n",
      "mv: CTCGCC_L2_2.fq.gz: No such file or directory\n",
      "mv: GTGCCT_L2_1.fq.gz: No such file or directory\n",
      "mv: GTGCCT_L2_2.fq.gz: No such file or directory\n",
      "mv: TAGTAT_L2_1.fq.gz: No such file or directory\n",
      "mv: TAGTAT_L2_2.fq.gz: No such file or directory\n",
      "mv: ACAAGA_L2_1.fq.gz: No such file or directory\n",
      "mv: ACAAGA_L2_2.fq.gz: No such file or directory\n",
      "mv: TTCCGT_L2_1.fq.gz: No such file or directory\n",
      "mv: TTCCGT_L2_2.fq.gz: No such file or directory\n",
      "mv: CCCTAA_L2_1.fq.gz: No such file or directory\n",
      "mv: CCCTAA_L2_2.fq.gz: No such file or directory\n",
      "mv: AGGGTC_L2_1.fq.gz: No such file or directory\n",
      "mv: AGGGTC_L2_2.fq.gz: No such file or directory\n",
      "mv: CAATCG_L2_1.fq.gz: No such file or directory\n",
      "mv: CAATCG_L2_2.fq.gz: No such file or directory\n",
      "mv: GAACCA_L2_1.fq.gz: No such file or directory\n",
      "mv: GAACCA_L2_2.fq.gz: No such file or directory\n",
      "mv: TCGGAC_L2_1.fq.gz: No such file or directory\n",
      "mv: TCGGAC_L2_2.fq.gz: No such file or directory\n",
      "mv: ATGTTG_L2_1.fq.gz: No such file or directory\n",
      "mv: ATGTTG_L2_2.fq.gz: No such file or directory\n",
      "mv: AACTGA_L2_1.fq.gz: No such file or directory\n",
      "mv: AACTGA_L2_2.fq.gz: No such file or directory\n",
      "mv: TCTTCT_L2_1.fq.gz: No such file or directory\n",
      "mv: TCTTCT_L2_2.fq.gz: No such file or directory\n",
      "mv: GGACTT_L2_1.fq.gz: No such file or directory\n",
      "mv: GGACTT_L2_2.fq.gz: No such file or directory\n",
      "mv: GGGCGC_L2_1.fq.gz: No such file or directory\n",
      "mv: GGGCGC_L2_2.fq.gz: No such file or directory\n",
      "mv: ATCAAA_L2_1.fq.gz: No such file or directory\n",
      "mv: ATCAAA_L2_2.fq.gz: No such file or directory\n",
      "mv: GGCAAG_L2_1.fq.gz: No such file or directory\n",
      "mv: GGCAAG_L2_2.fq.gz: No such file or directory\n",
      "mv: CATACT_L2_1.fq.gz: No such file or directory\n",
      "mv: CATACT_L2_2.fq.gz: No such file or directory\n",
      "mv: CGCGTG_L2_1.fq.gz: No such file or directory\n",
      "mv: CGCGTG_L2_2.fq.gz: No such file or directory\n",
      "mv: AACGTT_L2_1.fq.gz: No such file or directory\n",
      "mv: AACGTT_L2_2.fq.gz: No such file or directory\n",
      "mv: AAGACG_L2_1.fq.gz: No such file or directory\n",
      "mv: AAGACG_L2_2.fq.gz: No such file or directory\n",
      "mv: AAGCTA_L2_1.fq.gz: No such file or directory\n",
      "mv: AAGCTA_L2_2.fq.gz: No such file or directory\n",
      "mv: AATGAG_L2_1.fq.gz: No such file or directory\n",
      "mv: AATGAG_L2_2.fq.gz: No such file or directory\n",
      "mv: ACATAC_L2_1.fq.gz: No such file or directory\n",
      "mv: ACATAC_L2_2.fq.gz: No such file or directory\n",
      "mv: ACCATG_L2_1.fq.gz: No such file or directory\n",
      "mv: ACCATG_L2_2.fq.gz: No such file or directory\n",
      "mv: ACCCCC_L2_1.fq.gz: No such file or directory\n",
      "mv: ACCCCC_L2_2.fq.gz: No such file or directory\n",
      "mv: ACTGGC_L2_1.fq.gz: No such file or directory\n",
      "mv: ACTGGC_L2_2.fq.gz: No such file or directory\n",
      "mv: AGCGCA_L2_1.fq.gz: No such file or directory\n",
      "mv: AGCGCA_L2_2.fq.gz: No such file or directory\n",
      "mv: AGGTGT_L2_1.fq.gz: No such file or directory\n",
      "mv: AGGTGT_L2_2.fq.gz: No such file or directory\n",
      "mv: AGTTAA_L2_1.fq.gz: No such file or directory\n",
      "mv: AGTTAA_L2_2.fq.gz: No such file or directory\n",
      "mv: ATGCAC_L2_1.fq.gz: No such file or directory\n",
      "mv: ATGCAC_L2_2.fq.gz: No such file or directory\n",
      "mv: GAAATG_L1_1.fq.gz: No such file or directory\n",
      "mv: GAAATG_L1_2.fq.gz: No such file or directory\n",
      "mv: GACGAC_L1_1.fq.gz: No such file or directory\n",
      "mv: GACGAC_L1_2.fq.gz: No such file or directory\n",
      "mv: GCATGG_L1_1.fq.gz: No such file or directory\n",
      "mv: GCATGG_L1_2.fq.gz: No such file or directory\n",
      "mv: GCGACC_L1_1.fq.gz: No such file or directory\n",
      "mv: GCGACC_L1_2.fq.gz: No such file or directory\n",
      "mv: GCGCTG_L1_1.fq.gz: No such file or directory\n",
      "mv: GCGCTG_L1_2.fq.gz: No such file or directory\n",
      "mv: GCTCAA_L1_1.fq.gz: No such file or directory\n",
      "mv: GCTCAA_L1_2.fq.gz: No such file or directory\n",
      "mv: GGTACA_L1_1.fq.gz: No such file or directory\n",
      "mv: GGTACA_L1_2.fq.gz: No such file or directory\n",
      "mv: GTAAGT_L1_1.fq.gz: No such file or directory\n",
      "mv: GTAAGT_L1_2.fq.gz: No such file or directory\n",
      "mv: GTATCC_L1_1.fq.gz: No such file or directory\n",
      "mv: GTATCC_L1_2.fq.gz: No such file or directory\n",
      "mv: GTGTAA_L1_1.fq.gz: No such file or directory\n",
      "mv: GTGTAA_L1_2.fq.gz: No such file or directory\n",
      "mv: GTTGGA_L1_1.fq.gz: No such file or directory\n",
      "mv: GTTGGA_L1_2.fq.gz: No such file or directory\n",
      "mv: TTTGTC_L1_1.fq.gz: No such file or directory\n",
      "mv: TTTGTC_L1_2.fq.gz: No such file or directory\n",
      "mv: TAAGCT_L2_1.fq.gz: No such file or directory\n",
      "mv: TAAGCT_L2_2.fq.gz: No such file or directory\n",
      "mv: TACACA_L2_1.fq.gz: No such file or directory\n",
      "mv: TACACA_L2_2.fq.gz: No such file or directory\n",
      "mv: TACGGG_L2_1.fq.gz: No such file or directory\n",
      "mv: TACGGG_L2_2.fq.gz: No such file or directory\n",
      "mv: TCAAAG_L2_1.fq.gz: No such file or directory\n",
      "mv: TCAAAG_L2_2.fq.gz: No such file or directory\n",
      "mv: TCCTGC_L2_1.fq.gz: No such file or directory\n",
      "mv: TCCTGC_L2_2.fq.gz: No such file or directory\n",
      "mv: TCGCCA_L2_1.fq.gz: No such file or directory\n",
      "mv: TCGCCA_L2_2.fq.gz: No such file or directory\n",
      "mv: TGAACC_L2_1.fq.gz: No such file or directory\n",
      "mv: TGAACC_L2_2.fq.gz: No such file or directory\n",
      "mv: TGCCCG_L2_1.fq.gz: No such file or directory\n",
      "mv: TGCCCG_L2_2.fq.gz: No such file or directory\n",
      "mv: TGCTTA_L2_1.fq.gz: No such file or directory\n",
      "mv: TGCTTA_L2_2.fq.gz: No such file or directory\n",
      "mv: TTATGA_L2_1.fq.gz: No such file or directory\n",
      "mv: TTATGA_L2_2.fq.gz: No such file or directory\n",
      "mv: TTGAGC_L2_1.fq.gz: No such file or directory\n",
      "mv: TTGAGC_L2_2.fq.gz: No such file or directory\n",
      "mv: GAAATG_L2_1.fq.gz: No such file or directory\n",
      "mv: GAAATG_L2_2.fq.gz: No such file or directory\n",
      "mv: GACGAC_L2_1.fq.gz: No such file or directory\n",
      "mv: GACGAC_L2_2.fq.gz: No such file or directory\n",
      "mv: GCATGG_L2_1.fq.gz: No such file or directory\n",
      "mv: GCATGG_L2_2.fq.gz: No such file or directory\n",
      "mv: GCGACC_L2_1.fq.gz: No such file or directory\n",
      "mv: GCGACC_L2_2.fq.gz: No such file or directory\n",
      "mv: GCGCTG_L2_1.fq.gz: No such file or directory\n",
      "mv: GCGCTG_L2_2.fq.gz: No such file or directory\n",
      "mv: GCTCAA_L2_1.fq.gz: No such file or directory\n",
      "mv: GCTCAA_L2_2.fq.gz: No such file or directory\n",
      "mv: GGTACA_L2_1.fq.gz: No such file or directory\n",
      "mv: GGTACA_L2_2.fq.gz: No such file or directory\n",
      "mv: GTAAGT_L2_1.fq.gz: No such file or directory\n",
      "mv: GTAAGT_L2_2.fq.gz: No such file or directory\n",
      "mv: GTATCC_L2_1.fq.gz: No such file or directory\n",
      "mv: GTATCC_L2_2.fq.gz: No such file or directory\n",
      "mv: GTGTAA_L2_1.fq.gz: No such file or directory\n",
      "mv: GTGTAA_L2_2.fq.gz: No such file or directory\n",
      "mv: GTTGGA_L2_1.fq.gz: No such file or directory\n",
      "mv: GTTGGA_L2_2.fq.gz: No such file or directory\n",
      "mv: TTTGTC_L2_1.fq.gz: No such file or directory\n",
      "mv: TTTGTC_L2_2.fq.gz: No such file or directory\n",
      "mv: TAAGCT_L1_1.fq.gz: No such file or directory\n",
      "mv: TAAGCT_L1_2.fq.gz: No such file or directory\n",
      "mv: TACACA_L1_1.fq.gz: No such file or directory\n",
      "mv: TACACA_L1_2.fq.gz: No such file or directory\n",
      "mv: TACGGG_L1_1.fq.gz: No such file or directory\n",
      "mv: TACGGG_L1_2.fq.gz: No such file or directory\n",
      "mv: TCAAAG_L1_1.fq.gz: No such file or directory\n",
      "mv: TCAAAG_L1_2.fq.gz: No such file or directory\n",
      "mv: TCCTGC_L1_1.fq.gz: No such file or directory\n",
      "mv: TCCTGC_L1_2.fq.gz: No such file or directory\n",
      "mv: TCGCCA_L1_1.fq.gz: No such file or directory\n",
      "mv: TCGCCA_L1_2.fq.gz: No such file or directory\n",
      "mv: TGAACC_L1_1.fq.gz: No such file or directory\n",
      "mv: TGAACC_L1_2.fq.gz: No such file or directory\n",
      "mv: TGCCCG_L1_1.fq.gz: No such file or directory\n",
      "mv: TGCCCG_L1_2.fq.gz: No such file or directory\n",
      "mv: TGCTTA_L1_1.fq.gz: No such file or directory\n",
      "mv: TGCTTA_L1_2.fq.gz: No such file or directory\n",
      "mv: TTATGA_L1_1.fq.gz: No such file or directory\n",
      "mv: TTATGA_L1_2.fq.gz: No such file or directory\n",
      "mv: TTGAGC_L1_1.fq.gz: No such file or directory\n",
      "mv: TTGAGC_L1_2.fq.gz: No such file or directory\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2005_387_1.fq.gz\n",
      "Loading RAD-Tags...done\n",
      "Loaded 337044 RAD-Tags.\n",
      "  Inserted 107931 elements into the RAD-Tags hash map.\n",
      "  196 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 9.87009; Std Dev: 16.9809; Max: 1341\n",
      "Deleveraging trigger: 27; Removal trigger: 44\n",
      "21754 initial stacks were populated; 86177 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 71 (74 k-mers per read); 3 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 457 stacks.\n",
      "  21390 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 9.13598; Std Dev: 33.5736; Max: 44\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  21390 stacks merged into 20263 stacks; deleveraged 123 stacks; removed 47 stacks.\n",
      "After merging, coverage depth Mean: 9.5447; Std Dev: 34.6834; Max: 86\n",
      "Merging remainder radtags\n",
      "  122330 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 23 (122 k-mers per read); 7 k-mer hits required.\n",
      "  Matched 41293 remainder reads; unable to match 81037 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 11.5967; Std Dev: 34.9479; Max: 252\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 256007\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2005_387_1.fq.gz... read 337044 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2005_387_2.fq.gz\n",
      "Loading RAD-Tags...done\n",
      "Loaded 337044 RAD-Tags.\n",
      "  Inserted 217488 elements into the RAD-Tags hash map.\n",
      "  3228 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 5.88971; Std Dev: 2.26641; Max: 65\n",
      "Deleveraging trigger: 8; Removal trigger: 10\n",
      "5567 initial stacks were populated; 211921 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 75 (76 k-mers per read); 1 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 164 stacks.\n",
      "  5530 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 5.64131; Std Dev: 3.53766; Max: 10\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 37 (114 k-mers per read); 3 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  5530 stacks merged into 5458 stacks; deleveraged 8 stacks; removed 0 stacks.\n",
      "After merging, coverage depth Mean: 5.7175; Std Dev: 3.66137; Max: 21\n",
      "Merging remainder radtags\n",
      "  304256 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 25 (126 k-mers per read); 1 k-mer hits required.\n",
      "  Matched 4983 remainder reads; unable to match 299273 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 6.65222; Std Dev: 4.94798; Max: 78\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 37771\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2005_387_2.fq.gz... read 337044 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2005_388_1.fq.gz\n",
      "Loading RAD-Tags...done\n",
      "Loaded 102934 RAD-Tags.\n",
      "  Inserted 47899 elements into the RAD-Tags hash map.\n",
      "  39 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 7.47604; Std Dev: 9.10383; Max: 308\n",
      "Deleveraging trigger: 17; Removal trigger: 26\n",
      "4758 initial stacks were populated; 43141 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 71 (74 k-mers per read); 3 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 140 stacks.\n",
      "  4652 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 6.82265; Std Dev: 16.715; Max: 26\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  4652 stacks merged into 4494 stacks; deleveraged 28 stacks; removed 8 stacks.\n",
      "After merging, coverage depth Mean: 6.99483; Std Dev: 17.1908; Max: 40\n",
      "Merging remainder radtags\n",
      "  67363 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 23 (122 k-mers per read); 7 k-mer hits required.\n",
      "  Matched 6914 remainder reads; unable to match 60449 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 8.54784; Std Dev: 17.7987; Max: 106\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 42485\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2005_388_1.fq.gz... read 102934 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2005_388_2.fq.gz\n",
      "Loading RAD-Tags...done\n",
      "Loaded 102934 RAD-Tags.\n",
      "  Inserted 68647 elements into the RAD-Tags hash map.\n",
      "  989 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 5.63989; Std Dev: 1.29719; Max: 28\n",
      "Deleveraging trigger: 7; Removal trigger: 8\n",
      "1494 initial stacks were populated; 67153 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 75 (76 k-mers per read); 1 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 39 stacks.\n",
      "  1491 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 5.50378; Std Dev: 2.72755; Max: 8\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 37 (114 k-mers per read); 3 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  1491 stacks merged into 1486 stacks; deleveraged 0 stacks; removed 0 stacks.\n",
      "After merging, coverage depth Mean: 5.52276; Std Dev: 2.75485; Max: 13\n",
      "Merging remainder radtags\n",
      "  94508 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 25 (126 k-mers per read); 1 k-mer hits required.\n",
      "  Matched 884 remainder reads; unable to match 93624 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 6.13241; Std Dev: 3.36175; Max: 33\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 9310\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2005_388_2.fq.gz... read 102934 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2005_389_1.fq.gz\n",
      "Loading RAD-Tags...done\n",
      "Loaded 383344 RAD-Tags.\n",
      "  Inserted 121511 elements into the RAD-Tags hash map.\n",
      "  186 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 10.3051; Std Dev: 18.8839; Max: 1699\n",
      "Deleveraging trigger: 29; Removal trigger: 48\n",
      "24055 initial stacks were populated; 97456 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 71 (74 k-mers per read); 3 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 558 stacks.\n",
      "  23600 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 9.51347; Std Dev: 37.3479; Max: 48\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  23600 stacks merged into 22247 stacks; deleveraged 135 stacks; removed 45 stacks.\n",
      "After merging, coverage depth Mean: 9.9828; Std Dev: 38.6664; Max: 86\n",
      "Merging remainder radtags\n",
      "  135455 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 23 (122 k-mers per read); 7 k-mer hits required.\n",
      "  Matched 50659 remainder reads; unable to match 84796 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 12.2752; Std Dev: 38.896; Max: 164\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 298548\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2005_389_1.fq.gz... read 383344 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2005_389_2.fq.gz\n",
      "Loading RAD-Tags...done\n",
      "Loaded 383344 RAD-Tags.\n",
      "  Inserted 246397 elements into the RAD-Tags hash map.\n",
      "  3653 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 5.95825; Std Dev: 2.46478; Max: 73\n",
      "Deleveraging trigger: 8; Removal trigger: 11\n",
      "6659 initial stacks were populated; 239738 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 75 (76 k-mers per read); 1 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 208 stacks.\n",
      "  6588 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 5.67958; Std Dev: 3.93443; Max: 11\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 37 (114 k-mers per read); 3 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  6588 stacks merged into 6483 stacks; deleveraged 11 stacks; removed 0 stacks.\n",
      "After merging, coverage depth Mean: 5.77356; Std Dev: 4.09073; Max: 27\n",
      "Merging remainder radtags\n",
      "  343668 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 25 (126 k-mers per read); 1 k-mer hits required.\n",
      "  Matched 6053 remainder reads; unable to match 337615 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 6.72739; Std Dev: 5.26791; Max: 55\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 45729\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2005_389_2.fq.gz... read 383344 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2005_457_1.fq.gz\n",
      "Loading RAD-Tags...1M...2M...done\n",
      "Loaded 2908839 RAD-Tags.\n",
      "  Inserted 512509 elements into the RAD-Tags hash map.\n",
      "  1310 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 50.5151; Std Dev: 109.249; Max: 15090\n",
      "Deleveraging trigger: 160; Removal trigger: 269\n",
      "47258 initial stacks were populated; 465251 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 71 (74 k-mers per read); 3 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 1433 stacks.\n",
      "  45990 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 47.1101; Std Dev: 218.955; Max: 269\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  45990 stacks merged into 39554 stacks; deleveraged 313 stacks; removed 120 stacks.\n",
      "After merging, coverage depth Mean: 54.0858; Std Dev: 236.829; Max: 467\n",
      "Merging remainder radtags\n",
      "  521596 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 23 (122 k-mers per read); 7 k-mer hits required.\n",
      "  Matched 377873 remainder reads; unable to match 143723 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 63.7085; Std Dev: 237.286; Max: 1034\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 2765116\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2005_457_1.fq.gz... read 2908839 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2005_457_2.fq.gz\n",
      "Loading RAD-Tags...1M...2M...done\n",
      "Loaded 2908839 RAD-Tags.\n",
      "  Inserted 1846417 elements into the RAD-Tags hash map.\n",
      "  26708 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 6.5805; Std Dev: 6.03966; Max: 279\n",
      "Deleveraging trigger: 13; Removal trigger: 19\n",
      "57738 initial stacks were populated; 1788679 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 75 (76 k-mers per read); 1 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 1756 stacks.\n",
      "  56605 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 6.04875; Std Dev: 9.35332; Max: 19\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 37 (114 k-mers per read); 3 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  56605 stacks merged into 55173 stacks; deleveraged 212 stacks; removed 36 stacks.\n",
      "After merging, coverage depth Mean: 6.18707; Std Dev: 9.62445; Max: 46\n",
      "Merging remainder radtags\n",
      "  2528894 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 25 (126 k-mers per read); 1 k-mer hits required.\n",
      "  Matched 77675 remainder reads; unable to match 2451219 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 7.61193; Std Dev: 10.9441; Max: 221\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 457620\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2005_457_2.fq.gz... read 2908839 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2009_172_1.fq.gz\n",
      "Loading RAD-Tags...done\n",
      "Loaded 37751 RAD-Tags.\n",
      "  Inserted 20915 elements into the RAD-Tags hash map.\n",
      "  15 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 6.83012; Std Dev: 5.30218; Max: 102\n",
      "Deleveraging trigger: 12; Removal trigger: 17\n",
      "1142 initial stacks were populated; 19773 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 71 (74 k-mers per read); 3 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 50 stacks.\n",
      "  1112 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 6.1511; Std Dev: 10.4541; Max: 16\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  1112 stacks merged into 1084 stacks; deleveraged 2 stacks; removed 0 stacks.\n",
      "After merging, coverage depth Mean: 6.31297; Std Dev: 10.6624; Max: 25\n",
      "Merging remainder radtags\n",
      "  29951 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 23 (122 k-mers per read); 7 k-mer hits required.\n",
      "  Matched 1587 remainder reads; unable to match 28364 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 7.80451; Std Dev: 11.6801; Max: 68\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 9387\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2009_172_1.fq.gz... read 37751 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2009_172_2.fq.gz\n",
      "Loading RAD-Tags...done\n",
      "Loaded 37751 RAD-Tags.\n",
      "  Inserted 25843 elements into the RAD-Tags hash map.\n",
      "  349 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 5.45666; Std Dev: 0.814802; Max: 10\n",
      "Deleveraging trigger: 6; Removal trigger: 7\n",
      "473 initial stacks were populated; 25370 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 75 (76 k-mers per read); 1 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 14 stacks.\n",
      "  473 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 5.37037; Std Dev: 2.46454; Max: 7\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 37 (114 k-mers per read); 3 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  473 stacks merged into 473 stacks; deleveraged 0 stacks; removed 0 stacks.\n",
      "After merging, coverage depth Mean: 5.37037; Std Dev: 2.46454; Max: 7\n",
      "Merging remainder radtags\n",
      "  35170 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 25 (126 k-mers per read); 1 k-mer hits required.\n",
      "  Matched 179 remainder reads; unable to match 34991 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 5.76035; Std Dev: 2.93145; Max: 26\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 2760\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2009_172_2.fq.gz... read 37751 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2010_114_1.fq.gz\n",
      "Loading RAD-Tags...1M...2M...3M...4M...done\n",
      "Loaded 4427652 RAD-Tags.\n",
      "  Inserted 723567 elements into the RAD-Tags hash map.\n",
      "  2040 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 75.6356; Std Dev: 181.781; Max: 25990\n",
      "Deleveraging trigger: 257; Removal trigger: 439\n",
      "48752 initial stacks were populated; 674815 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 71 (74 k-mers per read); 3 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 1748 stacks.\n",
      "  47157 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 70.6055; Std Dev: 382.219; Max: 437\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  47157 stacks merged into 40480 stacks; deleveraged 349 stacks; removed 157 stacks.\n",
      "After merging, coverage depth Mean: 81.1733; Std Dev: 413.959; Max: 807\n",
      "Merging remainder radtags\n",
      "  740264 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 23 (122 k-mers per read); 7 k-mer hits required.\n",
      "  Matched 529863 remainder reads; unable to match 210401 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 94.3638; Std Dev: 414.474; Max: 1473\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 4217251\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2010_114_1.fq.gz... read 4427652 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2010_114_2.fq.gz\n",
      "Loading RAD-Tags...1M...2M...3M...4M...done\n",
      "Loaded 4427652 RAD-Tags.\n",
      "  Inserted 2788850 elements into the RAD-Tags hash map.\n",
      "  41247 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 6.89688; Std Dev: 8.195; Max: 523\n",
      "Deleveraging trigger: 15; Removal trigger: 23\n",
      "95561 initial stacks were populated; 2693289 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 75 (76 k-mers per read); 1 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 3076 stacks.\n",
      "  93448 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 6.21576; Std Dev: 12.8153; Max: 23\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 37 (114 k-mers per read); 3 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  93448 stacks merged into 90643 stacks; deleveraged 450 stacks; removed 99 stacks.\n",
      "After merging, coverage depth Mean: 6.37406; Std Dev: 13.1731; Max: 48\n",
      "Merging remainder radtags\n",
      "  3768579 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 25 (126 k-mers per read); 1 k-mer hits required.\n",
      "  Matched 143237 remainder reads; unable to match 3625342 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 7.97303; Std Dev: 14.5501; Max: 371\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 802310\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2010_114_2.fq.gz... read 4427652 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2010_115_1.fq.gz\n",
      "Loading RAD-Tags...1M...2M...3M...done\n",
      "Loaded 3143244 RAD-Tags.\n",
      "  Inserted 519706 elements into the RAD-Tags hash map.\n",
      "  1414 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 56.1717; Std Dev: 118.954; Max: 13584\n",
      "Deleveraging trigger: 175; Removal trigger: 294\n",
      "46678 initial stacks were populated; 473028 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 71 (74 k-mers per read); 3 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 1387 stacks.\n",
      "  45461 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 52.0076; Std Dev: 252.903; Max: 294\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  45461 stacks merged into 39684 stacks; deleveraged 304 stacks; removed 113 stacks.\n",
      "After merging, coverage depth Mean: 58.8407; Std Dev: 271.558; Max: 482\n",
      "Merging remainder radtags\n",
      "  521261 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 23 (122 k-mers per read); 7 k-mer hits required.\n",
      "  Matched 366671 remainder reads; unable to match 154590 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 68.1469; Std Dev: 272.165; Max: 2225\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 2988654\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2010_115_1.fq.gz... read 3143244 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2010_115_2.fq.gz\n",
      "Loading RAD-Tags...1M...2M...3M...done\n",
      "Loaded 3143244 RAD-Tags.\n",
      "  Inserted 2040198 elements into the RAD-Tags hash map.\n",
      "  28805 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 6.70175; Std Dev: 6.87525; Max: 321\n",
      "Deleveraging trigger: 14; Removal trigger: 20\n",
      "57831 initial stacks were populated; 1982367 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 75 (76 k-mers per read); 1 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 1953 stacks.\n",
      "  56555 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 6.07221; Std Dev: 10.5137; Max: 20\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 37 (114 k-mers per read); 3 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  56555 stacks merged into 54958 stacks; deleveraged 255 stacks; removed 54 stacks.\n",
      "After merging, coverage depth Mean: 6.21912; Std Dev: 10.8281; Max: 41\n",
      "Merging remainder radtags\n",
      "  2755675 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 25 (126 k-mers per read); 1 k-mer hits required.\n",
      "  Matched 87172 remainder reads; unable to match 2668503 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 7.82665; Std Dev: 12.2677; Max: 259\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 474741\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2010_115_2.fq.gz... read 3143244 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2010_116_1.fq.gz\n",
      "Loading RAD-Tags...1M...2M...done\n",
      "Loaded 2778596 RAD-Tags.\n",
      "  Inserted 477610 elements into the RAD-Tags hash map.\n",
      "  1321 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 50.0551; Std Dev: 133.027; Max: 17419\n",
      "Deleveraging trigger: 183; Removal trigger: 316\n",
      "46043 initial stacks were populated; 431567 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 71 (74 k-mers per read); 3 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 1367 stacks.\n",
      "  44799 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 46.4636; Std Dev: 273.869; Max: 312\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  44799 stacks merged into 39063 stacks; deleveraged 301 stacks; removed 130 stacks.\n",
      "After merging, coverage depth Mean: 52.3894; Std Dev: 294.352; Max: 513\n",
      "Merging remainder radtags\n",
      "  473911 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 23 (122 k-mers per read); 7 k-mer hits required.\n",
      "  Matched 333431 remainder reads; unable to match 140480 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 60.9808; Std Dev: 294.644; Max: 583\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 2638116\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2010_116_1.fq.gz... read 2778596 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2010_116_2.fq.gz\n",
      "Loading RAD-Tags...1M...2M...done\n",
      "Loaded 2778596 RAD-Tags.\n",
      "  Inserted 1801256 elements into the RAD-Tags hash map.\n",
      "  25736 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 6.88259; Std Dev: 8.3336; Max: 356\n",
      "Deleveraging trigger: 15; Removal trigger: 24\n",
      "49929 initial stacks were populated; 1751327 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 75 (76 k-mers per read); 1 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 1858 stacks.\n",
      "  48620 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 6.09473; Std Dev: 12.8187; Max: 24\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 37 (114 k-mers per read); 3 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  48620 stacks merged into 47147 stacks; deleveraged 220 stacks; removed 56 stacks.\n",
      "After merging, coverage depth Mean: 6.24432; Std Dev: 13.1908; Max: 54\n",
      "Merging remainder radtags\n",
      "  2434955 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 25 (126 k-mers per read); 1 k-mer hits required.\n",
      "  Matched 77066 remainder reads; unable to match 2357889 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 7.90015; Std Dev: 14.673; Max: 249\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 420707\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2010_116_2.fq.gz... read 2778596 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2010_117_1.fq.gz\n",
      "Loading RAD-Tags...1M...2M...3M...4M...done\n",
      "Loaded 4052820 RAD-Tags.\n",
      "  Inserted 662406 elements into the RAD-Tags hash map.\n",
      "  1808 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 68.5725; Std Dev: 164.084; Max: 22108\n",
      "Deleveraging trigger: 233; Removal trigger: 397\n",
      "49207 initial stacks were populated; 613199 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 71 (74 k-mers per read); 3 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 1802 stacks.\n",
      "  47560 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 64.0854; Std Dev: 333.468; Max: 396\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  47560 stacks merged into 40584 stacks; deleveraged 373 stacks; removed 144 stacks.\n",
      "After merging, coverage depth Mean: 74.0004; Std Dev: 362.204; Max: 628\n",
      "Merging remainder radtags\n",
      "  678572 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 23 (122 k-mers per read); 7 k-mer hits required.\n",
      "  Matched 478542 remainder reads; unable to match 200030 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 85.8793; Std Dev: 362.628; Max: 817\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 3852790\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2010_117_1.fq.gz... read 4052820 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2010_117_2.fq.gz\n",
      "Loading RAD-Tags...1M...2M...3M...4M...done\n",
      "Loaded 4052820 RAD-Tags.\n",
      "  Inserted 2540585 elements into the RAD-Tags hash map.\n",
      "  37645 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 6.80969; Std Dev: 7.54951; Max: 492\n",
      "Deleveraging trigger: 14; Removal trigger: 22\n",
      "87804 initial stacks were populated; 2452781 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 75 (76 k-mers per read); 1 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 2752 stacks.\n",
      "  85945 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 6.18089; Std Dev: 11.5834; Max: 22\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 37 (114 k-mers per read); 3 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  85945 stacks merged into 83481 stacks; deleveraged 391 stacks; removed 83 stacks.\n",
      "After merging, coverage depth Mean: 6.33396; Std Dev: 11.9129; Max: 46\n",
      "Merging remainder radtags\n",
      "  3454902 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 25 (126 k-mers per read); 1 k-mer hits required.\n",
      "  Matched 125862 remainder reads; unable to match 3329040 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 7.85946; Std Dev: 13.2188; Max: 254\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 723780\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2010_117_2.fq.gz... read 4052820 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2010_118_1.fq.gz\n",
      "Loading RAD-Tags...1M...2M...3M...4M...done\n",
      "Loaded 4671321 RAD-Tags.\n",
      "  Inserted 750432 elements into the RAD-Tags hash map.\n",
      "  2150 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 78.3372; Std Dev: 172.151; Max: 26924\n",
      "Deleveraging trigger: 250; Removal trigger: 423\n",
      "49714 initial stacks were populated; 700718 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 71 (74 k-mers per read); 3 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 1871 stacks.\n",
      "  48003 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 73.4917; Std Dev: 374.846; Max: 422\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  48003 stacks merged into 40908 stacks; deleveraged 381 stacks; removed 165 stacks.\n",
      "After merging, coverage depth Mean: 85.1; Std Dev: 407.442; Max: 776\n",
      "Merging remainder radtags\n",
      "  776865 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 23 (122 k-mers per read); 7 k-mer hits required.\n",
      "  Matched 565892 remainder reads; unable to match 210973 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 99.044; Std Dev: 408.065; Max: 2409\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 4460348\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2010_118_1.fq.gz... read 4671321 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2010_118_2.fq.gz\n",
      "Loading RAD-Tags...1M...2M...3M...4M...done\n",
      "Loaded 4671321 RAD-Tags.\n",
      "  Inserted 2917632 elements into the RAD-Tags hash map.\n",
      "  42637 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 6.80444; Std Dev: 7.58676; Max: 546\n",
      "Deleveraging trigger: 14; Removal trigger: 22\n",
      "100899 initial stacks were populated; 2816733 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 75 (76 k-mers per read); 1 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 3071 stacks.\n",
      "  98799 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 6.19977; Std Dev: 11.9405; Max: 22\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 37 (114 k-mers per read); 3 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  98799 stacks merged into 95720 stacks; deleveraged 489 stacks; removed 106 stacks.\n",
      "After merging, coverage depth Mean: 6.36344; Std Dev: 12.311; Max: 51\n",
      "Merging remainder radtags\n",
      "  3984760 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 25 (126 k-mers per read); 1 k-mer hits required.\n",
      "  Matched 146611 remainder reads; unable to match 3838149 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 7.91253; Std Dev: 13.6328; Max: 424\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 833172\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2010_118_2.fq.gz... read 4671321 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2010_119_1.fq.gz\n",
      "Loading RAD-Tags...1M...2M...3M...done\n",
      "Loaded 3669174 RAD-Tags.\n",
      "  Inserted 614790 elements into the RAD-Tags hash map.\n",
      "  1720 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 62.6005; Std Dev: 136.23; Max: 16601\n",
      "Deleveraging trigger: 199; Removal trigger: 335\n",
      "48554 initial stacks were populated; 566236 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 71 (74 k-mers per read); 3 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 1623 stacks.\n",
      "  47108 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 58.4327; Std Dev: 274.496; Max: 327\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  47108 stacks merged into 40388 stacks; deleveraged 328 stacks; removed 132 stacks.\n",
      "After merging, coverage depth Mean: 67.3503; Std Dev: 297.349; Max: 571\n",
      "Merging remainder radtags\n",
      "  629670 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 23 (122 k-mers per read); 7 k-mer hits required.\n",
      "  Matched 448394 remainder reads; unable to match 181276 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 78.5381; Std Dev: 297.788; Max: 788\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "^CTraceback (most recent call last):\n",
      "  File \"pypipe_ustacks.py\", line 109, in <module>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!python pypipe_ustacks.py barcodes_samplenames.txt ./process_radtags_out ./ustacks_out"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The kernel crashed some point in the middle of the night so I don't have all my data :( going to try to rerun some of it now.\n",
    "\n",
    "I'm only going to use the forward reads for this class project, so I manually removed the reverse read files that ustacks made and put them in a different folder.\n",
    "\n",
    "## ``cstacks``\n",
    "\n",
    "Then, I run cstacks on 10 samples that represent the samples with the most reads.\n",
    "\n",
    "<br>\n",
    "My ``cstacks`` script looks like this:\n",
    "\n",
    "```\n",
    "##########################################################################################\n",
    "# \n",
    "###--- ``cstacks`` script\n",
    "#\n",
    "# PURPOSE: ustacks creates a catalog from a subset of individuals to call SNPs\n",
    "# INPUT: ustacks out put files for specified number of individuals with most sequence reads\n",
    "# OUTPUT: catalog file + associated files\n",
    "# \n",
    "#### WHEN RUNNING THIS SCRIPT, INPUTS AT THE COMMAND LINE ARE:\n",
    "# python \n",
    "# {0}[pypipe_cstacks.py] \n",
    "# {1}[shell script with changed file names] \t\n",
    "# {2}[directory of input files] \n",
    "# {3}[# individuals for cstacks] \n",
    "# {4}[batch number] \n",
    "# {5}[output directory] \n",
    "# {6}[num mismatches allowed] \n",
    "# {7}[num threads]\n",
    "# \n",
    "### DEPENDENCIES: \n",
    "# [1] Your file names coming out of ustacks cannot have a period other than before file extension (not sure if true! check?)\n",
    "# \n",
    "### WARNINGS:\n",
    "# [1] If you have to rerun this script, it will append onto it! make sure no file w name or else won't even run!\n",
    "# \n",
    "##########################################################################################\n",
    "\n",
    "### --- [A] Call necessary modules\n",
    "\n",
    "import sys \n",
    "import subprocess\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "### --- [B] Count lines in each sequence file\n",
    "\n",
    "myfile = open(sys.argv[1], \"r\")\t#open the file with your list of barcodes and sample IDs\n",
    "lines = myfile.readlines()[2:]\n",
    "\n",
    "\n",
    "dirfrom = sys.argv[2] # get directory for input files\n",
    "firststr = \"cd \" + sys.argv[2] + \"\\n\" # write first line of shell to cd to this directory\n",
    "\n",
    "filestring = \"\"\n",
    "filestring += firststr\n",
    "\n",
    "samplename_list = [] # to be used in loop later in this script\n",
    "\n",
    "for line in lines: \t\t\t\t\t#for each line in the barcode file\n",
    "\tlinelist = line.strip().split()\t\t#make a list of character strings broken by tabs\n",
    "\tsampID = linelist[2]\t\t\t\t#pick out file name\n",
    "\tsamplename_list.append(sampID)\n",
    "\tnewstring = \"gunzip -c \" + sampID + \" | wc -l >> ../cstacks_linecount.txt\\n\" # make line of code to run at command line\n",
    "\tfilestring += newstring # add to a list of strings we'll write to a file\n",
    "myfile.close()\n",
    "\n",
    "\n",
    "#create a new file where the ustacks code will go, write string to file, close\n",
    "newfile = open(\"cstacks_linecount_shell.txt\", \"w\")\n",
    "newfile.write(filestring)\n",
    "newfile.close()\n",
    "\n",
    "# run shell script that will calculate line counts\n",
    "# subprocess.call(['sh cstacks_linecount_shell.txt'], shell=True)\n",
    "# \n",
    "\n",
    "# \n",
    "# \n",
    "# \n",
    "# ### --- [C] Get sample names for specified # of samples with most sequence reads \n",
    "# \n",
    "linecounts = open(\"cstacks_linecount.txt\", \"r\") # read in line counts file\n",
    "linecounts_list = [] # initiate a list for the line counts so I can get.item later\n",
    "\n",
    "for line in linecounts:\n",
    "\tcount = line.strip().split() # get line count\n",
    "\tlinecounts_list.append(line)\n",
    "\n",
    "list_samp_ct = [] # initiate empty list\n",
    "i = 0 # start counter\n",
    "\n",
    "# --- CHECK^ if lists look normal\n",
    "# print \"sample name list \"\n",
    "# print samplename_list\n",
    "# print \"line counts list \"\n",
    "# print linecounts_list\n",
    "\n",
    "for item in linecounts_list:\n",
    "\tnew_item = [samplename_list[i], linecounts_list[i]]\n",
    "\tlist_samp_ct.append(new_item)\n",
    "\ti += 1\n",
    "\t\n",
    "def getKey(item): # so that sorted will sort by second item in list\n",
    "\treturn item[1]\n",
    "sortedlist = sorted(list_samp_ct, key = getKey, reverse = True)\n",
    "# print sortedlist # CHECK^\n",
    "\n",
    "with open('all_sorted_name_counts.txt', 'w') as file:\n",
    "    file.writelines('\\t'.join(i) + '\\n' for i in sortedlist) # makes file\n",
    "    \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "### --- [D] Write and run shell script for ``cstacks``\n",
    "\n",
    "\n",
    "# ---------\n",
    "# I had an automated way to pick the ten samples with most reads, but my ustacks run on 20161108 failed halfway throgh\n",
    "# and so I have less samples. So I manually picked ten with the most reads of the files that went through ustacks\n",
    "\n",
    "# so hash comment this out in the future!\n",
    "\n",
    "\n",
    "samples_for_use = []\n",
    "foruse = open(\"files_for_cstacks.txt\", \"r\")\n",
    "for line in foruse:\n",
    "\tlinelist = line.strip().split()\n",
    "\tsamples_for_use.append(linelist[0])\n",
    "foruse.close()\n",
    "\n",
    "\n",
    "\n",
    "#----------\n",
    "\n",
    "\n",
    "\n",
    "cstacks_shell = \"\"\n",
    "dirstr = \"cd \" + sys.argv[2] + \"\\n\"\n",
    "firststr = \"cstacks -b \" + sys.argv[4] + \" \"\n",
    "cstacks_shell += firststr\n",
    "\n",
    "endrange = int(sys.argv[3]) # set end of range for loop\n",
    "for i in range(0, endrange):\n",
    "\tfilename = samples_for_use[i]\n",
    "\ttrmd_filename = filename.rsplit(\".\",2)[0]\n",
    "\t# print trmd_filename # CHECK^\n",
    "\tstring = \"-s \" + trmd_filename + \" \"\n",
    "\tcstacks_shell += string\n",
    "laststr = \"-o \" + sys.argv[5] + \" -n \" + sys.argv[6] + \" -p \" + sys.argv[7]\n",
    "cstacks_shell += laststr\n",
    "# print cstacks_shell # CHECK^\n",
    "\n",
    "cstacks_shell_txt = open(\"cstacks_shell.txt\", \"w\")\n",
    "cstacks_shell_txt.write(cstacks_shell)\n",
    "cstacks_shell_txt.close()\n",
    "\n",
    "# run shell script\n",
    "subprocess.call([\"sh cstacks_shell.txt\"], shell = True)\n",
    "\n",
    "##########################################################################################\n",
    "\n",
    "\n",
    "##########################################################################################\n",
    "\n",
    "\n",
    "### --- DOCUMENTATION FOR CSTACKS\n",
    "\n",
    "# cstacks -b batch_id -s sample_file [-s sample_file_2 ...] [-o path] [-n num] [-g] [-p num_threads] [--catalog path] [-h]\n",
    "# p  enable parallel execution with num_threads threads.\n",
    "# b  MySQL ID of this batch.\n",
    "# s  TSV file from which to load radtags.\n",
    "# o  output path to write results.\n",
    "# m  include tags in the catalog that match to more than one entry.\n",
    "# n  number of mismatches allowed between sample tags when generating the catalog.\n",
    "# g  base catalog matching on genomic location, not sequence identity.\n",
    "# h  display this help messsage.\n",
    "# Catalog editing:\n",
    "# \n",
    "# --catalog [path]  provide the path to an existing catalog. cstacks will add data to this existing catalog.\n",
    "# Advanced options:\n",
    "# \n",
    "# --report_mmatches  report query loci that match more than one catalog locus.\n",
    "\n",
    "```\n",
    "\n",
    "This is how I would run it (below), except that my ustacks run failed and only 25% of my files went through, meaning a couple steps in my ustacks script had to be changed. So I did a couple parts manually, and at the terminal.\n",
    "\n",
    "```\n",
    "!python pypipe_cstacks.py new_filenames_shell.txt ustacks_out 10 1 cstacks_out 3 5\n",
    "```\n",
    "\n",
    "And then the output catalog files look like this:\n",
    "\n",
    "<br>\n",
    "![image](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Notebooks/images_for_notebooks/Screen%20Shot%202016-11-18%20at%202.25.06%20PM.png?raw=true)\n",
    "<br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## ``sstacks``\n",
    "\n",
    "<br>\n",
    "\n",
    "Then comes sstacks, where the program matches each individuals reads to the catalog.\n",
    "\n",
    "My code looks like this:\n",
    "\n",
    "```\n",
    "##########################################################################################\n",
    "# \n",
    "### ``sstacks``\n",
    "#\n",
    "# PURPOSE: To match individual samples against your catalog for genotyping\n",
    "# INPUT: TSV output files from cstacks\n",
    "# OUTPUT: match files\n",
    "# \n",
    "# \n",
    "### WHEN RUNNING THIS SCRIPT, YOUR INPUTS AT THE COMMAND LINE ARE:\n",
    "# python  \n",
    "# {0}[pipeline filename] \n",
    "# {1}[shell script w sample names]\n",
    "# {2}[batch ID number]\n",
    "# {3}[filepath to directory with catalog filename without file extension]\n",
    "# {4}[filepath to directory w ustacks output files per sample]\n",
    "# {5}[number of threads to use]\n",
    "# \n",
    "### DEPENDENCIES\n",
    "# \n",
    "# [1]\n",
    "# \n",
    "### WARNINGS\n",
    "# \n",
    "# [1] \n",
    "# \n",
    "##########################################################################################\n",
    "\n",
    "# --- [A] call necessary modules\n",
    "\n",
    "import subprocess\n",
    "import sys \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# --- [B] make shell script for sstacks\n",
    "\n",
    "trim_names = [] # initiate list\n",
    "\n",
    "rename_shell = open(sys.argv[1], \"r\") # open file w filenames \n",
    "lines = rename_shell.readlines()[2:]\n",
    "\n",
    "for line in lines:\n",
    "\tlinelist = line.strip().split()\n",
    "\ttrim_name = linelist[2].rsplit(\".\",2)[0]\n",
    "\ttrim_names.append(trim_name)\n",
    "# print trim_names # CHECK^\n",
    "\n",
    "rename_shell.close()\n",
    "\n",
    "numsamples = len(trim_names)\n",
    "\n",
    "newfile = open(\"sstacks_shell.txt\", \"w\") # create new file for shell script\n",
    "\n",
    "filestring = \"\"\n",
    "\n",
    "for i in range(0,numsamples):\n",
    "\tfilestring += \"sstacks -b \" + sys.argv[2] + \" -c \" + sys.argv[3]\n",
    "\tsubstr = \" -s \" + sys.argv[4] + \"/\" + trim_names[i] + \" -p \" + sys.argv[5] + \"\\n\"\n",
    "\tfilestring += substr\t\n",
    "\n",
    "# print filestring # ^CHECK\n",
    "\n",
    "newfile.write(filestring)\n",
    "newfile.close()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# --- [C] run shell script for sstacks\n",
    "\n",
    "subprocess.call([\"sh sstacks_shell.txt\"], shell=True)\n",
    "\n",
    "```\n",
    "\n",
    "Again, this run on \"all\" my data sort of failed when ustacks stopped overnight, so I'll need a new way of making a list of samples that I actually want to run sstacks on. Or, perhaps I can run it as is and it will skip those files and report that it couldn't find them. Let's try that..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_387_1' [1 of 1]\n",
      "  Parsing ustacks_out/2005_387_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2005_387_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2005_387_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "20123 stacks compared against the catalog containing 61592 loci.\n",
      "  19776 matching loci, 1158 contained no verified haplotypes.\n",
      "  151 loci matched more than one catalog locus and were excluded.\n",
      "  1007 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  21083 total haplotypes examined from matching loci, 19679 verified.\n",
      "Outputing to file ./2005_387_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_387_2' [1 of 1]\n",
      "  Parsing ustacks_out/2005_387_2.tags.tsv.gz\n",
      "  Parsing ustacks_out/2005_387_2.snps.tsv.gz\n",
      "  Parsing ustacks_out/2005_387_2.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "5331 stacks compared against the catalog containing 61592 loci.\n",
      "  0 matching loci, 0 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  0 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  0 total haplotypes examined from matching loci, 0 verified.\n",
      "Outputing to file ./2005_387_2.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_388_1' [1 of 1]\n",
      "  Parsing ustacks_out/2005_388_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2005_388_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2005_388_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "4452 stacks compared against the catalog containing 61592 loci.\n",
      "  4279 matching loci, 244 contained no verified haplotypes.\n",
      "  63 loci matched more than one catalog locus and were excluded.\n",
      "  181 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  4539 total haplotypes examined from matching loci, 4191 verified.\n",
      "Outputing to file ./2005_388_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_388_2' [1 of 1]\n",
      "  Parsing ustacks_out/2005_388_2.tags.tsv.gz\n",
      "  Parsing ustacks_out/2005_388_2.snps.tsv.gz\n",
      "  Parsing ustacks_out/2005_388_2.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "1450 stacks compared against the catalog containing 61592 loci.\n",
      "  0 matching loci, 0 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  0 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  0 total haplotypes examined from matching loci, 0 verified.\n",
      "Outputing to file ./2005_388_2.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_389_1' [1 of 1]\n",
      "  Parsing ustacks_out/2005_389_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2005_389_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2005_389_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "22099 stacks compared against the catalog containing 61592 loci.\n",
      "  21727 matching loci, 1361 contained no verified haplotypes.\n",
      "  180 loci matched more than one catalog locus and were excluded.\n",
      "  1181 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  23369 total haplotypes examined from matching loci, 21695 verified.\n",
      "Outputing to file ./2005_389_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_389_2' [1 of 1]\n",
      "  Parsing ustacks_out/2005_389_2.tags.tsv.gz\n",
      "  Parsing ustacks_out/2005_389_2.snps.tsv.gz\n",
      "  Parsing ustacks_out/2005_389_2.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "6346 stacks compared against the catalog containing 61592 loci.\n",
      "  0 matching loci, 0 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  0 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  0 total haplotypes examined from matching loci, 0 verified.\n",
      "Outputing to file ./2005_389_2.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_457_1' [1 of 1]\n",
      "  Parsing ustacks_out/2005_457_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2005_457_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2005_457_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39269 stacks compared against the catalog containing 61592 loci.\n",
      "  38120 matching loci, 1112 contained no verified haplotypes.\n",
      "  155 loci matched more than one catalog locus and were excluded.\n",
      "  957 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42684 total haplotypes examined from matching loci, 41268 verified.\n",
      "Outputing to file ./2005_457_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_457_2' [1 of 1]\n",
      "  Parsing ustacks_out/2005_457_2.tags.tsv.gz\n",
      "  Parsing ustacks_out/2005_457_2.snps.tsv.gz\n",
      "  Parsing ustacks_out/2005_457_2.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "54514 stacks compared against the catalog containing 61592 loci.\n",
      "  3 matching loci, 0 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  0 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  3 total haplotypes examined from matching loci, 3 verified.\n",
      "Outputing to file ./2005_457_2.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_172_1' [1 of 1]\n",
      "  Parsing ustacks_out/2009_172_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2009_172_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2009_172_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "1064 stacks compared against the catalog containing 61592 loci.\n",
      "  1010 matching loci, 58 contained no verified haplotypes.\n",
      "  30 loci matched more than one catalog locus and were excluded.\n",
      "  28 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  1085 total haplotypes examined from matching loci, 982 verified.\n",
      "Outputing to file ./2009_172_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_172_2' [1 of 1]\n",
      "  Parsing ustacks_out/2009_172_2.tags.tsv.gz\n",
      "  Parsing ustacks_out/2009_172_2.snps.tsv.gz\n",
      "  Parsing ustacks_out/2009_172_2.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "459 stacks compared against the catalog containing 61592 loci.\n",
      "  0 matching loci, 0 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  0 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  0 total haplotypes examined from matching loci, 0 verified.\n",
      "Outputing to file ./2009_172_2.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_114_1' [1 of 1]\n",
      "  Parsing ustacks_out/2010_114_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_114_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_114_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40170 stacks compared against the catalog containing 61592 loci.\n",
      "  40127 matching loci, 139 contained no verified haplotypes.\n",
      "  137 loci matched more than one catalog locus and were excluded.\n",
      "  2 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  46576 total haplotypes examined from matching loci, 46251 verified.\n",
      "Outputing to file ./2010_114_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_114_2' [1 of 1]\n",
      "  Parsing ustacks_out/2010_114_2.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_114_2.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_114_2.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "89581 stacks compared against the catalog containing 61592 loci.\n",
      "  4 matching loci, 1 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  1 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  4 total haplotypes examined from matching loci, 3 verified.\n",
      "Outputing to file ./2010_114_2.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_115_1' [1 of 1]\n",
      "  Parsing ustacks_out/2010_115_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_115_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_115_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39401 stacks compared against the catalog containing 61592 loci.\n",
      "  38070 matching loci, 1030 contained no verified haplotypes.\n",
      "  149 loci matched more than one catalog locus and were excluded.\n",
      "  881 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42533 total haplotypes examined from matching loci, 41196 verified.\n",
      "Outputing to file ./2010_115_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_115_2' [1 of 1]\n",
      "  Parsing ustacks_out/2010_115_2.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_115_2.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_115_2.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "54227 stacks compared against the catalog containing 61592 loci.\n",
      "  3 matching loci, 3 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  3 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  3 total haplotypes examined from matching loci, 0 verified.\n",
      "Outputing to file ./2010_115_2.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_116_1' [1 of 1]\n",
      "  Parsing ustacks_out/2010_116_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_116_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_116_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38810 stacks compared against the catalog containing 61592 loci.\n",
      "  37805 matching loci, 1020 contained no verified haplotypes.\n",
      "  145 loci matched more than one catalog locus and were excluded.\n",
      "  875 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42181 total haplotypes examined from matching loci, 40866 verified.\n",
      "Outputing to file ./2010_116_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_116_2' [1 of 1]\n",
      "  Parsing ustacks_out/2010_116_2.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_116_2.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_116_2.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "46542 stacks compared against the catalog containing 61592 loci.\n",
      "  1 matching loci, 1 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  0 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  2 total haplotypes examined from matching loci, 0 verified.\n",
      "Outputing to file ./2010_116_2.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_117_1' [1 of 1]\n",
      "  Parsing ustacks_out/2010_117_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_117_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_117_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40285 stacks compared against the catalog containing 61592 loci.\n",
      "  38839 matching loci, 1022 contained no verified haplotypes.\n",
      "  146 loci matched more than one catalog locus and were excluded.\n",
      "  876 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  43617 total haplotypes examined from matching loci, 42233 verified.\n",
      "Outputing to file ./2010_117_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_117_2' [1 of 1]\n",
      "  Parsing ustacks_out/2010_117_2.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_117_2.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_117_2.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "82505 stacks compared against the catalog containing 61592 loci.\n",
      "  5 matching loci, 3 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  3 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  7 total haplotypes examined from matching loci, 2 verified.\n",
      "Outputing to file ./2010_117_2.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_118_1' [1 of 1]\n",
      "  Parsing ustacks_out/2010_118_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_118_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_118_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40583 stacks compared against the catalog containing 61592 loci.\n",
      "  40532 matching loci, 129 contained no verified haplotypes.\n",
      "  128 loci matched more than one catalog locus and were excluded.\n",
      "  1 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  47031 total haplotypes examined from matching loci, 46712 verified.\n",
      "Outputing to file ./2010_118_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_118_2' [1 of 1]\n",
      "  Parsing ustacks_out/2010_118_2.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_118_2.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_118_2.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "94643 stacks compared against the catalog containing 61592 loci.\n",
      "  5 matching loci, 1 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  1 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  5 total haplotypes examined from matching loci, 4 verified.\n",
      "Outputing to file ./2010_118_2.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_119_1' [1 of 1]\n",
      "  Parsing ustacks_out/2010_119_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_119_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_119_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40080 stacks compared against the catalog containing 61592 loci.\n",
      "  38641 matching loci, 1085 contained no verified haplotypes.\n",
      "  138 loci matched more than one catalog locus and were excluded.\n",
      "  947 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  43336 total haplotypes examined from matching loci, 41921 verified.\n",
      "Outputing to file ./2010_119_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_119_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_119_2'\n",
      "Unable to parse 'ustacks_out/2010_119_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_123_1' [1 of 1]\n",
      "  Parsing ustacks_out/2010_123_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_123_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_123_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39593 stacks compared against the catalog containing 61592 loci.\n",
      "  38366 matching loci, 1070 contained no verified haplotypes.\n",
      "  140 loci matched more than one catalog locus and were excluded.\n",
      "  930 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42763 total haplotypes examined from matching loci, 41398 verified.\n",
      "Outputing to file ./2010_123_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_123_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_123_2'\n",
      "Unable to parse 'ustacks_out/2010_123_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_459_1' [1 of 1]\n",
      "  Parsing ustacks_out/2005_459_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2005_459_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2005_459_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38594 stacks compared against the catalog containing 61592 loci.\n",
      "  38550 matching loci, 122 contained no verified haplotypes.\n",
      "  119 loci matched more than one catalog locus and were excluded.\n",
      "  3 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  44212 total haplotypes examined from matching loci, 43932 verified.\n",
      "Outputing to file ./2005_459_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_459_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_459_2'\n",
      "Unable to parse 'ustacks_out/2005_459_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_462_1' [1 of 1]\n",
      "  Parsing ustacks_out/2005_462_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2005_462_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2005_462_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "37319 stacks compared against the catalog containing 61592 loci.\n",
      "  37287 matching loci, 150 contained no verified haplotypes.\n",
      "  147 loci matched more than one catalog locus and were excluded.\n",
      "  3 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42668 total haplotypes examined from matching loci, 42321 verified.\n",
      "Outputing to file ./2005_462_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_462_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_462_2'\n",
      "Unable to parse 'ustacks_out/2005_462_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_463_1' [1 of 1]\n",
      "  Parsing ustacks_out/2005_463_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2005_463_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2005_463_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39641 stacks compared against the catalog containing 61592 loci.\n",
      "  38460 matching loci, 1073 contained no verified haplotypes.\n",
      "  125 loci matched more than one catalog locus and were excluded.\n",
      "  948 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42761 total haplotypes examined from matching loci, 41427 verified.\n",
      "Outputing to file ./2005_463_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_463_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_463_2'\n",
      "Unable to parse 'ustacks_out/2005_463_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_464_1' [1 of 1]\n",
      "  Parsing ustacks_out/2005_464_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2005_464_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2005_464_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38911 stacks compared against the catalog containing 61592 loci.\n",
      "  38911 matching loci, 18 contained no verified haplotypes.\n",
      "  18 loci matched more than one catalog locus and were excluded.\n",
      "  0 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  44730 total haplotypes examined from matching loci, 44685 verified.\n",
      "Outputing to file ./2005_464_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_464_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_464_2'\n",
      "Unable to parse 'ustacks_out/2005_464_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_173_1' [1 of 1]\n",
      "  Parsing ustacks_out/2009_173_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2009_173_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2009_173_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "3761 stacks compared against the catalog containing 61592 loci.\n",
      "  3586 matching loci, 221 contained no verified haplotypes.\n",
      "  64 loci matched more than one catalog locus and were excluded.\n",
      "  157 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  3826 total haplotypes examined from matching loci, 3505 verified.\n",
      "Outputing to file ./2009_173_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_173_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_173_2'\n",
      "Unable to parse 'ustacks_out/2009_173_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_127_1' [1 of 1]\n",
      "  Parsing ustacks_out/2010_127_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_127_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_127_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40481 stacks compared against the catalog containing 61592 loci.\n",
      "  40436 matching loci, 134 contained no verified haplotypes.\n",
      "  132 loci matched more than one catalog locus and were excluded.\n",
      "  2 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  47095 total haplotypes examined from matching loci, 46788 verified.\n",
      "Outputing to file ./2010_127_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_127_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_127_2'\n",
      "Unable to parse 'ustacks_out/2010_127_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_184_1' [1 of 1]\n",
      "  Parsing ustacks_out/2010_184_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_184_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_184_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38663 stacks compared against the catalog containing 61592 loci.\n",
      "  38642 matching loci, 105 contained no verified haplotypes.\n",
      "  98 loci matched more than one catalog locus and were excluded.\n",
      "  7 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  44194 total haplotypes examined from matching loci, 43953 verified.\n",
      "Outputing to file ./2010_184_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_184_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_184_2'\n",
      "Unable to parse 'ustacks_out/2010_184_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_185_1' [1 of 1]\n",
      "  Parsing ustacks_out/2010_185_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_185_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_185_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40010 stacks compared against the catalog containing 61592 loci.\n",
      "  38616 matching loci, 1152 contained no verified haplotypes.\n",
      "  133 loci matched more than one catalog locus and were excluded.\n",
      "  1019 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  43092 total haplotypes examined from matching loci, 41627 verified.\n",
      "Outputing to file ./2010_185_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_185_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_185_2'\n",
      "Unable to parse 'ustacks_out/2010_185_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_187_1' [1 of 1]\n",
      "  Parsing ustacks_out/2010_187_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_187_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_187_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39772 stacks compared against the catalog containing 61592 loci.\n",
      "  38612 matching loci, 1051 contained no verified haplotypes.\n",
      "  142 loci matched more than one catalog locus and were excluded.\n",
      "  909 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  43260 total haplotypes examined from matching loci, 41881 verified.\n",
      "Outputing to file ./2010_187_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_187_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_187_2'\n",
      "Unable to parse 'ustacks_out/2010_187_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_188_1' [1 of 1]\n",
      "  Parsing ustacks_out/2010_188_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_188_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_188_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40324 stacks compared against the catalog containing 61592 loci.\n",
      "  38960 matching loci, 1106 contained no verified haplotypes.\n",
      "  145 loci matched more than one catalog locus and were excluded.\n",
      "  961 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  43562 total haplotypes examined from matching loci, 42131 verified.\n",
      "Outputing to file ./2010_188_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_188_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_188_2'\n",
      "Unable to parse 'ustacks_out/2010_188_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_189_1' [1 of 1]\n",
      "  Parsing ustacks_out/2010_189_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_189_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_189_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40315 stacks compared against the catalog containing 61592 loci.\n",
      "  40280 matching loci, 149 contained no verified haplotypes.\n",
      "  149 loci matched more than one catalog locus and were excluded.\n",
      "  0 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  46684 total haplotypes examined from matching loci, 46319 verified.\n",
      "Outputing to file ./2010_189_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_189_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_189_2'\n",
      "Unable to parse 'ustacks_out/2010_189_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_211_1' [1 of 1]\n",
      "  Parsing ustacks_out/2010_211_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2010_211_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2010_211_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40187 stacks compared against the catalog containing 61592 loci.\n",
      "  38847 matching loci, 1117 contained no verified haplotypes.\n",
      "  145 loci matched more than one catalog locus and were excluded.\n",
      "  972 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  43613 total haplotypes examined from matching loci, 42146 verified.\n",
      "Outputing to file ./2010_211_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_211_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_211_2'\n",
      "Unable to parse 'ustacks_out/2010_211_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_465_1' [1 of 1]\n",
      "  Parsing ustacks_out/2005_465_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2005_465_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2005_465_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "43177 stacks compared against the catalog containing 61592 loci.\n",
      "  43035 matching loci, 81 contained no verified haplotypes.\n",
      "  75 loci matched more than one catalog locus and were excluded.\n",
      "  6 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  51436 total haplotypes examined from matching loci, 51237 verified.\n",
      "Outputing to file ./2005_465_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_465_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_465_2'\n",
      "Unable to parse 'ustacks_out/2005_465_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_466_1' [1 of 1]\n",
      "  Parsing ustacks_out/2005_466_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2005_466_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2005_466_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "43745 stacks compared against the catalog containing 61592 loci.\n",
      "  43570 matching loci, 80 contained no verified haplotypes.\n",
      "  70 loci matched more than one catalog locus and were excluded.\n",
      "  10 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  53686 total haplotypes examined from matching loci, 53512 verified.\n",
      "Outputing to file ./2005_466_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_466_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_466_2'\n",
      "Unable to parse 'ustacks_out/2005_466_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_467_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_467_1'\n",
      "Unable to parse 'ustacks_out/2005_467_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_467_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_467_2'\n",
      "Unable to parse 'ustacks_out/2005_467_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_495_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_495_1'\n",
      "Unable to parse 'ustacks_out/2005_495_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_495_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_495_2'\n",
      "Unable to parse 'ustacks_out/2005_495_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_176_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_176_1'\n",
      "Unable to parse 'ustacks_out/2009_176_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_176_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_176_2'\n",
      "Unable to parse 'ustacks_out/2009_176_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_212_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_212_1'\n",
      "Unable to parse 'ustacks_out/2010_212_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_212_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_212_2'\n",
      "Unable to parse 'ustacks_out/2010_212_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_213_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_213_1'\n",
      "Unable to parse 'ustacks_out/2010_213_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_213_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_213_2'\n",
      "Unable to parse 'ustacks_out/2010_213_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_215_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_215_1'\n",
      "Unable to parse 'ustacks_out/2010_215_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_215_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_215_2'\n",
      "Unable to parse 'ustacks_out/2010_215_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_216_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_216_1'\n",
      "Unable to parse 'ustacks_out/2010_216_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_216_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_216_2'\n",
      "Unable to parse 'ustacks_out/2010_216_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_218_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_218_1'\n",
      "Unable to parse 'ustacks_out/2010_218_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_218_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_218_2'\n",
      "Unable to parse 'ustacks_out/2010_218_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_219_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_219_1'\n",
      "Unable to parse 'ustacks_out/2010_219_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_219_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_219_2'\n",
      "Unable to parse 'ustacks_out/2010_219_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_220_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_220_1'\n",
      "Unable to parse 'ustacks_out/2010_220_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_220_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_220_2'\n",
      "Unable to parse 'ustacks_out/2010_220_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_559_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_559_1'\n",
      "Unable to parse 'ustacks_out/2005_559_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_559_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_559_2'\n",
      "Unable to parse 'ustacks_out/2005_559_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_297_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_297_1'\n",
      "Unable to parse 'ustacks_out/2005_297_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_297_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_297_2'\n",
      "Unable to parse 'ustacks_out/2005_297_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_298_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_298_1'\n",
      "Unable to parse 'ustacks_out/2005_298_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_298_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_298_2'\n",
      "Unable to parse 'ustacks_out/2005_298_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_171_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_171_1'\n",
      "Unable to parse 'ustacks_out/2009_171_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_171_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_171_2'\n",
      "Unable to parse 'ustacks_out/2009_171_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_221_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_221_1'\n",
      "Unable to parse 'ustacks_out/2010_221_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_221_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_221_2'\n",
      "Unable to parse 'ustacks_out/2010_221_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_222_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_222_1'\n",
      "Unable to parse 'ustacks_out/2010_222_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_222_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_222_2'\n",
      "Unable to parse 'ustacks_out/2010_222_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_240_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_240_1'\n",
      "Unable to parse 'ustacks_out/2010_240_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_240_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_240_2'\n",
      "Unable to parse 'ustacks_out/2010_240_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_242_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_242_1'\n",
      "Unable to parse 'ustacks_out/2010_242_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_242_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_242_2'\n",
      "Unable to parse 'ustacks_out/2010_242_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_243_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_243_1'\n",
      "Unable to parse 'ustacks_out/2010_243_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_243_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_243_2'\n",
      "Unable to parse 'ustacks_out/2010_243_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_244_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_244_1'\n",
      "Unable to parse 'ustacks_out/2010_244_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_244_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_244_2'\n",
      "Unable to parse 'ustacks_out/2010_244_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_246_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_246_1'\n",
      "Unable to parse 'ustacks_out/2010_246_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_246_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_246_2'\n",
      "Unable to parse 'ustacks_out/2010_246_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_247_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_247_1'\n",
      "Unable to parse 'ustacks_out/2010_247_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_247_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_247_2'\n",
      "Unable to parse 'ustacks_out/2010_247_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_251_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_251_1'\n",
      "Unable to parse 'ustacks_out/2010_251_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_251_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_251_2'\n",
      "Unable to parse 'ustacks_out/2010_251_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_266_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_266_1'\n",
      "Unable to parse 'ustacks_out/2010_266_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_266_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_266_2'\n",
      "Unable to parse 'ustacks_out/2010_266_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_564_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_564_1'\n",
      "Unable to parse 'ustacks_out/2010_564_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_564_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_564_2'\n",
      "Unable to parse 'ustacks_out/2010_564_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_565_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_565_1'\n",
      "Unable to parse 'ustacks_out/2010_565_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_565_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_565_2'\n",
      "Unable to parse 'ustacks_out/2010_565_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_568_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_568_1'\n",
      "Unable to parse 'ustacks_out/2010_568_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_568_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_568_2'\n",
      "Unable to parse 'ustacks_out/2010_568_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_569_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_569_1'\n",
      "Unable to parse 'ustacks_out/2010_569_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_569_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_569_2'\n",
      "Unable to parse 'ustacks_out/2010_569_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_572_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_572_1'\n",
      "Unable to parse 'ustacks_out/2010_572_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_572_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_572_2'\n",
      "Unable to parse 'ustacks_out/2010_572_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_858_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_858_1'\n",
      "Unable to parse 'ustacks_out/2010_858_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_858_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_858_2'\n",
      "Unable to parse 'ustacks_out/2010_858_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_250_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_250_1'\n",
      "Unable to parse 'ustacks_out/2010_250_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2010_250_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2010_250_2'\n",
      "Unable to parse 'ustacks_out/2010_250_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_537_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_537_1'\n",
      "Unable to parse 'ustacks_out/2009_537_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_537_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_537_2'\n",
      "Unable to parse 'ustacks_out/2009_537_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_124_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_124_1'\n",
      "Unable to parse 'ustacks_out/2009_124_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_124_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_124_2'\n",
      "Unable to parse 'ustacks_out/2009_124_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_169_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_169_1'\n",
      "Unable to parse 'ustacks_out/2009_169_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_169_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_169_2'\n",
      "Unable to parse 'ustacks_out/2009_169_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_465_1' [1 of 1]\n",
      "  Parsing ustacks_out/2005_465_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2005_465_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2005_465_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "43177 stacks compared against the catalog containing 61592 loci.\n",
      "  43035 matching loci, 81 contained no verified haplotypes.\n",
      "  75 loci matched more than one catalog locus and were excluded.\n",
      "  6 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  51436 total haplotypes examined from matching loci, 51237 verified.\n",
      "Outputing to file ./2005_465_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_465_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_465_2'\n",
      "Unable to parse 'ustacks_out/2005_465_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_123_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_123_1'\n",
      "Unable to parse 'ustacks_out/2009_123_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_123_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_123_2'\n",
      "Unable to parse 'ustacks_out/2009_123_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_147_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_147_1'\n",
      "Unable to parse 'ustacks_out/2009_147_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_147_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_147_2'\n",
      "Unable to parse 'ustacks_out/2009_147_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_164_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_164_1'\n",
      "Unable to parse 'ustacks_out/2009_164_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_164_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_164_2'\n",
      "Unable to parse 'ustacks_out/2009_164_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_170_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_170_1'\n",
      "Unable to parse 'ustacks_out/2009_170_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_170_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_170_2'\n",
      "Unable to parse 'ustacks_out/2009_170_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_174_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_174_1'\n",
      "Unable to parse 'ustacks_out/2009_174_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_174_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_174_2'\n",
      "Unable to parse 'ustacks_out/2009_174_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_178_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_178_1'\n",
      "Unable to parse 'ustacks_out/2009_178_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_178_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_178_2'\n",
      "Unable to parse 'ustacks_out/2009_178_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_532_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_532_1'\n",
      "Unable to parse 'ustacks_out/2009_532_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_532_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_532_2'\n",
      "Unable to parse 'ustacks_out/2009_532_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_536_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_536_1'\n",
      "Unable to parse 'ustacks_out/2009_536_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_536_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_536_2'\n",
      "Unable to parse 'ustacks_out/2009_536_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_466_1' [1 of 1]\n",
      "  Parsing ustacks_out/2005_466_1.tags.tsv.gz\n",
      "  Parsing ustacks_out/2005_466_1.snps.tsv.gz\n",
      "  Parsing ustacks_out/2005_466_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "43745 stacks compared against the catalog containing 61592 loci.\n",
      "  43570 matching loci, 80 contained no verified haplotypes.\n",
      "  70 loci matched more than one catalog locus and were excluded.\n",
      "  10 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  53686 total haplotypes examined from matching loci, 53512 verified.\n",
      "Outputing to file ./2005_466_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_466_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_466_2'\n",
      "Unable to parse 'ustacks_out/2005_466_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_141_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_141_1'\n",
      "Unable to parse 'ustacks_out/2009_141_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_141_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_141_2'\n",
      "Unable to parse 'ustacks_out/2009_141_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_124_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_124_1'\n",
      "Unable to parse 'ustacks_out/2009_124_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_124_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_124_2'\n",
      "Unable to parse 'ustacks_out/2009_124_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_299_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_299_1'\n",
      "Unable to parse 'ustacks_out/2005_299_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_299_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_299_2'\n",
      "Unable to parse 'ustacks_out/2005_299_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_460_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_460_1'\n",
      "Unable to parse 'ustacks_out/2005_460_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_460_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_460_2'\n",
      "Unable to parse 'ustacks_out/2005_460_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_461_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_461_1'\n",
      "Unable to parse 'ustacks_out/2005_461_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_461_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_461_2'\n",
      "Unable to parse 'ustacks_out/2005_461_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_384_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_384_1'\n",
      "Unable to parse 'ustacks_out/2005_384_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_384_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_384_2'\n",
      "Unable to parse 'ustacks_out/2005_384_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_385_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_385_1'\n",
      "Unable to parse 'ustacks_out/2005_385_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2005_385_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2005_385_2'\n",
      "Unable to parse 'ustacks_out/2005_385_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_127_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_127_1'\n",
      "Unable to parse 'ustacks_out/2009_127_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_127_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_127_2'\n",
      "Unable to parse 'ustacks_out/2009_127_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_128_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_128_1'\n",
      "Unable to parse 'ustacks_out/2009_128_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_128_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_128_2'\n",
      "Unable to parse 'ustacks_out/2009_128_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_129_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_129_1'\n",
      "Unable to parse 'ustacks_out/2009_129_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_129_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_129_2'\n",
      "Unable to parse 'ustacks_out/2009_129_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_131_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_131_1'\n",
      "Unable to parse 'ustacks_out/2009_131_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_131_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_131_2'\n",
      "Unable to parse 'ustacks_out/2009_131_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_135_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_135_1'\n",
      "Unable to parse 'ustacks_out/2009_135_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_135_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_135_2'\n",
      "Unable to parse 'ustacks_out/2009_135_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_141_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_141_1'\n",
      "Unable to parse 'ustacks_out/2009_141_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2009_141_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2009_141_2'\n",
      "Unable to parse 'ustacks_out/2009_141_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_174_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_174_1'\n",
      "Unable to parse 'ustacks_out/2014_174_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_174_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_174_2'\n",
      "Unable to parse 'ustacks_out/2014_174_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_203_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_203_1'\n",
      "Unable to parse 'ustacks_out/2014_203_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_203_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_203_2'\n",
      "Unable to parse 'ustacks_out/2014_203_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_204_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_204_1'\n",
      "Unable to parse 'ustacks_out/2014_204_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_204_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_204_2'\n",
      "Unable to parse 'ustacks_out/2014_204_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_205_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_205_1'\n",
      "Unable to parse 'ustacks_out/2014_205_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_205_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_205_2'\n",
      "Unable to parse 'ustacks_out/2014_205_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_207_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_207_1'\n",
      "Unable to parse 'ustacks_out/2014_207_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_207_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_207_2'\n",
      "Unable to parse 'ustacks_out/2014_207_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_016_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_016_1'\n",
      "Unable to parse 'ustacks_out/2015_016_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_016_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_016_2'\n",
      "Unable to parse 'ustacks_out/2015_016_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_100_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_100_1'\n",
      "Unable to parse 'ustacks_out/2015_100_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_100_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_100_2'\n",
      "Unable to parse 'ustacks_out/2015_100_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_101_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_101_1'\n",
      "Unable to parse 'ustacks_out/2015_101_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_101_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_101_2'\n",
      "Unable to parse 'ustacks_out/2015_101_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_102_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_102_1'\n",
      "Unable to parse 'ustacks_out/2015_102_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_102_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_102_2'\n",
      "Unable to parse 'ustacks_out/2015_102_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_139_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_139_1'\n",
      "Unable to parse 'ustacks_out/2015_139_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_139_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_139_2'\n",
      "Unable to parse 'ustacks_out/2015_139_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_140_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_140_1'\n",
      "Unable to parse 'ustacks_out/2015_140_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_140_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_140_2'\n",
      "Unable to parse 'ustacks_out/2015_140_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_141_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_141_1'\n",
      "Unable to parse 'ustacks_out/2015_141_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_141_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_141_2'\n",
      "Unable to parse 'ustacks_out/2015_141_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_232_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_232_1'\n",
      "Unable to parse 'ustacks_out/2014_232_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_232_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_232_2'\n",
      "Unable to parse 'ustacks_out/2014_232_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_208_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_208_1'\n",
      "Unable to parse 'ustacks_out/2014_208_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_208_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_208_2'\n",
      "Unable to parse 'ustacks_out/2014_208_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_218_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_218_1'\n",
      "Unable to parse 'ustacks_out/2014_218_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_218_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_218_2'\n",
      "Unable to parse 'ustacks_out/2014_218_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_223_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_223_1'\n",
      "Unable to parse 'ustacks_out/2014_223_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_223_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_223_2'\n",
      "Unable to parse 'ustacks_out/2014_223_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_224_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_224_1'\n",
      "Unable to parse 'ustacks_out/2014_224_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_224_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_224_2'\n",
      "Unable to parse 'ustacks_out/2014_224_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_103_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_103_1'\n",
      "Unable to parse 'ustacks_out/2015_103_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_103_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_103_2'\n",
      "Unable to parse 'ustacks_out/2015_103_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_105_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_105_1'\n",
      "Unable to parse 'ustacks_out/2015_105_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_105_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_105_2'\n",
      "Unable to parse 'ustacks_out/2015_105_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_138_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_138_1'\n",
      "Unable to parse 'ustacks_out/2015_138_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2015_138_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2015_138_2'\n",
      "Unable to parse 'ustacks_out/2015_138_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_128_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_128_1'\n",
      "Unable to parse 'ustacks_out/2014_128_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_128_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_128_2'\n",
      "Unable to parse 'ustacks_out/2014_128_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_129_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_129_1'\n",
      "Unable to parse 'ustacks_out/2014_129_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_129_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_129_2'\n",
      "Unable to parse 'ustacks_out/2014_129_2'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_130_1' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_130_1'\n",
      "Unable to parse 'ustacks_out/2014_130_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing cstacks_out/batch_1.catalog.tags.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.snps.tsv.gz\n",
      "  Parsing cstacks_out/batch_1.catalog.alleles.tsv.gz\n",
      "Processing sample 'ustacks_out/2014_130_2' [1 of 1]\n",
      " Unable to open 'ustacks_out/2014_130_2'\n",
      "Unable to parse 'ustacks_out/2014_130_2'\n"
     ]
    }
   ],
   "source": [
    "!python pypipe_sstacks.py new_filenames_shell.txt 1 cstacks_out/batch_1 ustacks_out 10 -o sstacks_out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "populations parameters to use -p 2, -r .50, -m 5, -t 10"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
