{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Purpose\n",
    "\n",
    "The purpose of this notebook is to document my full application of the Stacks pipeline on my cod data. I've been writing scripts to work on a subset of my files, and now I want to start running them on all of my files.\n",
    "\n",
    "## ``process_radtags`` \n",
    "\n",
    "I don't have to run this on my data because the previous user of the data already did. In my script for running ``process_radtags``, however, I also use this opportunity to make folders for the output of each of the Stacks programs. So I'm doing that manually here. Below, you can see these directories plus two folders of cod data (two lanes of data):\n",
    "\n",
    "![image](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Notebooks/images_for_notebooks/Screen%20Shot%202016-11-16%20at%203.10.45%20PM.png?raw=true)\n",
    "\n",
    "## ``ustacks`` \n",
    "\n",
    "I'll be working off of my hard drive, so first I had to change directory to there."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Time\\ Machine\\ Backups/Cod-Time-Series-Data/ "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Another issue is that I have two lanes. It'd be easiest if I could work in only one directory, but their barcodes are redundant. So my first step is to rename the files by adding *_L1* for Library 1 and *_L2* for Library 2. I'll do that in python before the file extension. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here's the content of the script ``add_lib_to_filename.py``:\n",
    "\n",
    "```\n",
    "# arguments\n",
    "# [1] directory name with files to rename \n",
    "# [2] text you would like to add to the end of each filename before file extensions\n",
    "\n",
    "# assumptions\n",
    "# [1] file extention = fq.qz\n",
    "# [2] you are one directory above the directory that you want to rename the files of\n",
    "# [3] your files aren't already renamed! --- write an if else loop to make sure it hasn't been run yet when you have the time\n",
    "\n",
    "import sys\n",
    "import subprocess\n",
    "\n",
    "# make a text file that has each thing in this directory on a line; each line should be each file in the lane\n",
    "\n",
    "string1 = \"\" # initiate string\n",
    "\n",
    "string1 += 'ls ' + sys.argv[1] + ' > ./dircontents.txt' # add ls, wd, redirect to text file\n",
    "\n",
    "firstshell = open(\"getcontents_shell.txt\", \"w\")\n",
    "firstshell.write(string1)\n",
    "firstshell.close()\n",
    "\n",
    "subprocess.call([\"sh getcontents_shell.txt\"], shell = True)\n",
    "\n",
    "# now that file is made for contents of directory, read that file in\n",
    "contents = open(\"dircontents.txt\", \"r\")\n",
    "\n",
    "\n",
    "# second shell = renaming script\n",
    "rename_w_lib_shell = open(\"rename_w_lib_shell.txt\", \"w\")\n",
    "\n",
    "string2 = \"\"\n",
    "string2 += \"cd \" + sys.argv[1] + \"\\n\"\n",
    "\n",
    "\n",
    "# loop that will break the script if it's already run to make sure you don't rename files into something wrong!\n",
    "for line in contents:\n",
    "\tlinelist = line.strip().split(\".\")\n",
    "\tif linelist[0][-4] == \"L\":\n",
    "\t\tprint \"CHECK TO SEE IF YOU HAVE ALREADY RENAMED THESE FILES. SCRIPT PAUSED AS WARNING BECAUSE FILES APPEAR RENAMED.\"\n",
    "\t\tsys.exit() # exit script\n",
    "\telse:\n",
    "\t\tcontinue\n",
    "contents.close()\n",
    "\n",
    "\n",
    "\n",
    "contents = open(\"dircontents.txt\", \"r\")\n",
    "for line in contents:\n",
    "\toriginal = line.strip()\n",
    "\tfilename_list = line.strip().split(\".\") # make list out of whole file name with extensions\n",
    "\tfilename_wo_ext = filename_list[0]\n",
    "\tfile_ext = \".\" + filename_list[1] + \".\" + filename_list[2]\n",
    "\twordlist = filename_wo_ext.split(\"_\") # split filename without extensions into parts by underscore\n",
    "\tnewfilename = wordlist[0] + sys.argv[2] + \"_\" + wordlist[1]\n",
    "\tnewfilename += file_ext\n",
    "\tstring2 += \"mv\" + \"\\t\" + original + \"\\t\" + newfilename + \"\\n\"\n",
    "\n",
    "rename_w_lib_shell.write(string2)\n",
    "rename_w_lib_shell.close()\n",
    "contents.close()\n",
    "\n",
    "# call renaming shell script\n",
    "subprocess.call([\"sh rename_w_lib_shell.txt\"], shell = True)\n",
    "```\n",
    "\n",
    "Running the code looks like this:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!python add_lib_to_filename.py process_radtags_out/cod_lib1 _L1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!python add_lib_to_filename.py process_radtags_out/cod_lib2 _L2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now my files are renamed with the library information after the barcode before their 1 or 2 that was used to signify forward and reverse for paired end data.\n",
    "<br><br>\n",
    "From\n",
    "<br>\n",
    "![image](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Notebooks/images_for_notebooks/Screen%20Shot%202016-11-16%20at%206.15.59%20PM.png?raw=true)\n",
    "<br>\n",
    "To\n",
    "![image](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Notebooks/images_for_notebooks/Screen%20Shot%202016-11-16%20at%206.15.48%20PM.png?raw=true)\n",
    "<br>\n",
    "<br>\n",
    "And then I can group the two directories of libraries together.\n",
    "<br>\n",
    "![image](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Notebooks/images_for_notebooks/Screen%20Shot%202016-11-17%20at%209.09.21%20AM.png?raw=true)\n",
    "<br>\n",
    "Now I want to run ``ustacks`` on all the files in this folder. I can use my handy dandy script I wrote a couple of weeks ago. It looks like this:\n",
    "\n",
    "```\n",
    "##########################################################################################\n",
    "# \n",
    "### ``ustacks``\n",
    "#\n",
    "# PURPOSE: This step aligns identical RAD tags within an individual into stacks & provides data for calling SNPs\n",
    "# INPUT: fastq or gzfastq files\n",
    "# OUTPUT: 4 files - alleles, modules, snps, tags\n",
    "# \n",
    "# \n",
    "### WHEN RUNNING THIS SCRIPT, YOUR INPUTS AT THE COMMAND LINE ARE:\n",
    "# python  \n",
    "# {0}[pipeline filename] \n",
    "# {1}[barcodes & samples textfile] \n",
    "# {2}[start directory] \n",
    "# {3}[end directory]\n",
    "# \n",
    "### DEPENDENCIES\n",
    "# \n",
    "# [1] You need a file where first column is barcode and second is unique sample name\n",
    "# \n",
    "### WARNINGS\n",
    "# \n",
    "# [1] ustacks only works when your working directory is one direcotry above the folders you are\n",
    "#\t\tcalling and storing data in \n",
    "# \n",
    "##########################################################################################\n",
    "\n",
    "# --- [A] call necessary modules\n",
    "\n",
    "import subprocess\n",
    "import sys \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# --- [B] Rename your files by the sample name\n",
    "\n",
    "new_file = open(\"new_filenames_shell.txt\", \"w\") # new txt file\n",
    "dir = sys.argv[2] # directory files that need names changed\n",
    "firststr = \"cd \" + dir + \"\\n\" + \"pwd\\n\"\n",
    "new_file.write(firststr)\n",
    "\n",
    "myfile = open(sys.argv[1], \"r\")\t\t\t\t# myfile =  tab delimited file mentioned, called after your script\n",
    "\n",
    "for line in myfile:\t\t\t\t\t\t   # loop through each line\n",
    "\tlinelist=line.strip().split()\t\t   # splits into list by white space\n",
    "\tbarcodefile1 = linelist[0] + \"_1.fq.gz\" # forward\n",
    "\tbarcodefile2 = linelist[0] + \"_2.fq.gz\" # reverse\n",
    "\tsamplefile1 = linelist[1] + \"_1.fq.gz\" # forward\n",
    "\tsamplefile2 = linelist[1] + \"_2.fq.gz\" # reverse\n",
    "\tnewstring = \"mv\" + \"\\t\" + barcodefile1 + \"\\t\" + samplefile1 + \"\\n\" + \"mv\" + \"\\t\" + barcodefile2 + \"\\t\" + samplefile2 + \"\\n\"\n",
    "\t# print newstring  # troubleshooting loop\n",
    "\tnew_file.write(newstring)\n",
    "\n",
    "myfile.close()\n",
    "new_file.close()\n",
    "\n",
    "# run the script you just made as a shell script to rename your files\n",
    "subprocess.call(['sh new_filenames_shell.txt'], shell=True)\n",
    "\n",
    " \n",
    "\n",
    "\n",
    "\n",
    "### --- [C] Make shell script to run all samples through command line through ``ustacks``\n",
    "\n",
    "# ``ustacks`` requires an arbitrary integer for every sample, although unclear how it gets used as it does not become the name of the file\n",
    "\n",
    "\n",
    "# name your 'from' and 'to' directories that will go in each line of your ustacks shell script\n",
    "dirfrom = sys.argv[2]\n",
    "dirto = sys.argv[3]\n",
    "\n",
    "newfile2 = open(\"ustacks_shell.txt\", \"w\")\t # make ustacks shell script to run through terminal\n",
    "myfile = open(\"new_filenames_shell.txt\", \"r\")\t#open the file with a list of barcodes + sample IDs\n",
    "\n",
    "# dir = sys.argv[2] # directory files that need names changed\n",
    "# firststr = \"cd \" + dir + \"\\n\"\n",
    "# new_file.write(firststr)\n",
    "\n",
    "dir2 = sys.argv[2] # directory with files that we want to run ustacks on\n",
    "\n",
    "ID_int = 001\t\t\t\t\t\t\t\t# start integer counter\n",
    "\n",
    "lines = myfile.readlines()[2:] # skip first two lines because just cd and pwd\n",
    "\n",
    "\n",
    "ID_int = 001\t\t\t\t\t\t\t\t# start integer counter\n",
    "for line in lines: \t\t\t#for each line in the barcode file\t\n",
    "\tlinelist=line.strip().split()\t\n",
    "\tsampID = linelist[2] \t\t\t\t\t#save the second object as \"sampID\"\n",
    "\tif ID_int < 10: \n",
    "\t\tustacks_code = \"ustacks -t gzfastq -f \" + dirfrom + \"/\" + sampID + \" -r -d -o \" + dirto + \" -i 00\" + str(ID_int) + \" -m 5 -M 3 -p 10\" + \"\\n\"\n",
    "\t\t\t\t\t\t\t\t#create a line of code for ustacks that includes the new sample ID (with 2 leading 0s)\n",
    "\telif ID_int >= 10 & ID_int < 100: \n",
    "\t\tustacks_code = \"ustacks -t gzfastq -f \" + dirfrom + \"/\" + sampID + \" -r -d -o \" + dirto + \" -i 0\" + str(ID_int) + \" -m 5 -M 3 -p 10\" + \"\\n\"\n",
    "\t\t\t\t\t\t\t\t#create a line of code for ustacks that includes the new sample ID (with 1 leading 0)\n",
    "\telse: \n",
    "\t\tustacks_code = \"ustacks -t gzfastq -f \" + dirfrom + \"/\" + sampID + \" -r -d -o \" + dirto + \" -i \" + str(ID_int) + \" -m 5 -M 3 -p 10\" + \"\\n\"\n",
    "\t\t\t\t\t\t\t\t#create a line of code for ustacks that includes the new sample ID (with no leading 0s)\n",
    "\tnewfile2.write(ustacks_code)\t#append this new line of code to the output file\n",
    "\tID_int += 1\n",
    "\n",
    "myfile.close()\n",
    "newfile2.close()\n",
    "\n",
    "# run this new script through the terminal\n",
    "subprocess.call(['sh ustacks_shell.txt'], shell=True)\n",
    "\n",
    "##########################################################################################\n",
    "\n",
    "```\n",
    "\n",
    "And I run it like this:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data/process_radtags_out\n",
      "mv: AAACGG_L1_1.fq.gz: No such file or directory\n",
      "mv: AAACGG_L1_2.fq.gz: No such file or directory\n",
      "mv: GCCGTA_L1_1.fq.gz: No such file or directory\n",
      "mv: GCCGTA_L1_2.fq.gz: No such file or directory\n",
      "mv: ACTCTT_L1_1.fq.gz: No such file or directory\n",
      "mv: ACTCTT_L1_2.fq.gz: No such file or directory\n",
      "mv: TTCTAG_L1_1.fq.gz: No such file or directory\n",
      "mv: TTCTAG_L1_2.fq.gz: No such file or directory\n",
      "mv: ATTCCG_L1_1.fq.gz: No such file or directory\n",
      "mv: ATTCCG_L1_2.fq.gz: No such file or directory\n",
      "mv: CCGCAT_L1_1.fq.gz: No such file or directory\n",
      "mv: CCGCAT_L1_2.fq.gz: No such file or directory\n",
      "mv: CGAGGC_L1_1.fq.gz: No such file or directory\n",
      "mv: CGAGGC_L1_2.fq.gz: No such file or directory\n",
      "mv: CGCAGA_L1_1.fq.gz: No such file or directory\n",
      "mv: CGCAGA_L1_2.fq.gz: No such file or directory\n",
      "mv: GAGAGA_L1_1.fq.gz: No such file or directory\n",
      "mv: GAGAGA_L1_2.fq.gz: No such file or directory\n",
      "mv: GGGGCG_L1_1.fq.gz: No such file or directory\n",
      "mv: GGGGCG_L1_2.fq.gz: No such file or directory\n",
      "mv: TAATTC_L1_1.fq.gz: No such file or directory\n",
      "mv: TAATTC_L1_2.fq.gz: No such file or directory\n",
      "mv: TTTAAT_L1_1.fq.gz: No such file or directory\n",
      "mv: TTTAAT_L1_2.fq.gz: No such file or directory\n",
      "mv: AGCCAT_L1_1.fq.gz: No such file or directory\n",
      "mv: AGCCAT_L1_2.fq.gz: No such file or directory\n",
      "mv: AGTAGG_L1_1.fq.gz: No such file or directory\n",
      "mv: AGTAGG_L1_2.fq.gz: No such file or directory\n",
      "mv: ATAGTA_L1_1.fq.gz: No such file or directory\n",
      "mv: ATAGTA_L1_2.fq.gz: No such file or directory\n",
      "mv: CGGTCC_L1_1.fq.gz: No such file or directory\n",
      "mv: CGGTCC_L1_2.fq.gz: No such file or directory\n",
      "mv: CCATTT_L1_1.fq.gz: No such file or directory\n",
      "mv: CCATTT_L1_2.fq.gz: No such file or directory\n",
      "mv: CAGGCA_L1_1.fq.gz: No such file or directory\n",
      "mv: CAGGCA_L1_2.fq.gz: No such file or directory\n",
      "mv: GCAGAT_L1_1.fq.gz: No such file or directory\n",
      "mv: GCAGAT_L1_2.fq.gz: No such file or directory\n",
      "mv: GTCATC_L1_1.fq.gz: No such file or directory\n",
      "mv: GTCATC_L1_2.fq.gz: No such file or directory\n",
      "mv: GACTCT_L1_1.fq.gz: No such file or directory\n",
      "mv: GACTCT_L1_2.fq.gz: No such file or directory\n",
      "mv: TATCAC_L1_1.fq.gz: No such file or directory\n",
      "mv: TATCAC_L1_2.fq.gz: No such file or directory\n",
      "mv: TGGGGA_L1_1.fq.gz: No such file or directory\n",
      "mv: TGGGGA_L1_2.fq.gz: No such file or directory\n",
      "mv: TCTCGG_L1_1.fq.gz: No such file or directory\n",
      "mv: TCTCGG_L1_2.fq.gz: No such file or directory\n",
      "mv: TGACAA_L2_1.fq.gz: No such file or directory\n",
      "mv: TGACAA_L2_2.fq.gz: No such file or directory\n",
      "mv: AATATC_L2_1.fq.gz: No such file or directory\n",
      "mv: AATATC_L2_2.fq.gz: No such file or directory\n",
      "mv: ACAGCG_L2_1.fq.gz: No such file or directory\n",
      "mv: ACAGCG_L2_2.fq.gz: No such file or directory\n",
      "mv: GATCGT_L2_1.fq.gz: No such file or directory\n",
      "mv: GATCGT_L2_2.fq.gz: No such file or directory\n",
      "mv: GGTTTG_L2_1.fq.gz: No such file or directory\n",
      "mv: GGTTTG_L2_2.fq.gz: No such file or directory\n",
      "mv: CTCGCC_L2_1.fq.gz: No such file or directory\n",
      "mv: CTCGCC_L2_2.fq.gz: No such file or directory\n",
      "mv: GTGCCT_L2_1.fq.gz: No such file or directory\n",
      "mv: GTGCCT_L2_2.fq.gz: No such file or directory\n",
      "mv: TAGTAT_L2_1.fq.gz: No such file or directory\n",
      "mv: TAGTAT_L2_2.fq.gz: No such file or directory\n",
      "mv: ACAAGA_L2_1.fq.gz: No such file or directory\n",
      "mv: ACAAGA_L2_2.fq.gz: No such file or directory\n",
      "mv: TTCCGT_L2_1.fq.gz: No such file or directory\n",
      "mv: TTCCGT_L2_2.fq.gz: No such file or directory\n",
      "mv: CCCTAA_L2_1.fq.gz: No such file or directory\n",
      "mv: CCCTAA_L2_2.fq.gz: No such file or directory\n",
      "mv: AGGGTC_L2_1.fq.gz: No such file or directory\n",
      "mv: AGGGTC_L2_2.fq.gz: No such file or directory\n",
      "mv: CAATCG_L2_1.fq.gz: No such file or directory\n",
      "mv: CAATCG_L2_2.fq.gz: No such file or directory\n",
      "mv: GAACCA_L2_1.fq.gz: No such file or directory\n",
      "mv: GAACCA_L2_2.fq.gz: No such file or directory\n",
      "mv: TCGGAC_L2_1.fq.gz: No such file or directory\n",
      "mv: TCGGAC_L2_2.fq.gz: No such file or directory\n",
      "mv: ATGTTG_L2_1.fq.gz: No such file or directory\n",
      "mv: ATGTTG_L2_2.fq.gz: No such file or directory\n",
      "mv: AACTGA_L2_1.fq.gz: No such file or directory\n",
      "mv: AACTGA_L2_2.fq.gz: No such file or directory\n",
      "mv: TCTTCT_L2_1.fq.gz: No such file or directory\n",
      "mv: TCTTCT_L2_2.fq.gz: No such file or directory\n",
      "mv: GGACTT_L2_1.fq.gz: No such file or directory\n",
      "mv: GGACTT_L2_2.fq.gz: No such file or directory\n",
      "mv: GGGCGC_L2_1.fq.gz: No such file or directory\n",
      "mv: GGGCGC_L2_2.fq.gz: No such file or directory\n",
      "mv: ATCAAA_L2_1.fq.gz: No such file or directory\n",
      "mv: ATCAAA_L2_2.fq.gz: No such file or directory\n",
      "mv: GGCAAG_L2_1.fq.gz: No such file or directory\n",
      "mv: GGCAAG_L2_2.fq.gz: No such file or directory\n",
      "mv: CATACT_L2_1.fq.gz: No such file or directory\n",
      "mv: CATACT_L2_2.fq.gz: No such file or directory\n",
      "mv: CGCGTG_L2_1.fq.gz: No such file or directory\n",
      "mv: CGCGTG_L2_2.fq.gz: No such file or directory\n",
      "mv: AACGTT_L2_1.fq.gz: No such file or directory\n",
      "mv: AACGTT_L2_2.fq.gz: No such file or directory\n",
      "mv: AAGACG_L2_1.fq.gz: No such file or directory\n",
      "mv: AAGACG_L2_2.fq.gz: No such file or directory\n",
      "mv: AAGCTA_L2_1.fq.gz: No such file or directory\n",
      "mv: AAGCTA_L2_2.fq.gz: No such file or directory\n",
      "mv: AATGAG_L2_1.fq.gz: No such file or directory\n",
      "mv: AATGAG_L2_2.fq.gz: No such file or directory\n",
      "mv: ACATAC_L2_1.fq.gz: No such file or directory\n",
      "mv: ACATAC_L2_2.fq.gz: No such file or directory\n",
      "mv: ACCATG_L2_1.fq.gz: No such file or directory\n",
      "mv: ACCATG_L2_2.fq.gz: No such file or directory\n",
      "mv: ACCCCC_L2_1.fq.gz: No such file or directory\n",
      "mv: ACCCCC_L2_2.fq.gz: No such file or directory\n",
      "mv: ACTGGC_L2_1.fq.gz: No such file or directory\n",
      "mv: ACTGGC_L2_2.fq.gz: No such file or directory\n",
      "mv: AGCGCA_L2_1.fq.gz: No such file or directory\n",
      "mv: AGCGCA_L2_2.fq.gz: No such file or directory\n",
      "mv: AGGTGT_L2_1.fq.gz: No such file or directory\n",
      "mv: AGGTGT_L2_2.fq.gz: No such file or directory\n",
      "mv: AGTTAA_L2_1.fq.gz: No such file or directory\n",
      "mv: AGTTAA_L2_2.fq.gz: No such file or directory\n",
      "mv: ATGCAC_L2_1.fq.gz: No such file or directory\n",
      "mv: ATGCAC_L2_2.fq.gz: No such file or directory\n",
      "mv: GAAATG_L1_1.fq.gz: No such file or directory\n",
      "mv: GAAATG_L1_2.fq.gz: No such file or directory\n",
      "mv: GACGAC_L1_1.fq.gz: No such file or directory\n",
      "mv: GACGAC_L1_2.fq.gz: No such file or directory\n",
      "mv: GCATGG_L1_1.fq.gz: No such file or directory\n",
      "mv: GCATGG_L1_2.fq.gz: No such file or directory\n",
      "mv: GCGACC_L1_1.fq.gz: No such file or directory\n",
      "mv: GCGACC_L1_2.fq.gz: No such file or directory\n",
      "mv: GCGCTG_L1_1.fq.gz: No such file or directory\n",
      "mv: GCGCTG_L1_2.fq.gz: No such file or directory\n",
      "mv: GCTCAA_L1_1.fq.gz: No such file or directory\n",
      "mv: GCTCAA_L1_2.fq.gz: No such file or directory\n",
      "mv: GGTACA_L1_1.fq.gz: No such file or directory\n",
      "mv: GGTACA_L1_2.fq.gz: No such file or directory\n",
      "mv: GTAAGT_L1_1.fq.gz: No such file or directory\n",
      "mv: GTAAGT_L1_2.fq.gz: No such file or directory\n",
      "mv: GTATCC_L1_1.fq.gz: No such file or directory\n",
      "mv: GTATCC_L1_2.fq.gz: No such file or directory\n",
      "mv: GTGTAA_L1_1.fq.gz: No such file or directory\n",
      "mv: GTGTAA_L1_2.fq.gz: No such file or directory\n",
      "mv: GTTGGA_L1_1.fq.gz: No such file or directory\n",
      "mv: GTTGGA_L1_2.fq.gz: No such file or directory\n",
      "mv: TTTGTC_L1_1.fq.gz: No such file or directory\n",
      "mv: TTTGTC_L1_2.fq.gz: No such file or directory\n",
      "mv: TAAGCT_L2_1.fq.gz: No such file or directory\n",
      "mv: TAAGCT_L2_2.fq.gz: No such file or directory\n",
      "mv: TACACA_L2_1.fq.gz: No such file or directory\n",
      "mv: TACACA_L2_2.fq.gz: No such file or directory\n",
      "mv: TACGGG_L2_1.fq.gz: No such file or directory\n",
      "mv: TACGGG_L2_2.fq.gz: No such file or directory\n",
      "mv: TCAAAG_L2_1.fq.gz: No such file or directory\n",
      "mv: TCAAAG_L2_2.fq.gz: No such file or directory\n",
      "mv: TCCTGC_L2_1.fq.gz: No such file or directory\n",
      "mv: TCCTGC_L2_2.fq.gz: No such file or directory\n",
      "mv: TCGCCA_L2_1.fq.gz: No such file or directory\n",
      "mv: TCGCCA_L2_2.fq.gz: No such file or directory\n",
      "mv: TGAACC_L2_1.fq.gz: No such file or directory\n",
      "mv: TGAACC_L2_2.fq.gz: No such file or directory\n",
      "mv: TGCCCG_L2_1.fq.gz: No such file or directory\n",
      "mv: TGCCCG_L2_2.fq.gz: No such file or directory\n",
      "mv: TGCTTA_L2_1.fq.gz: No such file or directory\n",
      "mv: TGCTTA_L2_2.fq.gz: No such file or directory\n",
      "mv: TTATGA_L2_1.fq.gz: No such file or directory\n",
      "mv: TTATGA_L2_2.fq.gz: No such file or directory\n",
      "mv: TTGAGC_L2_1.fq.gz: No such file or directory\n",
      "mv: TTGAGC_L2_2.fq.gz: No such file or directory\n",
      "mv: GAAATG_L2_1.fq.gz: No such file or directory\n",
      "mv: GAAATG_L2_2.fq.gz: No such file or directory\n",
      "mv: GACGAC_L2_1.fq.gz: No such file or directory\n",
      "mv: GACGAC_L2_2.fq.gz: No such file or directory\n",
      "mv: GCATGG_L2_1.fq.gz: No such file or directory\n",
      "mv: GCATGG_L2_2.fq.gz: No such file or directory\n",
      "mv: GCGACC_L2_1.fq.gz: No such file or directory\n",
      "mv: GCGACC_L2_2.fq.gz: No such file or directory\n",
      "mv: GCGCTG_L2_1.fq.gz: No such file or directory\n",
      "mv: GCGCTG_L2_2.fq.gz: No such file or directory\n",
      "mv: GCTCAA_L2_1.fq.gz: No such file or directory\n",
      "mv: GCTCAA_L2_2.fq.gz: No such file or directory\n",
      "mv: GGTACA_L2_1.fq.gz: No such file or directory\n",
      "mv: GGTACA_L2_2.fq.gz: No such file or directory\n",
      "mv: GTAAGT_L2_1.fq.gz: No such file or directory\n",
      "mv: GTAAGT_L2_2.fq.gz: No such file or directory\n",
      "mv: GTATCC_L2_1.fq.gz: No such file or directory\n",
      "mv: GTATCC_L2_2.fq.gz: No such file or directory\n",
      "mv: GTGTAA_L2_1.fq.gz: No such file or directory\n",
      "mv: GTGTAA_L2_2.fq.gz: No such file or directory\n",
      "mv: GTTGGA_L2_1.fq.gz: No such file or directory\n",
      "mv: GTTGGA_L2_2.fq.gz: No such file or directory\n",
      "mv: TTTGTC_L2_1.fq.gz: No such file or directory\n",
      "mv: TTTGTC_L2_2.fq.gz: No such file or directory\n",
      "mv: TAAGCT_L1_1.fq.gz: No such file or directory\n",
      "mv: TAAGCT_L1_2.fq.gz: No such file or directory\n",
      "mv: TACACA_L1_1.fq.gz: No such file or directory\n",
      "mv: TACACA_L1_2.fq.gz: No such file or directory\n",
      "mv: TACGGG_L1_1.fq.gz: No such file or directory\n",
      "mv: TACGGG_L1_2.fq.gz: No such file or directory\n",
      "mv: TCAAAG_L1_1.fq.gz: No such file or directory\n",
      "mv: TCAAAG_L1_2.fq.gz: No such file or directory\n",
      "mv: TCCTGC_L1_1.fq.gz: No such file or directory\n",
      "mv: TCCTGC_L1_2.fq.gz: No such file or directory\n",
      "mv: TCGCCA_L1_1.fq.gz: No such file or directory\n",
      "mv: TCGCCA_L1_2.fq.gz: No such file or directory\n",
      "mv: TGAACC_L1_1.fq.gz: No such file or directory\n",
      "mv: TGAACC_L1_2.fq.gz: No such file or directory\n",
      "mv: TGCCCG_L1_1.fq.gz: No such file or directory\n",
      "mv: TGCCCG_L1_2.fq.gz: No such file or directory\n",
      "mv: TGCTTA_L1_1.fq.gz: No such file or directory\n",
      "mv: TGCTTA_L1_2.fq.gz: No such file or directory\n",
      "mv: TTATGA_L1_1.fq.gz: No such file or directory\n",
      "mv: TTATGA_L1_2.fq.gz: No such file or directory\n",
      "mv: TTGAGC_L1_1.fq.gz: No such file or directory\n",
      "mv: TTGAGC_L1_2.fq.gz: No such file or directory\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2005_387_1.fq.gz\n",
      "Loading RAD-Tags...done\n",
      "Loaded 337044 RAD-Tags.\n",
      "  Inserted 107931 elements into the RAD-Tags hash map.\n",
      "  196 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 9.87009; Std Dev: 16.9809; Max: 1341\n",
      "Deleveraging trigger: 27; Removal trigger: 44\n",
      "21754 initial stacks were populated; 86177 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 71 (74 k-mers per read); 3 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 457 stacks.\n",
      "  21390 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 9.13598; Std Dev: 33.5736; Max: 44\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  21390 stacks merged into 20263 stacks; deleveraged 123 stacks; removed 47 stacks.\n",
      "After merging, coverage depth Mean: 9.54346; Std Dev: 34.6848; Max: 86\n",
      "Merging remainder radtags\n",
      "  122330 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 23 (122 k-mers per read); 7 k-mer hits required.\n",
      "  Matched 41331 remainder reads; unable to match 80999 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 11.5974; Std Dev: 34.9511; Max: 252\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 256045\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2005_387_1.fq.gz... read 337044 sequence IDs.\n",
      "done.\n",
      "ustacks paramters selected:\n",
      "  Min depth of coverage to create a stack: 5\n",
      "  Max distance allowed between stacks: 3\n",
      "  Max distance allowed to align secondary reads: 5\n",
      "  Max number of stacks allowed per de novo locus: 3\n",
      "  Deleveraging algorithm: enabled\n",
      "  Removal algorithm: enabled\n",
      "  Model type: SNP\n",
      "  Alpha significance level for model: 0.05\n",
      "  Gapped alignments: disabled\n",
      "Parsing ./process_radtags_out/2005_387_2.fq.gz\n",
      "Loading RAD-Tags...done\n",
      "Loaded 337044 RAD-Tags.\n",
      "  Inserted 217488 elements into the RAD-Tags hash map.\n",
      "  3228 reads contained uncalled nucleotides that were modified.\n",
      "Initial coverage mean: 5.88971; Std Dev: 2.26641; Max: 65\n",
      "Deleveraging trigger: 8; Removal trigger: 10\n",
      "5567 initial stacks were populated; 211921 stacks were set aside as secondary reads.\n",
      "Calculating distance for removing repetitive stacks.\n",
      "  Distance allowed between stacks: 1; searching with a k-mer length of 75 (76 k-mers per read); 1 k-mer hits required.\n",
      "Removing repetitive stacks.\n",
      "  Removed 164 stacks.\n",
      "  5530 stacks remain for merging.\n",
      "Post-Repeat Removal, coverage depth Mean: 5.64131; Std Dev: 3.53766; Max: 10\n",
      "Calculating distance between stacks...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 37 (114 k-mers per read); 3 k-mer hits required.\n",
      "Merging stacks, maximum allowed distance: 3 nucleotide(s)\n",
      "  5530 stacks merged into 5458 stacks; deleveraged 8 stacks; removed 0 stacks.\n",
      "After merging, coverage depth Mean: 5.7175; Std Dev: 3.66137; Max: 21\n",
      "Merging remainder radtags\n",
      "  304256 remainder sequences left to merge.\n",
      "  Distance allowed between stacks: 5; searching with a k-mer length of 25 (126 k-mers per read); 1 k-mer hits required.\n",
      "  Matched 4983 remainder reads; unable to match 299273 remainder reads.\n",
      "After remainders merged, coverage depth Mean: 6.65222; Std Dev: 4.94798; Max: 78\n",
      "Calling final consensus sequences, invoking SNP-calling model...\n",
      "Number of utilized reads: 37771\n",
      "Writing loci, SNPs, and alleles to './ustacks_out/'...\n",
      "  Refetching sequencing IDs from ./process_radtags_out/2005_387_2.fq.gz... read 337044 sequence IDs.\n"
     ]
    }
   ],
   "source": [
    "!python pypipe_ustacks.py barcodes_samplenames.txt ./process_radtags_out ./ustacks_out"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note that a lot of the files aren't my data, I should only have 106 individuals out of the two lanes, and it's paired end so there should be 212 files that are mine. Also, this kept crashing my jupyter notebook, so I ran it in the terminal outside the notebook. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
