{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## STACKS+BOWTIE+BLAST Pipeline for Population Genomics Analysis\n",
    "\n",
    "This is the full pipeline that I'm using to analyze my Pacific cod time series data for FISH546. Our lab uses the [Stacks](https://www.google.com/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=stacks%20catchen) pipeline, followed by filtering our catalog with Bowtie and BLAST to filter down our catalog of loci into a de novo reference genome, and then rerunning the Stacks pipeline. My lab mate [Mary Fisher](https://github.com/mfisher5) made a great diagram of the flow of steps in our lab's pipeline.\n",
    "\n",
    "![image](https://github.com/mfisher5/mf-fish546-2016/raw/master/Diagrams/flowchart1.png)\n",
    "\n",
    "\n",
    "### Steps\n",
    "\n",
    "#### 1. Add library identifier to file name (optional)\n",
    "When combining sequence data from different sequencing lanes but with redundant barcodes, I add an identifier for the library number (or lane number) so that I can process all of the files together.\n",
    "\n",
    "#### 2. ``ustacks``\n",
    "``ustacks`` reads fastq files and makes \"stacks\" of identical reads, which can be used to locate tags call SNPs within individuals.\n",
    "\n",
    "#### 3.  ``cstacks``\n",
    "``cstacks`` uses a subset of your individuals to make a catalog of loci that you will use to genotype all of your individuals. \n",
    "\n",
    "#### 4. ``sstacks``\n",
    "``sstacks`` uses the catalog made in ``cstacks`` to genotype your individuals.\n",
    "\n",
    "#### 5. ``populations``\n",
    "``populations`` assigns individuals to populations and produces a genepop file and other summary statistics files.\n",
    "\n",
    "#### 6. Filter catalog with ``bowtie``\n",
    "We use ``bowtie`` align our catalog to itself. This filters out loci that align to different sequences in addition to aligning to themselves. This is supposed to remove highly repetitive tags that could interfere with our analysis.\n",
    "\n",
    "#### 7. Filter catalog with ``BLAST``\n",
    "We ``BLAST`` our catalog to itself to filter out more loci. This is supposed to remove highly repetitive tags that could interfere with our analysis.\n",
    "\n",
    "#### 8. Create final reference genome and align reads with ``bowtie``\n",
    "We use ``bowtie`` for a second time to align all of our reads to our filtered catalog, which is the input for ``pstacks``.\n",
    "\n",
    "#### 9. ``pstacks``\n",
    "``Pstacks`` works like ``ustacks``, except instead of aligning reads to themselves, it identifies stacks in the alignment file. It uses these stacks to find tags and call SNPs within inviduals.\n",
    "\n",
    "#### 10. ``cstacks``\n",
    "This second round of ``cstacks`` uses the ``pstacks`` output files to make a better catalog, as opposed to using one where we clustered de novo.\n",
    "\n",
    "#### 11. ``sstacks``\n",
    "This second round of ``sstacks`` genotypes indiviuals using the new ``cstacks`` catalog.\n",
    "\n",
    "#### 12. ``populations``\n",
    "``populations`` assigns individuals to populations and produces a genepop file and other summary statistics files. This second round of ``populations`` is much cleaner and more conservative.\n",
    "\n",
    "#### 13. Additional filtering with Marine's scripts\n",
    "Our postdoc Marine made additional filtering scripts for after the second ``populations`` to \n",
    "\n",
    "#### 14. Statistial tests in R\n",
    "\n",
    "\n",
    "<br>\n",
    "<br>\n",
    "<br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Go to working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[Errno 2] No such file or directory: '/Volumes/Time Machine Backups/Cod-Time-Series-Data/'\n",
      "/Users/natalielowell/Git-repos/FISH546/Cod-Time-Series-Project/Notebooks\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Time\\ Machine\\ Backups/Cod-Time-Series-Data/ "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1. Adding library identifier to file names\n",
    "\n",
    "If you are analyzing data run on multiple lanes, it may be useful to rename your files such that they have the unique library identifier (eg., \\_L1 or \\_L2) because barcodes will be redundant between libraries. I wrote a [script](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Scripts/add_lib_to_filename.py) that will add this to your filenames.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!python add_lib_to_filename.py process_radtags_out/cod_lib1 _L1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!python add_lib_to_filename.py process_radtags_out/cod_lib2 _L2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2. Running ``ustacks``\n",
    "\n",
    "``ustacks`` [documentation](http://catchenlab.life.illinois.edu/stacks/comp/ustacks.php) highlights:\n",
    "\n",
    "<br>ustacks -t file_type -f file_path [-d] [-r] [-o path] [-i id] [-m min_cov] [-M max_dist] [-p num_threads] [-R] [-H] [-h]\n",
    "<br>t — input file Type. Supported types: fasta, fastq, gzfasta, or gzfastq.\n",
    "<br>f — input file path.\n",
    "<br>o — output path to write results.\n",
    "<br>i — SQL ID to insert into the output to identify this sample.\n",
    "<br>m — Minimum depth of coverage required to create a stack (default 2).\n",
    "<br>M — Maximum distance (in nucleotides) allowed between stacks (default 2).\n",
    "<br>N — Maximum distance allowed to align secondary reads to primary stacks (default: M + 2).\n",
    "<br>R — retain unused reads.\n",
    "<br>H — disable calling haplotypes from secondary reads.\n",
    "<br>p — enable parallel execution with num_threads threads.\n",
    "<br>h — display this help messsage.\n",
    "\n",
    "<br>\n",
    "<br>\n",
    "Running custom python [script for ``ustacks``](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Scripts/pypipe_ustacks.py):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!python pypipe_ustacks.py barcodes_samplenames.txt ./process_radtags_out ./ustacks_out"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3. Running ``cstacks``\n",
    "``cstacks`` [documentation](http://catchenlab.life.illinois.edu/stacks/comp/cstacks.php) highlights:\n",
    "\n",
    "<br>cstacks -b batch_id -s sample_file [-s sample_file_2 ...] [-o path] [-n num] [-g] [-p num_threads] [--catalog path] [-h]\n",
    "<br>p — enable parallel execution with num_threads threads.\n",
    "<br>b — MySQL ID of this batch.\n",
    "<br>s — TSV file from which to load radtags.\n",
    "<br>o — output path to write results.\n",
    "<br>m — include tags in the catalog that match to more than one entry.\n",
    "<br>n — number of mismatches allowed between sample tags when generating the catalog.\n",
    "<br>g — base catalog matching on genomic location, not sequence identity.\n",
    "<br>h — display this help messsage.\n",
    "\n",
    "<br>\n",
    "<br>\n",
    "Running custom python [script for ``cstacks``](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Scripts/pypipe_cstacks.py):\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Time\\ Machine\\ Backups/Cod-Time-Series-Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cstacks paramters selected:\n",
      "  Loci matched based on sequence identity.\n",
      "  Number of mismatches allowed between stacks: 3\n",
      "  Gapped alignments: disabled\n",
      "Constructing catalog from 10 samples.\n",
      "Initializing new catalog...\n",
      "  Parsing stacks_b3/2015_101_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2015_101_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2015_101_1.alleles.tsv.gz\n",
      "  37987 loci were newly added to the catalog.\n",
      "Processing sample stacks_b3/2015_101_1 [2 of 10]\n",
      "  Parsing stacks_b3/2005_464_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_464_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_464_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "  37987 loci in the catalog, 3925472 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  34143 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  4583 loci were newly added to the catalog.\n",
      "  185 loci matched more than one catalog locus and were excluded.\n",
      "Processing sample stacks_b3/2005_464_1 [3 of 10]\n",
      "  Parsing stacks_b3/2010_184_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_184_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_184_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "  42570 loci in the catalog, 4336822 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  35970 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  2526 loci were newly added to the catalog.\n",
      "  167 loci matched more than one catalog locus and were excluded.\n",
      "Processing sample stacks_b3/2010_184_1 [4 of 10]\n",
      "  Parsing stacks_b3/2005_459_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_459_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_459_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "  45096 loci in the catalog, 4541782 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  36655 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  1741 loci were newly added to the catalog.\n",
      "  198 loci matched more than one catalog locus and were excluded.\n",
      "Processing sample stacks_b3/2005_459_1 [5 of 10]\n",
      "  Parsing stacks_b3/2010_244_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_244_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_244_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "  46837 loci in the catalog, 4691662 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  36145 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  2415 loci were newly added to the catalog.\n",
      "  179 loci matched more than one catalog locus and were excluded.\n",
      "Processing sample stacks_b3/2010_244_1 [6 of 10]\n",
      "  Parsing stacks_b3/2014_223_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2014_223_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2014_223_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "  49252 loci in the catalog, 4913922 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  36470 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  1319 loci were newly added to the catalog.\n",
      "  181 loci matched more than one catalog locus and were excluded.\n",
      "Processing sample stacks_b3/2014_223_1 [7 of 10]\n",
      " Unable to open 'stacks_b3/2014_174_1'\n",
      "Failed to load sample 7\n",
      "Processing sample stacks_b3/2014_174_1 [7 of 10]\n",
      "  Parsing stacks_b3/2015_103_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2015_103_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2015_103_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "  50571 loci in the catalog, 5025064 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  36229 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  986 loci were newly added to the catalog.\n",
      "  161 loci matched more than one catalog locus and were excluded.\n",
      "Processing sample stacks_b3/2015_103_1 [8 of 10]\n",
      "  Parsing stacks_b3/2009_176_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_176_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_176_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "  51557 loci in the catalog, 5107165 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  32303 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  749 loci were newly added to the catalog.\n",
      "  156 loci matched more than one catalog locus and were excluded.\n",
      "Processing sample stacks_b3/2009_176_1 [9 of 10]\n",
      "  Parsing stacks_b3/2009_135_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_135_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_135_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  Distance allowed between stacks: 3; searching with a k-mer length of 35 (110 k-mers per read); 5 k-mer hits required.\n",
      "  52306 loci in the catalog, 5227444 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  12404 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  727 loci were newly added to the catalog.\n",
      "  69 loci matched more than one catalog locus and were excluded.\n",
      "Writing catalog to 'stacks_b3/... done.\n"
     ]
    }
   ],
   "source": [
    "!python pypipe_cstacks.py prt_out_filenames.txt stacks_b3 10 3 stacks_b3 3 5"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4. Running ``sstacks``\n",
    "``sstacks`` [documentation](http://catchenlab.life.illinois.edu/stacks/comp/sstacks.php) highlights:\n",
    "\n",
    "<br>sstacks -b batch_id -c catalog_file -s sample_file [-s sample_file_2 ...] [-o path] [-p num_threads] [-g] [-x] [-v] [-h]\n",
    "<br>p — enable parallel execution with num_threads threads.\n",
    "<br>b — MySQL ID of this batch.\n",
    "<br>c — TSV file from which to load the catalog loci.\n",
    "<br>s — TSV file from which to load sample loci.\n",
    "<br>o — output path to write results.\n",
    "<br>g — base matching on genomic location, not sequence identity.\n",
    "<br>x — don’t verify haplotype of matching locus.\n",
    "<br>v — print program version.\n",
    "<br>h — display this help messsage.\n",
    "\n",
    "Running custom python [script for ``sstacks``](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Scripts/pypipe_sstacks.py):\n",
    "\n",
    "\n",
    "<br>WHEN RUNNING THIS SCRIPT, YOUR INPUTS AT THE COMMAND LINE ARE:\n",
    "<br>python  \n",
    "<br>{0}[pipeline filename] \n",
    "<br>{1}[files that came out of process rad tags]\n",
    "<br>{2}[batch ID number]\n",
    "<br>{3}[filepath to directory with catalog filename without file extension]\n",
    "<br>{4}[filepath to directory w ustacks output files per sample]\n",
    "<br>{5}[number of threads to use]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['2005_297_1', '2005_298_1', '2005_299_1', '2005_384_1', '2005_385_1', '2005_387_1', '2005_388_1', '2005_389_1', '2005_457_1', '2005_459_1', '2005_460_1', '2005_461_1', '2005_462_1', '2005_463_1', '2005_464_1', '2005_465_1', '2005_466_1', '2005_467_1', '2005_495_1', '2005_559_1', '2009_123_1', '2009_124_1', '2009_127_1', '2009_128_1', '2009_129_1', '2009_131_1', '2009_135_1', '2009_141_1', '2009_147_1', '2009_164_1', '2009_169_1', '2009_170_1', '2009_171_1', '2009_172_1', '2009_173_1', '2009_174_1', '2009_176_1', '2009_178_1', '2009_532_1', '2009_536_1', '2009_537_1', '2010_114_1', '2010_115_1', '2010_116_1', '2010_117_1', '2010_118_1', '2010_119_1', '2010_123_1', '2010_127_1', '2010_184_1', '2010_185_1', '2010_187_1', '2010_188_1', '2010_189_1', '2010_211_1', '2010_212_1', '2010_213_1', '2010_215_1', '2010_216_1', '2010_218_1', '2010_219_1', '2010_220_1', '2010_221_1', '2010_222_1', '2010_240_1', '2010_242_1', '2010_243_1', '2010_244_1', '2010_246_1', '2010_247_1', '2010_250_1', '2010_251_1', '2010_266_1', '2010_564_1', '2010_565_1', '2010_568_1', '2010_569_1', '2010_572_1', '2010_858_1', '2014_128_1', '2014_129_1', '2014_130_1', '2014_174_1', '2014_203_1', '2014_204_1', '2014_205_1', '2014_207_1', '2014_208_1', '2014_218_1', '2014_223_1', '2014_224_1', '2014_232_1', '2015_016_1', '2015_100_1', '2015_101_1', '2015_102_1', '2015_103_1', '2015_105_1', '2015_138_1', '2015_139_1', '2015_140_1', '2015_141_1']\n",
      "You are working with 102 samples.\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_297_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_297_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_297_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_297_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "34567 stacks compared against the catalog containing 53033 loci.\n",
      "  32551 matching loci, 2904 contained no verified haplotypes.\n",
      "  121 loci matched more than one catalog locus and were excluded.\n",
      "  2783 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  35110 total haplotypes examined from matching loci, 31915 verified.\n",
      "Outputing to file ./2005_297_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_298_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_298_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_298_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_298_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "37751 stacks compared against the catalog containing 53033 loci.\n",
      "  35600 matching loci, 2028 contained no verified haplotypes.\n",
      "  120 loci matched more than one catalog locus and were excluded.\n",
      "  1908 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  38772 total haplotypes examined from matching loci, 36460 verified.\n",
      "Outputing to file ./2005_298_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_299_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_299_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_299_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_299_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38073 stacks compared against the catalog containing 53033 loci.\n",
      "  32628 matching loci, 5318 contained no verified haplotypes.\n",
      "  84 loci matched more than one catalog locus and were excluded.\n",
      "  5234 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  34638 total haplotypes examined from matching loci, 28881 verified.\n",
      "Outputing to file ./2005_299_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_384_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_384_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_384_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_384_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "29161 stacks compared against the catalog containing 53033 loci.\n",
      "  25564 matching loci, 3598 contained no verified haplotypes.\n",
      "  83 loci matched more than one catalog locus and were excluded.\n",
      "  3515 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  26965 total haplotypes examined from matching loci, 23055 verified.\n",
      "Outputing to file ./2005_384_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_385_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_385_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_385_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_385_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "35074 stacks compared against the catalog containing 53033 loci.\n",
      "  30429 matching loci, 4391 contained no verified haplotypes.\n",
      "  89 loci matched more than one catalog locus and were excluded.\n",
      "  4302 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  32070 total haplotypes examined from matching loci, 27330 verified.\n",
      "Outputing to file ./2005_385_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_387_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_387_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_387_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_387_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "20123 stacks compared against the catalog containing 53033 loci.\n",
      "  19696 matching loci, 1190 contained no verified haplotypes.\n",
      "  156 loci matched more than one catalog locus and were excluded.\n",
      "  1034 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  20950 total haplotypes examined from matching loci, 19514 verified.\n",
      "Outputing to file ./2005_387_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_388_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_388_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_388_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_388_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "4452 stacks compared against the catalog containing 53033 loci.\n",
      "  4271 matching loci, 262 contained no verified haplotypes.\n",
      "  68 loci matched more than one catalog locus and were excluded.\n",
      "  194 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  4526 total haplotypes examined from matching loci, 4153 verified.\n",
      "Outputing to file ./2005_388_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_389_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_389_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_389_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_389_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "22099 stacks compared against the catalog containing 53033 loci.\n",
      "  21682 matching loci, 1418 contained no verified haplotypes.\n",
      "  172 loci matched more than one catalog locus and were excluded.\n",
      "  1246 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  23238 total haplotypes examined from matching loci, 21512 verified.\n",
      "Outputing to file ./2005_389_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_457_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_457_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_457_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_457_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39269 stacks compared against the catalog containing 53033 loci.\n",
      "  37661 matching loci, 1263 contained no verified haplotypes.\n",
      "  139 loci matched more than one catalog locus and were excluded.\n",
      "  1124 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41858 total haplotypes examined from matching loci, 40307 verified.\n",
      "Outputing to file ./2005_457_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_459_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_459_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_459_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_459_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38594 stacks compared against the catalog containing 53033 loci.\n",
      "  38540 matching loci, 135 contained no verified haplotypes.\n",
      "  127 loci matched more than one catalog locus and were excluded.\n",
      "  8 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  44186 total haplotypes examined from matching loci, 43885 verified.\n",
      "Outputing to file ./2005_459_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_460_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_460_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_460_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_460_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "43900 stacks compared against the catalog containing 53033 loci.\n",
      "  35905 matching loci, 3261 contained no verified haplotypes.\n",
      "  87 loci matched more than one catalog locus and were excluded.\n",
      "  3174 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  38533 total haplotypes examined from matching loci, 34907 verified.\n",
      "Outputing to file ./2005_460_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_461_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_461_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_461_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_461_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38246 stacks compared against the catalog containing 53033 loci.\n",
      "  33568 matching loci, 5149 contained no verified haplotypes.\n",
      "  100 loci matched more than one catalog locus and were excluded.\n",
      "  5049 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  35799 total haplotypes examined from matching loci, 30207 verified.\n",
      "Outputing to file ./2005_461_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_462_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_462_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_462_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_462_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "37319 stacks compared against the catalog containing 53033 loci.\n",
      "  36152 matching loci, 1392 contained no verified haplotypes.\n",
      "  152 loci matched more than one catalog locus and were excluded.\n",
      "  1240 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  39847 total haplotypes examined from matching loci, 38182 verified.\n",
      "Outputing to file ./2005_462_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_463_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_463_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_463_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_463_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39641 stacks compared against the catalog containing 53033 loci.\n",
      "  37909 matching loci, 1215 contained no verified haplotypes.\n",
      "  118 loci matched more than one catalog locus and were excluded.\n",
      "  1097 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41863 total haplotypes examined from matching loci, 40406 verified.\n",
      "Outputing to file ./2005_463_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_464_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_464_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_464_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_464_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38911 stacks compared against the catalog containing 53033 loci.\n",
      "  38868 matching loci, 116 contained no verified haplotypes.\n",
      "  108 loci matched more than one catalog locus and were excluded.\n",
      "  8 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  44608 total haplotypes examined from matching loci, 44346 verified.\n",
      "Outputing to file ./2005_464_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_465_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_465_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_465_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_465_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "43179 stacks compared against the catalog containing 53033 loci.\n",
      "  37417 matching loci, 1836 contained no verified haplotypes.\n",
      "  92 loci matched more than one catalog locus and were excluded.\n",
      "  1744 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  40583 total haplotypes examined from matching loci, 38414 verified.\n",
      "Outputing to file ./2005_465_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_466_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_466_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_466_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_466_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "43736 stacks compared against the catalog containing 53033 loci.\n",
      "  36597 matching loci, 1544 contained no verified haplotypes.\n",
      "  92 loci matched more than one catalog locus and were excluded.\n",
      "  1452 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  39688 total haplotypes examined from matching loci, 37829 verified.\n",
      "Outputing to file ./2005_466_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_467_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_467_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_467_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_467_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40113 stacks compared against the catalog containing 53033 loci.\n",
      "  37848 matching loci, 1301 contained no verified haplotypes.\n",
      "  98 loci matched more than one catalog locus and were excluded.\n",
      "  1203 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41715 total haplotypes examined from matching loci, 40164 verified.\n",
      "Outputing to file ./2005_467_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_495_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_495_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_495_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_495_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39374 stacks compared against the catalog containing 53033 loci.\n",
      "  37544 matching loci, 1325 contained no verified haplotypes.\n",
      "  114 loci matched more than one catalog locus and were excluded.\n",
      "  1211 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41430 total haplotypes examined from matching loci, 39822 verified.\n",
      "Outputing to file ./2005_495_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2005_559_1' [1 of 1]\n",
      "  Parsing stacks_b3/2005_559_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2005_559_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2005_559_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38614 stacks compared against the catalog containing 53033 loci.\n",
      "  36805 matching loci, 1338 contained no verified haplotypes.\n",
      "  138 loci matched more than one catalog locus and were excluded.\n",
      "  1200 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  40453 total haplotypes examined from matching loci, 38798 verified.\n",
      "Outputing to file ./2005_559_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_123_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_123_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_123_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_123_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "12703 stacks compared against the catalog containing 53033 loci.\n",
      "  10686 matching loci, 1495 contained no verified haplotypes.\n",
      "  43 loci matched more than one catalog locus and were excluded.\n",
      "  1452 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  11045 total haplotypes examined from matching loci, 9442 verified.\n",
      "Outputing to file ./2009_123_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_124_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_124_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_124_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_124_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "27873 stacks compared against the catalog containing 53033 loci.\n",
      "  26922 matching loci, 1392 contained no verified haplotypes.\n",
      "  155 loci matched more than one catalog locus and were excluded.\n",
      "  1237 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  28869 total haplotypes examined from matching loci, 27214 verified.\n",
      "Outputing to file ./2009_124_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_127_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_127_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_127_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_127_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "357 stacks compared against the catalog containing 53033 loci.\n",
      "  282 matching loci, 13 contained no verified haplotypes.\n",
      "  2 loci matched more than one catalog locus and were excluded.\n",
      "  11 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  290 total haplotypes examined from matching loci, 273 verified.\n",
      "Outputing to file ./2009_127_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_128_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_128_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_128_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_128_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "351 stacks compared against the catalog containing 53033 loci.\n",
      "  272 matching loci, 12 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  12 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  277 total haplotypes examined from matching loci, 264 verified.\n",
      "Outputing to file ./2009_128_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_129_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_129_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_129_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_129_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "136 stacks compared against the catalog containing 53033 loci.\n",
      "  113 matching loci, 3 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  3 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  118 total haplotypes examined from matching loci, 115 verified.\n",
      "Outputing to file ./2009_129_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_131_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_131_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_131_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_131_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "759 stacks compared against the catalog containing 53033 loci.\n",
      "  614 matching loci, 34 contained no verified haplotypes.\n",
      "  2 loci matched more than one catalog locus and were excluded.\n",
      "  32 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  625 total haplotypes examined from matching loci, 589 verified.\n",
      "Outputing to file ./2009_131_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_135_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_135_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_135_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_135_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "13200 stacks compared against the catalog containing 53033 loci.\n",
      "  13165 matching loci, 34 contained no verified haplotypes.\n",
      "  34 loci matched more than one catalog locus and were excluded.\n",
      "  0 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  16406 total haplotypes examined from matching loci, 16334 verified.\n",
      "Outputing to file ./2009_135_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_141_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_141_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_141_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_141_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "3779 stacks compared against the catalog containing 53033 loci.\n",
      "  3064 matching loci, 238 contained no verified haplotypes.\n",
      "  15 loci matched more than one catalog locus and were excluded.\n",
      "  223 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  3169 total haplotypes examined from matching loci, 2906 verified.\n",
      "Outputing to file ./2009_141_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_147_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_147_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_147_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_147_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "1148 stacks compared against the catalog containing 53033 loci.\n",
      "  934 matching loci, 64 contained no verified haplotypes.\n",
      "  4 loci matched more than one catalog locus and were excluded.\n",
      "  60 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  957 total haplotypes examined from matching loci, 884 verified.\n",
      "Outputing to file ./2009_147_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_164_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_164_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_164_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_164_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "1639 stacks compared against the catalog containing 53033 loci.\n",
      "  1306 matching loci, 92 contained no verified haplotypes.\n",
      "  11 loci matched more than one catalog locus and were excluded.\n",
      "  81 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  1325 total haplotypes examined from matching loci, 1218 verified.\n",
      "Outputing to file ./2009_164_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_169_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_169_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_169_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_169_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "22 stacks compared against the catalog containing 53033 loci.\n",
      "  16 matching loci, 3 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  3 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  16 total haplotypes examined from matching loci, 13 verified.\n",
      "Outputing to file ./2009_169_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_170_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_170_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_170_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_170_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "319 stacks compared against the catalog containing 53033 loci.\n",
      "  283 matching loci, 12 contained no verified haplotypes.\n",
      "  6 loci matched more than one catalog locus and were excluded.\n",
      "  6 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  295 total haplotypes examined from matching loci, 276 verified.\n",
      "Outputing to file ./2009_170_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_171_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_171_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_171_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_171_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "12577 stacks compared against the catalog containing 53033 loci.\n",
      "  11968 matching loci, 985 contained no verified haplotypes.\n",
      "  97 loci matched more than one catalog locus and were excluded.\n",
      "  888 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  12547 total haplotypes examined from matching loci, 11399 verified.\n",
      "Outputing to file ./2009_171_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_172_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_172_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_172_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_172_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "1064 stacks compared against the catalog containing 53033 loci.\n",
      "  996 matching loci, 55 contained no verified haplotypes.\n",
      "  25 loci matched more than one catalog locus and were excluded.\n",
      "  30 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  1063 total haplotypes examined from matching loci, 972 verified.\n",
      "Outputing to file ./2009_172_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_173_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_173_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_173_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_173_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "3761 stacks compared against the catalog containing 53033 loci.\n",
      "  3572 matching loci, 239 contained no verified haplotypes.\n",
      "  68 loci matched more than one catalog locus and were excluded.\n",
      "  171 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  3788 total haplotypes examined from matching loci, 3449 verified.\n",
      "Outputing to file ./2009_173_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_174_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_174_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_174_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_174_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "15728 stacks compared against the catalog containing 53033 loci.\n",
      "  13614 matching loci, 1916 contained no verified haplotypes.\n",
      "  61 loci matched more than one catalog locus and were excluded.\n",
      "  1855 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  14187 total haplotypes examined from matching loci, 12121 verified.\n",
      "Outputing to file ./2009_174_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_176_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_176_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_176_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_176_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "33208 stacks compared against the catalog containing 53033 loci.\n",
      "  33167 matching loci, 113 contained no verified haplotypes.\n",
      "  113 loci matched more than one catalog locus and were excluded.\n",
      "  0 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  38258 total haplotypes examined from matching loci, 38005 verified.\n",
      "Outputing to file ./2009_176_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_178_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_178_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_178_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_178_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "731 stacks compared against the catalog containing 53033 loci.\n",
      "  621 matching loci, 22 contained no verified haplotypes.\n",
      "  5 loci matched more than one catalog locus and were excluded.\n",
      "  17 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  632 total haplotypes examined from matching loci, 604 verified.\n",
      "Outputing to file ./2009_178_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_532_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_532_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_532_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_532_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "850 stacks compared against the catalog containing 53033 loci.\n",
      "  731 matching loci, 34 contained no verified haplotypes.\n",
      "  9 loci matched more than one catalog locus and were excluded.\n",
      "  25 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  754 total haplotypes examined from matching loci, 710 verified.\n",
      "Outputing to file ./2009_532_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_536_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_536_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_536_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_536_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "1157 stacks compared against the catalog containing 53033 loci.\n",
      "  936 matching loci, 50 contained no verified haplotypes.\n",
      "  7 loci matched more than one catalog locus and were excluded.\n",
      "  43 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  965 total haplotypes examined from matching loci, 904 verified.\n",
      "Outputing to file ./2009_536_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2009_537_1' [1 of 1]\n",
      "  Parsing stacks_b3/2009_537_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2009_537_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2009_537_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "2789 stacks compared against the catalog containing 53033 loci.\n",
      "  2305 matching loci, 149 contained no verified haplotypes.\n",
      "  14 loci matched more than one catalog locus and were excluded.\n",
      "  135 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  2353 total haplotypes examined from matching loci, 2178 verified.\n",
      "Outputing to file ./2009_537_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_114_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_114_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_114_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_114_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40170 stacks compared against the catalog containing 53033 loci.\n",
      "  38063 matching loci, 1304 contained no verified haplotypes.\n",
      "  136 loci matched more than one catalog locus and were excluded.\n",
      "  1168 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42290 total haplotypes examined from matching loci, 40687 verified.\n",
      "Outputing to file ./2010_114_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_115_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_115_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_115_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_115_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39401 stacks compared against the catalog containing 53033 loci.\n",
      "  37570 matching loci, 1155 contained no verified haplotypes.\n",
      "  140 loci matched more than one catalog locus and were excluded.\n",
      "  1015 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41702 total haplotypes examined from matching loci, 40256 verified.\n",
      "Outputing to file ./2010_115_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_116_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_116_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_116_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_116_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38810 stacks compared against the catalog containing 53033 loci.\n",
      "  37348 matching loci, 1140 contained no verified haplotypes.\n",
      "  124 loci matched more than one catalog locus and were excluded.\n",
      "  1016 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41415 total haplotypes examined from matching loci, 39991 verified.\n",
      "Outputing to file ./2010_116_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_117_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_117_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_117_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_117_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40285 stacks compared against the catalog containing 53033 loci.\n",
      "  38210 matching loci, 1216 contained no verified haplotypes.\n",
      "  137 loci matched more than one catalog locus and were excluded.\n",
      "  1079 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42532 total haplotypes examined from matching loci, 40956 verified.\n",
      "Outputing to file ./2010_117_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_118_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_118_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_118_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_118_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40583 stacks compared against the catalog containing 53033 loci.\n",
      "  38500 matching loci, 1253 contained no verified haplotypes.\n",
      "  137 loci matched more than one catalog locus and were excluded.\n",
      "  1116 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42762 total haplotypes examined from matching loci, 41178 verified.\n",
      "Outputing to file ./2010_118_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_119_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_119_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_119_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_119_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40080 stacks compared against the catalog containing 53033 loci.\n",
      "  37985 matching loci, 1242 contained no verified haplotypes.\n",
      "  143 loci matched more than one catalog locus and were excluded.\n",
      "  1099 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42293 total haplotypes examined from matching loci, 40729 verified.\n",
      "Outputing to file ./2010_119_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_123_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_123_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_123_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_123_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39593 stacks compared against the catalog containing 53033 loci.\n",
      "  37834 matching loci, 1234 contained no verified haplotypes.\n",
      "  139 loci matched more than one catalog locus and were excluded.\n",
      "  1095 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41890 total haplotypes examined from matching loci, 40378 verified.\n",
      "Outputing to file ./2010_123_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_127_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_127_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_127_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_127_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40481 stacks compared against the catalog containing 53033 loci.\n",
      "  38409 matching loci, 1276 contained no verified haplotypes.\n",
      "  143 loci matched more than one catalog locus and were excluded.\n",
      "  1133 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42750 total haplotypes examined from matching loci, 41152 verified.\n",
      "Outputing to file ./2010_127_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_184_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_184_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_184_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_184_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38663 stacks compared against the catalog containing 53033 loci.\n",
      "  38625 matching loci, 115 contained no verified haplotypes.\n",
      "  111 loci matched more than one catalog locus and were excluded.\n",
      "  4 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  44169 total haplotypes examined from matching loci, 43900 verified.\n",
      "Outputing to file ./2010_184_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_185_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_185_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_185_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_185_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40010 stacks compared against the catalog containing 53033 loci.\n",
      "  38067 matching loci, 1338 contained no verified haplotypes.\n",
      "  134 loci matched more than one catalog locus and were excluded.\n",
      "  1204 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42206 total haplotypes examined from matching loci, 40518 verified.\n",
      "Outputing to file ./2010_185_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_187_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_187_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_187_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_187_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39772 stacks compared against the catalog containing 53033 loci.\n",
      "  38082 matching loci, 1204 contained no verified haplotypes.\n",
      "  121 loci matched more than one catalog locus and were excluded.\n",
      "  1083 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42364 total haplotypes examined from matching loci, 40833 verified.\n",
      "Outputing to file ./2010_187_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_188_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_188_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_188_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_188_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40324 stacks compared against the catalog containing 53033 loci.\n",
      "  38344 matching loci, 1235 contained no verified haplotypes.\n",
      "  131 loci matched more than one catalog locus and were excluded.\n",
      "  1104 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42572 total haplotypes examined from matching loci, 41006 verified.\n",
      "Outputing to file ./2010_188_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_189_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_189_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_189_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_189_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40315 stacks compared against the catalog containing 53033 loci.\n",
      "  38374 matching loci, 1252 contained no verified haplotypes.\n",
      "  123 loci matched more than one catalog locus and were excluded.\n",
      "  1129 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42664 total haplotypes examined from matching loci, 41112 verified.\n",
      "Outputing to file ./2010_189_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_211_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_211_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_211_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_211_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40187 stacks compared against the catalog containing 53033 loci.\n",
      "  38267 matching loci, 1266 contained no verified haplotypes.\n",
      "  146 loci matched more than one catalog locus and were excluded.\n",
      "  1120 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42620 total haplotypes examined from matching loci, 41013 verified.\n",
      "Outputing to file ./2010_211_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_212_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_212_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_212_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_212_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "41885 stacks compared against the catalog containing 53033 loci.\n",
      "  38597 matching loci, 1283 contained no verified haplotypes.\n",
      "  105 loci matched more than one catalog locus and were excluded.\n",
      "  1178 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42718 total haplotypes examined from matching loci, 41135 verified.\n",
      "Outputing to file ./2010_212_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_213_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_213_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_213_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_213_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "42132 stacks compared against the catalog containing 53033 loci.\n",
      "  38629 matching loci, 1283 contained no verified haplotypes.\n",
      "  120 loci matched more than one catalog locus and were excluded.\n",
      "  1163 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42624 total haplotypes examined from matching loci, 41036 verified.\n",
      "Outputing to file ./2010_213_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_215_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_215_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_215_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_215_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "41141 stacks compared against the catalog containing 53033 loci.\n",
      "  38356 matching loci, 1318 contained no verified haplotypes.\n",
      "  114 loci matched more than one catalog locus and were excluded.\n",
      "  1204 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42242 total haplotypes examined from matching loci, 40615 verified.\n",
      "Outputing to file ./2010_215_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_216_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_216_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_216_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_216_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40940 stacks compared against the catalog containing 53033 loci.\n",
      "  38247 matching loci, 1280 contained no verified haplotypes.\n",
      "  103 loci matched more than one catalog locus and were excluded.\n",
      "  1177 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42176 total haplotypes examined from matching loci, 40633 verified.\n",
      "Outputing to file ./2010_216_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_218_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_218_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_218_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_218_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40329 stacks compared against the catalog containing 53033 loci.\n",
      "  37681 matching loci, 1220 contained no verified haplotypes.\n",
      "  119 loci matched more than one catalog locus and were excluded.\n",
      "  1101 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41589 total haplotypes examined from matching loci, 40076 verified.\n",
      "Outputing to file ./2010_218_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_219_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_219_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_219_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_219_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "48278 stacks compared against the catalog containing 53033 loci.\n",
      "  38582 matching loci, 1241 contained no verified haplotypes.\n",
      "  125 loci matched more than one catalog locus and were excluded.\n",
      "  1116 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42626 total haplotypes examined from matching loci, 41074 verified.\n",
      "Outputing to file ./2010_219_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_220_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_220_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_220_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_220_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "43243 stacks compared against the catalog containing 53033 loci.\n",
      "  38694 matching loci, 1315 contained no verified haplotypes.\n",
      "  116 loci matched more than one catalog locus and were excluded.\n",
      "  1199 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42556 total haplotypes examined from matching loci, 40926 verified.\n",
      "Outputing to file ./2010_220_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_221_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_221_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_221_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_221_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39958 stacks compared against the catalog containing 53033 loci.\n",
      "  37365 matching loci, 1204 contained no verified haplotypes.\n",
      "  126 loci matched more than one catalog locus and were excluded.\n",
      "  1078 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41263 total haplotypes examined from matching loci, 39758 verified.\n",
      "Outputing to file ./2010_221_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_222_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_222_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_222_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_222_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "41266 stacks compared against the catalog containing 53033 loci.\n",
      "  37792 matching loci, 1264 contained no verified haplotypes.\n",
      "  109 loci matched more than one catalog locus and were excluded.\n",
      "  1155 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41626 total haplotypes examined from matching loci, 40069 verified.\n",
      "Outputing to file ./2010_222_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_240_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_240_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_240_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_240_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40789 stacks compared against the catalog containing 53033 loci.\n",
      "  37771 matching loci, 1190 contained no verified haplotypes.\n",
      "  106 loci matched more than one catalog locus and were excluded.\n",
      "  1084 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41670 total haplotypes examined from matching loci, 40193 verified.\n",
      "Outputing to file ./2010_240_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_242_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_242_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_242_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_242_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "41119 stacks compared against the catalog containing 53033 loci.\n",
      "  37500 matching loci, 1276 contained no verified haplotypes.\n",
      "  109 loci matched more than one catalog locus and were excluded.\n",
      "  1167 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41208 total haplotypes examined from matching loci, 39694 verified.\n",
      "Outputing to file ./2010_242_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_243_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_243_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_243_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_243_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39726 stacks compared against the catalog containing 53033 loci.\n",
      "  37496 matching loci, 1283 contained no verified haplotypes.\n",
      "  120 loci matched more than one catalog locus and were excluded.\n",
      "  1163 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41259 total haplotypes examined from matching loci, 39681 verified.\n",
      "Outputing to file ./2010_243_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_244_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_244_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_244_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_244_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38739 stacks compared against the catalog containing 53033 loci.\n",
      "  38692 matching loci, 127 contained no verified haplotypes.\n",
      "  124 loci matched more than one catalog locus and were excluded.\n",
      "  3 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  44265 total haplotypes examined from matching loci, 43984 verified.\n",
      "Outputing to file ./2010_244_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_246_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_246_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_246_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_246_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "34971 stacks compared against the catalog containing 53033 loci.\n",
      "  33716 matching loci, 2258 contained no verified haplotypes.\n",
      "  121 loci matched more than one catalog locus and were excluded.\n",
      "  2137 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  36482 total haplotypes examined from matching loci, 33936 verified.\n",
      "Outputing to file ./2010_246_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_247_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_247_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_247_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_247_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38762 stacks compared against the catalog containing 53033 loci.\n",
      "  36974 matching loci, 1206 contained no verified haplotypes.\n",
      "  129 loci matched more than one catalog locus and were excluded.\n",
      "  1077 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  40746 total haplotypes examined from matching loci, 39268 verified.\n",
      "Outputing to file ./2010_247_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_250_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_250_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_250_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_250_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "37449 stacks compared against the catalog containing 53033 loci.\n",
      "  35108 matching loci, 2846 contained no verified haplotypes.\n",
      "  96 loci matched more than one catalog locus and were excluded.\n",
      "  2750 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  37774 total haplotypes examined from matching loci, 34600 verified.\n",
      "Outputing to file ./2010_250_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_251_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_251_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_251_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_251_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "35373 stacks compared against the catalog containing 53033 loci.\n",
      "  33501 matching loci, 2478 contained no verified haplotypes.\n",
      "  110 loci matched more than one catalog locus and were excluded.\n",
      "  2368 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  35922 total haplotypes examined from matching loci, 33156 verified.\n",
      "Outputing to file ./2010_251_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_266_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_266_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_266_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_266_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "35964 stacks compared against the catalog containing 53033 loci.\n",
      "  33844 matching loci, 3019 contained no verified haplotypes.\n",
      "  110 loci matched more than one catalog locus and were excluded.\n",
      "  2909 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  36345 total haplotypes examined from matching loci, 33001 verified.\n",
      "Outputing to file ./2010_266_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_564_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_564_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_564_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_564_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "43126 stacks compared against the catalog containing 53033 loci.\n",
      "  37895 matching loci, 1839 contained no verified haplotypes.\n",
      "  100 loci matched more than one catalog locus and were excluded.\n",
      "  1739 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41244 total haplotypes examined from matching loci, 39128 verified.\n",
      "Outputing to file ./2010_564_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_565_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_565_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_565_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_565_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39859 stacks compared against the catalog containing 53033 loci.\n",
      "  36940 matching loci, 2198 contained no verified haplotypes.\n",
      "  100 loci matched more than one catalog locus and were excluded.\n",
      "  2098 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  40143 total haplotypes examined from matching loci, 37627 verified.\n",
      "Outputing to file ./2010_565_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_568_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_568_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_568_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_568_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39421 stacks compared against the catalog containing 53033 loci.\n",
      "  36499 matching loci, 2467 contained no verified haplotypes.\n",
      "  108 loci matched more than one catalog locus and were excluded.\n",
      "  2359 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  39588 total haplotypes examined from matching loci, 36784 verified.\n",
      "Outputing to file ./2010_568_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_569_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_569_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_569_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_569_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "41477 stacks compared against the catalog containing 53033 loci.\n",
      "  37929 matching loci, 2022 contained no verified haplotypes.\n",
      "  92 loci matched more than one catalog locus and were excluded.\n",
      "  1930 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41505 total haplotypes examined from matching loci, 39177 verified.\n",
      "Outputing to file ./2010_569_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_572_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_572_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_572_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_572_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "41321 stacks compared against the catalog containing 53033 loci.\n",
      "  37632 matching loci, 1769 contained no verified haplotypes.\n",
      "  97 loci matched more than one catalog locus and were excluded.\n",
      "  1672 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  40955 total haplotypes examined from matching loci, 38892 verified.\n",
      "Outputing to file ./2010_572_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2010_858_1' [1 of 1]\n",
      "  Parsing stacks_b3/2010_858_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2010_858_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2010_858_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39878 stacks compared against the catalog containing 53033 loci.\n",
      "  37070 matching loci, 1839 contained no verified haplotypes.\n",
      "  116 loci matched more than one catalog locus and were excluded.\n",
      "  1723 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  40316 total haplotypes examined from matching loci, 38159 verified.\n",
      "Outputing to file ./2010_858_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2014_128_1' [1 of 1]\n",
      "  Parsing stacks_b3/2014_128_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2014_128_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2014_128_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39595 stacks compared against the catalog containing 53033 loci.\n",
      "  37595 matching loci, 1210 contained no verified haplotypes.\n",
      "  132 loci matched more than one catalog locus and were excluded.\n",
      "  1078 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41804 total haplotypes examined from matching loci, 40293 verified.\n",
      "Outputing to file ./2014_128_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2014_129_1' [1 of 1]\n",
      "  Parsing stacks_b3/2014_129_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2014_129_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2014_129_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "37970 stacks compared against the catalog containing 53033 loci.\n",
      "  36626 matching loci, 1152 contained no verified haplotypes.\n",
      "  131 loci matched more than one catalog locus and were excluded.\n",
      "  1021 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  40426 total haplotypes examined from matching loci, 38989 verified.\n",
      "Outputing to file ./2014_129_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2014_130_1' [1 of 1]\n",
      "  Parsing stacks_b3/2014_130_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2014_130_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2014_130_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39457 stacks compared against the catalog containing 53033 loci.\n",
      "  37399 matching loci, 1222 contained no verified haplotypes.\n",
      "  121 loci matched more than one catalog locus and were excluded.\n",
      "  1101 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41429 total haplotypes examined from matching loci, 39925 verified.\n",
      "Outputing to file ./2014_130_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2014_174_1' [1 of 1]\n",
      " Unable to open 'stacks_b3/2014_174_1'\n",
      "Unable to parse 'stacks_b3/2014_174_1'\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2014_203_1' [1 of 1]\n",
      "  Parsing stacks_b3/2014_203_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2014_203_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2014_203_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "37093 stacks compared against the catalog containing 53033 loci.\n",
      "  35551 matching loci, 1280 contained no verified haplotypes.\n",
      "  147 loci matched more than one catalog locus and were excluded.\n",
      "  1133 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  38942 total haplotypes examined from matching loci, 37375 verified.\n",
      "Outputing to file ./2014_203_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2014_204_1' [1 of 1]\n",
      "  Parsing stacks_b3/2014_204_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2014_204_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2014_204_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "41592 stacks compared against the catalog containing 53033 loci.\n",
      "  35455 matching loci, 1302 contained no verified haplotypes.\n",
      "  127 loci matched more than one catalog locus and were excluded.\n",
      "  1175 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  38908 total haplotypes examined from matching loci, 37338 verified.\n",
      "Outputing to file ./2014_204_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2014_205_1' [1 of 1]\n",
      "  Parsing stacks_b3/2014_205_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2014_205_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2014_205_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "37475 stacks compared against the catalog containing 53033 loci.\n",
      "  35729 matching loci, 1294 contained no verified haplotypes.\n",
      "  120 loci matched more than one catalog locus and were excluded.\n",
      "  1174 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  39178 total haplotypes examined from matching loci, 37651 verified.\n",
      "Outputing to file ./2014_205_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2014_207_1' [1 of 1]\n",
      "  Parsing stacks_b3/2014_207_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2014_207_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2014_207_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "37337 stacks compared against the catalog containing 53033 loci.\n",
      "  35857 matching loci, 1257 contained no verified haplotypes.\n",
      "  126 loci matched more than one catalog locus and were excluded.\n",
      "  1131 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  39407 total haplotypes examined from matching loci, 37924 verified.\n",
      "Outputing to file ./2014_207_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2014_208_1' [1 of 1]\n",
      "  Parsing stacks_b3/2014_208_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2014_208_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2014_208_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "37327 stacks compared against the catalog containing 53033 loci.\n",
      "  35971 matching loci, 1292 contained no verified haplotypes.\n",
      "  149 loci matched more than one catalog locus and were excluded.\n",
      "  1143 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  39668 total haplotypes examined from matching loci, 38113 verified.\n",
      "Outputing to file ./2014_208_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2014_218_1' [1 of 1]\n",
      "  Parsing stacks_b3/2014_218_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2014_218_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2014_218_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "36435 stacks compared against the catalog containing 53033 loci.\n",
      "  35200 matching loci, 1365 contained no verified haplotypes.\n",
      "  146 loci matched more than one catalog locus and were excluded.\n",
      "  1219 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  38683 total haplotypes examined from matching loci, 37063 verified.\n",
      "Outputing to file ./2014_218_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2014_223_1' [1 of 1]\n",
      "  Parsing stacks_b3/2014_223_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2014_223_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2014_223_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "37970 stacks compared against the catalog containing 53033 loci.\n",
      "  37924 matching loci, 130 contained no verified haplotypes.\n",
      "  126 loci matched more than one catalog locus and were excluded.\n",
      "  4 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  43332 total haplotypes examined from matching loci, 43033 verified.\n",
      "Outputing to file ./2014_223_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2014_224_1' [1 of 1]\n",
      "  Parsing stacks_b3/2014_224_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2014_224_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2014_224_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "34952 stacks compared against the catalog containing 53033 loci.\n",
      "  33829 matching loci, 1214 contained no verified haplotypes.\n",
      "  146 loci matched more than one catalog locus and were excluded.\n",
      "  1068 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  36901 total haplotypes examined from matching loci, 35441 verified.\n",
      "Outputing to file ./2014_224_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2014_232_1' [1 of 1]\n",
      "  Parsing stacks_b3/2014_232_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2014_232_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2014_232_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38331 stacks compared against the catalog containing 53033 loci.\n",
      "  36693 matching loci, 1321 contained no verified haplotypes.\n",
      "  140 loci matched more than one catalog locus and were excluded.\n",
      "  1181 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  40554 total haplotypes examined from matching loci, 38937 verified.\n",
      "Outputing to file ./2014_232_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2015_016_1' [1 of 1]\n",
      "  Parsing stacks_b3/2015_016_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2015_016_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2015_016_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38497 stacks compared against the catalog containing 53033 loci.\n",
      "  36868 matching loci, 1080 contained no verified haplotypes.\n",
      "  144 loci matched more than one catalog locus and were excluded.\n",
      "  936 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  40765 total haplotypes examined from matching loci, 39394 verified.\n",
      "Outputing to file ./2015_016_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2015_100_1' [1 of 1]\n",
      "  Parsing stacks_b3/2015_100_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2015_100_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2015_100_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "36438 stacks compared against the catalog containing 53033 loci.\n",
      "  35275 matching loci, 1066 contained no verified haplotypes.\n",
      "  151 loci matched more than one catalog locus and were excluded.\n",
      "  915 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  38724 total haplotypes examined from matching loci, 37393 verified.\n",
      "Outputing to file ./2015_100_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2015_101_1' [1 of 1]\n",
      "  Parsing stacks_b3/2015_101_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2015_101_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2015_101_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "37987 stacks compared against the catalog containing 53033 loci.\n",
      "  37987 matching loci, 10 contained no verified haplotypes.\n",
      "  10 loci matched more than one catalog locus and were excluded.\n",
      "  0 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  43470 total haplotypes examined from matching loci, 43436 verified.\n",
      "Outputing to file ./2015_101_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2015_102_1' [1 of 1]\n",
      "  Parsing stacks_b3/2015_102_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2015_102_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2015_102_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39035 stacks compared against the catalog containing 53033 loci.\n",
      "  37157 matching loci, 1267 contained no verified haplotypes.\n",
      "  145 loci matched more than one catalog locus and were excluded.\n",
      "  1122 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  41110 total haplotypes examined from matching loci, 39543 verified.\n",
      "Outputing to file ./2015_102_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2015_103_1' [1 of 1]\n",
      "  Parsing stacks_b3/2015_103_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2015_103_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2015_103_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "37376 stacks compared against the catalog containing 53033 loci.\n",
      "  37336 matching loci, 120 contained no verified haplotypes.\n",
      "  119 loci matched more than one catalog locus and were excluded.\n",
      "  1 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  42488 total haplotypes examined from matching loci, 42217 verified.\n",
      "Outputing to file ./2015_103_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2015_105_1' [1 of 1]\n",
      "  Parsing stacks_b3/2015_105_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2015_105_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2015_105_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38791 stacks compared against the catalog containing 53033 loci.\n",
      "  36956 matching loci, 1157 contained no verified haplotypes.\n",
      "  131 loci matched more than one catalog locus and were excluded.\n",
      "  1026 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  40838 total haplotypes examined from matching loci, 39432 verified.\n",
      "Outputing to file ./2015_105_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2015_138_1' [1 of 1]\n",
      "  Parsing stacks_b3/2015_138_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2015_138_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2015_138_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "36841 stacks compared against the catalog containing 53033 loci.\n",
      "  35466 matching loci, 1351 contained no verified haplotypes.\n",
      "  143 loci matched more than one catalog locus and were excluded.\n",
      "  1208 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  38867 total haplotypes examined from matching loci, 37256 verified.\n",
      "Outputing to file ./2015_138_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2015_139_1' [1 of 1]\n",
      "  Parsing stacks_b3/2015_139_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2015_139_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2015_139_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "38137 stacks compared against the catalog containing 53033 loci.\n",
      "  36480 matching loci, 1321 contained no verified haplotypes.\n",
      "  139 loci matched more than one catalog locus and were excluded.\n",
      "  1182 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  40116 total haplotypes examined from matching loci, 38524 verified.\n",
      "Outputing to file ./2015_139_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2015_140_1' [1 of 1]\n",
      "  Parsing stacks_b3/2015_140_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2015_140_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2015_140_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "37507 stacks compared against the catalog containing 53033 loci.\n",
      "  35941 matching loci, 1256 contained no verified haplotypes.\n",
      "  122 loci matched more than one catalog locus and were excluded.\n",
      "  1134 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  39467 total haplotypes examined from matching loci, 37981 verified.\n",
      "Outputing to file ./2015_140_1.matches.tsv.gz\n",
      "Searching for matches by sequence identity...\n",
      "  Parsing stacks_b3/batch_3.catalog.tags.tsv\n",
      "  Parsing stacks_b3/batch_3.catalog.snps.tsv.gz\n",
      "  Parsing stacks_b3/batch_3.catalog.alleles.tsv.gz\n",
      "Processing sample 'stacks_b3/2015_141_1' [1 of 1]\n",
      "  Parsing stacks_b3/2015_141_1.tags.tsv.gz\n",
      "  Parsing stacks_b3/2015_141_1.snps.tsv.gz\n",
      "  Parsing stacks_b3/2015_141_1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "37501 stacks compared against the catalog containing 53033 loci.\n",
      "  35851 matching loci, 1272 contained no verified haplotypes.\n",
      "  140 loci matched more than one catalog locus and were excluded.\n",
      "  1132 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  39257 total haplotypes examined from matching loci, 37728 verified.\n",
      "Outputing to file ./2015_141_1.matches.tsv.gz\n"
     ]
    }
   ],
   "source": [
    "!python pypipe_sstacks.py prt_out_filenames.txt 3 stacks_b3/batch_3 stacks_b3 10 -o stacks_b3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data/scripts\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Time\\ Machine\\ Backups/Cod-Time-Series-Data/scripts"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 5. Running ``populations``\n",
    "\n",
    "Documentation [here](http://catchenlab.life.illinois.edu/stacks/comp/populations.php).\n",
    "\n",
    "Example code:\n",
    "```\n",
    "!populations -b 2 -P ustacks_out -M popmap1.txt -t 10 -r 0.50 -p 2 -m 5 --genepop\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!populations -b 3 -P test_20k_cutoff -M popmap_b3.txt -t 10 -r 0.50 -p 1 -m 10 --genepop"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Counting retained loci after ``populations``:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15365\r\n"
     ]
    }
   ],
   "source": [
    "!python count_unique_loci_gp.py ../test_20k_cutoff/batch_3_20k.genepop"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I retained 15,365 loci. See Jupyter Notebook \"Solving low retained loci problem\" to see how I ended up using this batch of samples (only samples with at least 20k retained loci after ``ustacks``)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 6. Filter with ``Bowtie``\n",
    "\n",
    "<br>\n",
    "#### A. Make a fasta file for ``Bowtie`` \n",
    "\n",
    "Make a fasta file for ``Bowtie`` by running a custom python [script](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Scripts/genBOWTIEfasta.py) that Mary wrote, which requires (1) a text file with the header from the genepop file and (2) unzipped batch.catalog.tags file.\n",
    "\n",
    "[Bowtie Documentation](http://bowtie-bio.sourceforge.net/manual.shtml) highlights:\n",
    "\n",
    "<br>bowtie-build [options]\n",
    "<br>Main arguments\n",
    "\n",
    "reference_in\n",
    "A comma-separated list of FASTA files containing the reference sequences to be aligned to, or, if -c is specified, the sequences themselves. E.g., <reference_in> might be chr1.fa,chr2.fa,chrX.fa,chrY.fa, or, if -c is specified, this might be <br>GGTCATCCT,ACGGGTCGT,CCGTTCTATGCGGCTTA.\n",
    "\n",
    "-f\n",
    "The reference input files (specified as (reference_in)) are FASTA files (usually having extension .fa, .mfa, .fna or similar).\n",
    "\n",
    "ebwt_base\n",
    "The basename of the index files to write. By default, bowtie-build writes files named <br>NAME.1.ebwt, NAME.2.ebwt, NAME.3.ebwt, NAME.4.ebwt, NAME.rev.1.ebwt, and NAME.rev.2.ebwt, <br>where NAME is <ebwt_base>.\n",
    "<br>\n",
    "<br>\n",
    "<br>bowtie [options]\n",
    "<br>Main arguments\n",
    "\n",
    "-v (integer)\n",
    "Report alignments with at most (integer) mismatches. -e and -l options are ignored and quality values have no effect on what alignments are valid. -v is mutually exclusive with -n.\n",
    "\n",
    "-S/--sam\n",
    "Print alignments in SAM format. See the SAM output section of the manual for details. To suppress all SAM headers, use --sam-nohead in addition to -S/--sam.\n",
    "\n",
    "<br>\n",
    "Here, I will make batch_3_loci.txt for this round by manually opening the genepop file in Textwrangler, and cutting and pasting the header line with the tag names into its own text file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data/test_20k_cutoff\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Time\\ Machine\\ Backups/Cod-Time-Series-Data/test_20k_cutoff"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Unzip the .catalog.tags file, which you will use as an argument to make the Bowtie FASTA:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "batch_3.catalog.tags.tsv already exists -- do you wish to overwrite (y or n)? ^C\n"
     ]
    }
   ],
   "source": [
    "!gzip -d batch_3.catalog.tags.tsv.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Run script to make Bowtie FASTA, with first argument being text file you made with header from genepop, and second argument being the unzipped .catalog.tags file:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data/scripts\n"
     ]
    }
   ],
   "source": [
    "cd ../../scripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!python genBOWTIEfasta.py ../test_20k_cutoff/batch_3_loci.txt ../test_20k_cutoff/batch_3.catalog.tags.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make a directory for ``Bowtie`` files and navigate there. Store the software there. I haven't figured out how to run ``Bowtie`` from different directories, so I moved all of the associated output files from previous batches into their own directory, so that I don't get confused when I make my new output files. \n",
    "\n",
    "Then I will use ``Bowtie`` to make a reference index de novo. Also, genBOWTIEfasta.py stores the new file in the same folder you're running the script from, so it may be worth editing the script at some point to direct where it saves. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Time\\ Machine\\ Backups/Cod-Time-Series-Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mkdir Bowtie"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data/Bowtie/bowtie-1.1.2\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Time Machine Backups/Cod-Time-Series-Data/Bowtie/bowtie-1.1.2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make the Bowtie index, which produces some files that are all you need to align to the index."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Settings:\n",
      "  Output files: \"batch_3.*.ebwt\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 5 (one in 32)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  seqsforBOWTIE.fa\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:00\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "bmax according to bmaxDivN setting: 550152\n",
      "Using parameters --bmax 412614 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 412614 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:00\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "  Binary sorting into buckets\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Binary sorting into buckets time: 00:00:00\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 7; iterating...\n",
      "  Binary sorting into buckets\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Binary sorting into buckets time: 00:00:00\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 0; iterating...\n",
      "  Binary sorting into buckets\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Binary sorting into buckets time: 00:00:00\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 1; iterating...\n",
      "  Binary sorting into buckets\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Binary sorting into buckets time: 00:00:01\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 275075 (target: 412613)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 8\n",
      "  Reserving size (412614) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 366229\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 366230\n",
      "Getting block 2 of 8\n",
      "  Reserving size (412614) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 175515\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 175516\n",
      "Getting block 3 of 8\n",
      "  Reserving size (412614) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 401306\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 401307\n",
      "Getting block 4 of 8\n",
      "  Reserving size (412614) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 234224\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 234225\n",
      "Getting block 5 of 8\n",
      "  Reserving size (412614) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 286995\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 286996\n",
      "Getting block 6 of 8\n",
      "  Reserving size (412614) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 127262\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 127263\n",
      "Getting block 7 of 8\n",
      "  Reserving size (412614) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 404297\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 404298\n",
      "Getting block 8 of 8\n",
      "  Reserving size (412614) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 204773\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 204774\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 545484\n",
      "fchr[G]: 1095609\n",
      "fchr[T]: 1687665\n",
      "fchr[$]: 2200608\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 5154824 bytes to primary EBWT file: batch_3.1.ebwt\n",
      "Wrote 275084 bytes to secondary EBWT file: batch_3.2.ebwt\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 2200608\n",
      "    bwtLen: 2200609\n",
      "    sz: 550152\n",
      "    bwtSz: 550153\n",
      "    lineRate: 6\n",
      "    linesPerSide: 1\n",
      "    offRate: 5\n",
      "    offMask: 0xffffffe0\n",
      "    isaRate: -1\n",
      "    isaMask: 0xffffffff\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 68770\n",
      "    offsSz: 275080\n",
      "    isaLen: 0\n",
      "    isaSz: 0\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 56\n",
      "    sideBwtLen: 224\n",
      "    numSidePairs: 4913\n",
      "    numSides: 9826\n",
      "    numLines: 9826\n",
      "    ebwtTotLen: 628864\n",
      "    ebwtTotSz: 628864\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:00:02\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:00\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "bmax according to bmaxDivN setting: 550152\n",
      "Using parameters --bmax 412614 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 412614 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:00\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "  Binary sorting into buckets\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Binary sorting into buckets time: 00:00:00\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 7; iterating...\n",
      "  Binary sorting into buckets\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Binary sorting into buckets time: 00:00:00\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 314372 (target: 412613)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 7\n",
      "  Reserving size (412614) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 373302\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 373303\n",
      "Getting block 2 of 7\n",
      "  Reserving size (412614) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 91830\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 91831\n",
      "Getting block 3 of 7\n",
      "  Reserving size (412614) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 359294\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:01\n",
      "Returning block of 359295\n",
      "Getting block 4 of 7\n",
      "  Reserving size (412614) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 409761\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 409762\n",
      "Getting block 5 of 7\n",
      "  Reserving size (412614) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 208202\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 208203\n",
      "Getting block 6 of 7\n",
      "  Reserving size (412614) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 355632\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 355633\n",
      "Getting block 7 of 7\n",
      "  Reserving size (412614) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 402581\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 402582\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 545484\n",
      "fchr[G]: 1095609\n",
      "fchr[T]: 1687665\n",
      "fchr[$]: 2200608\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 5154824 bytes to primary EBWT file: batch_3.rev.1.ebwt\n",
      "Wrote 275084 bytes to secondary EBWT file: batch_3.rev.2.ebwt\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 2200608\n",
      "    bwtLen: 2200609\n",
      "    sz: 550152\n",
      "    bwtSz: 550153\n",
      "    lineRate: 6\n",
      "    linesPerSide: 1\n",
      "    offRate: 5\n",
      "    offMask: 0xffffffe0\n",
      "    isaRate: -1\n",
      "    isaMask: 0xffffffff\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 68770\n",
      "    offsSz: 275080\n",
      "    isaLen: 0\n",
      "    isaSz: 0\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 56\n",
      "    sideBwtLen: 224\n",
      "    numSidePairs: 4913\n",
      "    numSides: 9826\n",
      "    numLines: 9826\n",
      "    ebwtTotLen: 628864\n",
      "    ebwtTotSz: 628864\n",
      "    reverse: 0\n",
      "Total time for backward call to driver() for mirror index: 00:00:01\n"
     ]
    }
   ],
   "source": [
    "!./bowtie-build seqsforBOWTIE.fa batch_3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then, align it to itself and filter out any sequences that aligned to sequences other than themselves using a custom [script](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Scripts/parseBowtie_DD.py) that Dan wrote in our lab."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# reads processed: 15282\r\n",
      "# reads with at least one reported alignment: 15282 (100.00%)\r\n",
      "# reads that failed to align: 0 (0.00%)\r\n",
      "Reported 15282 alignments to 1 output stream(s)\r\n"
     ]
    }
   ],
   "source": [
    "!./bowtie -f -v 3 --sam --sam-nohead \\\n",
    "batch_3 \\\n",
    "seqsforBOWTIE.fa \\\n",
    "batch_3_BOWTIEout.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data/scripts\n"
     ]
    }
   ],
   "source": [
    "cd ../../scripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of Bowtie output lines read: 15282\n",
      "Number of sequences written to output: 15282\n"
     ]
    }
   ],
   "source": [
    "!python parseBowtie_DD.py ../Bowtie/bowtie-1.1.2/batch_3_BOWTIEout.sam ../Bowtie/bowtie-1.1.2/batch_3_BOWTIEout_filtered.fa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "``Bowtie`` filtering step filtered out 83 loci (15365 - 15282)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 7. Filter with ``BLAST``\n",
    "\n",
    "<br>\n",
    "Change directory to highest project directory, here Cod Time Series Data. Then, make a directory for Blast (if you haven't done so already) and make a Blast database out of the output from Bowtie. This requires me to move the filtered fasta file, which I did manually. \n",
    "\n",
    "I have figured out how to run Blast from anywhere, so I'll make a new directory for this batch and move the Bowtie filetered fasta file there.\n",
    "\n",
    "Then, we'll be filtering out any loci that match other loci equally well or better than to themselves, which is supposed to remove highly repetitive loci like microsatellites that can interfere with our data analysis. This [script](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Scripts/checkBlastResults_DD.py) for Blast filtering was also written by Dan."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mkdir Blast_b3_20k"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data/Blast_b3_20k\n"
     ]
    }
   ],
   "source": [
    "cd Blast_b3_20k"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make ``BLAST`` database:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 12/12/2016 13:38:08\n",
      "New DB name:   /Volumes/Time Machine Backups/Cod-Time-Series-Data/Blast_b3_20k/batch_3_BOWTIEfiltered\n",
      "New DB title:  batch_3_BOWTIEout_filtered.fa\n",
      "Sequence type: Nucleotide\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 15282 sequences in 0.735458 seconds.\n"
     ]
    }
   ],
   "source": [
    "!makeblastdb -in batch_3_BOWTIEout_filtered.fa \\\n",
    "-parse_seqids \\\n",
    "-dbtype nucl \\\n",
    "-out batch_3_BOWTIEfiltered"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "``BLAST`` against itself:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!blastn -query batch_3_BOWTIEout_filtered.fa \\\n",
    "-db batch_3_BOWTIEfiltered \\\n",
    "-out batch_3_BowtieBlastFiltered"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use python script to filter only wanted loci:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data/scripts\n"
     ]
    }
   ],
   "source": [
    "cd ../scripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Identifying which loci are 'good' and 'bad' based on BLAST alignments...\n",
      "Writing 'good' and 'bad' loci to their respective files...\n"
     ]
    }
   ],
   "source": [
    "!python checkBlastResults_DD.py \\\n",
    "../Blast_b3_20k/batch_3_BowtieBlastFiltered \\\n",
    "../Blast_b3_20k/batch_3_BOWTIEout_filtered.fa \\\n",
    "../Blast_b3_20k/batch_3_BowtieBlastFiltered_GOOD.fa \\\n",
    "../Blast_b3_20k/batch_3_BowtieBlastFiltered_BAD.fa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "And now to check how many loci were retained:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data/Blast_b3_20k\n"
     ]
    }
   ],
   "source": [
    "cd ../Blast_b3_20k/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   14961\r\n"
     ]
    }
   ],
   "source": [
    "!grep \">\" batch_3_BowtieBlastFiltered_GOOD.fa | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     321\r\n"
     ]
    }
   ],
   "source": [
    "!grep \">\" batch_3_BowtieBlastFiltered_BAD.fa | wc -l"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Blast filtered out an additional 321, so in total the Bowtie & Blast filter steps removed 404 loci leaving a total of 14961 loci."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "### 8. Create final reference genome with ``bowtie``\n",
    "\n",
    "Lastly, I need to use Bowtie again to build a final Bowtie index using the files cleaned in Blast, and then use Bowtie to align all of my fastq files to the Bowtie index for ``pstacks``. So I manually need to move the output files from Blast to the Bowtie folder."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data/Bowtie/bowtie-1.1.2\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Time Machine Backups/Cod-Time-Series-Data/Bowtie/bowtie-1.1.2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Settings:\n",
      "  Output files: \"batch_3_final_index.*.ebwt\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 5 (one in 32)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  batch_3_BowtieBlastFiltered_GOOD.fa\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:00\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "bmax according to bmaxDivN setting: 538596\n",
      "Using parameters --bmax 403947 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 403947 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:00\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "  Binary sorting into buckets\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Binary sorting into buckets time: 00:00:00\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 6; iterating...\n",
      "  Binary sorting into buckets\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Binary sorting into buckets time: 00:00:01\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 307768 (target: 403946)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 7\n",
      "  Reserving size (403947) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 314220\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 314221\n",
      "Getting block 2 of 7\n",
      "  Reserving size (403947) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 378936\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 378937\n",
      "Getting block 3 of 7\n",
      "  Reserving size (403947) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 378499\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 378500\n",
      "Getting block 4 of 7\n",
      "  Reserving size (403947) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 153534\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 153535\n",
      "Getting block 5 of 7\n",
      "  Reserving size (403947) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 356439\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 356440\n",
      "Getting block 6 of 7\n",
      "  Reserving size (403947) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 379542\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 379543\n",
      "Getting block 7 of 7\n",
      "  Reserving size (403947) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 193208\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 193209\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 531835\n",
      "fchr[G]: 1071658\n",
      "fchr[T]: 1649312\n",
      "fchr[$]: 2154384\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 5134545 bytes to primary EBWT file: batch_3_final_index.1.ebwt\n",
      "Wrote 269304 bytes to secondary EBWT file: batch_3_final_index.2.ebwt\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 2154384\n",
      "    bwtLen: 2154385\n",
      "    sz: 538596\n",
      "    bwtSz: 538597\n",
      "    lineRate: 6\n",
      "    linesPerSide: 1\n",
      "    offRate: 5\n",
      "    offMask: 0xffffffe0\n",
      "    isaRate: -1\n",
      "    isaMask: 0xffffffff\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 67325\n",
      "    offsSz: 269300\n",
      "    isaLen: 0\n",
      "    isaSz: 0\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 56\n",
      "    sideBwtLen: 224\n",
      "    numSidePairs: 4809\n",
      "    numSides: 9618\n",
      "    numLines: 9618\n",
      "    ebwtTotLen: 615552\n",
      "    ebwtTotSz: 615552\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:00:02\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:00\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "bmax according to bmaxDivN setting: 538596\n",
      "Using parameters --bmax 403947 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 403947 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:00\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "  Binary sorting into buckets\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Binary sorting into buckets time: 00:00:00\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 2, merged 6; iterating...\n",
      "  Binary sorting into buckets\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Binary sorting into buckets time: 00:00:00\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 269297 (target: 403946)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 8\n",
      "  Reserving size (403947) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 221858\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 221859\n",
      "Getting block 2 of 8\n",
      "  Reserving size (403947) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 398920\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 398921\n",
      "Getting block 3 of 8\n",
      "  Reserving size (403947) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 311284\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 311285\n",
      "Getting block 4 of 8\n",
      "  Reserving size (403947) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 307721\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 307722\n",
      "Getting block 5 of 8\n",
      "  Reserving size (403947) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 229586\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 229587\n",
      "Getting block 6 of 8\n",
      "  Reserving size (403947) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 179937\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 179938\n",
      "Getting block 7 of 8\n",
      "  Reserving size (403947) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 316675\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 316676\n",
      "Getting block 8 of 8\n",
      "  Reserving size (403947) for bucket\n",
      "  Calculating Z arrays\n",
      "  Calculating Z arrays time: 00:00:00\n",
      "  Entering block accumulator loop:\n",
      "  10%\n",
      "  20%\n",
      "  30%\n",
      "  40%\n",
      "  50%\n",
      "  60%\n",
      "  70%\n",
      "  80%\n",
      "  90%\n",
      "  100%\n",
      "  Block accumulator loop time: 00:00:00\n",
      "  Sorting block of length 188396\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 188397\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 531835\n",
      "fchr[G]: 1071658\n",
      "fchr[T]: 1649312\n",
      "fchr[$]: 2154384\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 5134545 bytes to primary EBWT file: batch_3_final_index.rev.1.ebwt\n",
      "Wrote 269304 bytes to secondary EBWT file: batch_3_final_index.rev.2.ebwt\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 2154384\n",
      "    bwtLen: 2154385\n",
      "    sz: 538596\n",
      "    bwtSz: 538597\n",
      "    lineRate: 6\n",
      "    linesPerSide: 1\n",
      "    offRate: 5\n",
      "    offMask: 0xffffffe0\n",
      "    isaRate: -1\n",
      "    isaMask: 0xffffffff\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 67325\n",
      "    offsSz: 269300\n",
      "    isaLen: 0\n",
      "    isaSz: 0\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 56\n",
      "    sideBwtLen: 224\n",
      "    numSidePairs: 4809\n",
      "    numSides: 9618\n",
      "    numLines: 9618\n",
      "    ebwtTotLen: 615552\n",
      "    ebwtTotSz: 615552\n",
      "    reverse: 0\n",
      "Total time for backward call to driver() for mirror index: 00:00:01\n"
     ]
    }
   ],
   "source": [
    "!./bowtie-build batch_3_BowtieBlastFiltered_GOOD.fa batch_3_final_index"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The fastq files are in a directory within the bowtie directory, and I need to unzip them. I used the following code through a terminal outside jupyter notebook because I wasn't sure if it would crash the kernel.\n",
    "\n",
    "```\n",
    "gunzip *.gz\n",
    "```\n",
    "\n",
    "Then, I can run this python script to make the final SAM file that will feed into ``pstacks``."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You are working with 82 files. Consider checking your code if you expected a different number of files.\n",
      "^CTraceback (most recent call last):\n",
      "  File \"final_bowtie_shell.py\", line 51, in <module>\n",
      "    subprocess.call([\"sh final_bowtie_shell.txt\"], shell = True) # run shell script\n",
      "  File \"//anaconda/lib/python2.7/subprocess.py\", line 523, in call\n",
      "    return Popen(*popenargs, **kwargs).wait()\n",
      "  File \"//anaconda/lib/python2.7/subprocess.py\", line 1392, in wait\n",
      "    pid, sts = _eintr_retry_call(os.waitpid, self.pid, 0)\n",
      "  File \"//anaconda/lib/python2.7/subprocess.py\", line 476, in _eintr_retry_call\n",
      "    return func(*args)\n",
      "KeyboardInterrupt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!python final_bowtie_shell.py prt_out_filenames_20k.txt 3 batch_3_final_index batch_3_final.sam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 9. ``pstacks``\n",
    "\n",
    "``pstacks`` [documentation](http://catchenlab.life.illinois.edu/stacks/comp/pstacks.php) highlights:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "pstacks -t file_type -f file_path [-o path] [-i id] [-m min_cov] [-p num_threads] [-h]\n",
    "<br>t — input file Type. Supported types: bowtie, sam, or bam.\n",
    "<br>f — input file path.\n",
    "<br>o — output path to write results.\n",
    "<br>i — SQL ID to insert into the output to identify this sample.\n",
    "<br>m — minimum depth of coverage to report a stack (default 1).\n",
    "<br>p — enable parallel execution with num_threads threads.\n",
    "<br>h — display this help messsage.\n",
    "<br>--pct_aln [num] — require read alignments to use at least this percentage of the read (default 85%).\n",
    "<br>--keep_sec_alns — keep secondary alignments (default: false, only keep primary alignments).\n",
    "\n",
    "In my custom script, arguments are as follows:\n",
    "python \n",
    "<br>{0}[pypipe_pstacks.py] \n",
    "<br>{1}[number of threads]\n",
    "<br>{2}[file type, either \"sam\" or \"bowtie\"]\n",
    "<br>{3}[input file path]\n",
    "<br>{4}[output path to write results]\n",
    "<br>{5}[SQL ID number]\n",
    "<br>{6}[minimum depth of coverage to report stack, default = 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "u'/Volumes/Time Machine Backups/Cod-Time-Series-Data/Bowtie/bowtie-1.1.2'"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Time Machine Backups/Cod-Time-Series-Data/scripts\n"
     ]
    }
   ],
   "source": [
    "cd ../../scripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Min depth of coverage to report a stack: 3\n",
      "Model type: SNP\n",
      "Alpha significance level for model: 0.05\n",
      "Error opening input file '../bowtie-1.1.2/batch_2_ReferenceGenome.'\n",
      "Parsing ../bowtie-1.1.2/batch_2_ReferenceGenome.\n",
      "Loading aligned sequences...done\n",
      "Error: Unable to load data from '../bowtie-1.1.2/batch_2_ReferenceGenome.'.\n",
      "Finished running pstacks_shell.txt script.\n"
     ]
    }
   ],
   "source": [
    "!python pypipe_pstacks.py 3 bowtie ../bowtie-1.1.2/batch_3_final. ../bowtie-1.1.2 3 3 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 10. ``sstacks``\n",
    "\n",
    "``sstacks`` [documentation](http://catchenlab.life.illinois.edu/stacks/comp/sstacks.php) highlights:\n",
    "\n",
    "<br>sstacks -b batch_id -c catalog_file -s sample_file [-s sample_file_2 ...] [-o path] [-p num_threads] [-g] [-x] [-v] [-h]\n",
    "<br>p — enable parallel execution with num_threads threads.\n",
    "<br>b — MySQL ID of this batch.\n",
    "<br>c — TSV file from which to load the catalog loci.\n",
    "<br>s — TSV file from which to load sample loci.\n",
    "<br>o — output path to write results.\n",
    "<br>g — base matching on genomic location, not sequence identity.\n",
    "<br>x — don’t verify haplotype of matching locus.\n",
    "<br>v — print program version.\n",
    "<br>h — display this help messsage.\n",
    "\n",
    "Running custom python [script for ``sstacks``](https://github.com/nclowell/FISH546/blob/master/Cod-Time-Series-Project/Scripts/pypipe_sstacks.py):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!python pypipe_sstacks.py new_filenames_shell.txt 2 ustacks_out/batch_1 ustacks_out 10 -o sstacks_out"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 11. ``populations``\n",
    "\n",
    "Documentation [here](http://catchenlab.life.illinois.edu/stacks/comp/populations.php).\n",
    "\n",
    "Example code:\n",
    "```\n",
    "!populations -b 2 -P ustacks_out -M popmap1.txt -t 10 -r 0.50 -p 2 -m 5 --genepop\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 12. Additional filtering with Marine's scripts"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "### 13. Statistical tests in R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
